

```{r,echo=FALSE,warning=FALSE}
setwd("/Users/expeditoolimi/Documents/Manuscript-AA-OE_SB/Microbiome_analysis/")
library(phyloseq);packageVersion("phyloseq")#‘1.46.0’
library(vegan);packageVersion("vegan")# ‘2.6.4’
library(genefilter);packageVersion("genefilter")#‘1.84.0’
library(psych);packageVersion("psych")#‘2.4.1’
library(DESeq2);packageVersion("DESeq2")#‘1.42.1’
library(ggplot2);packageVersion("ggplot2")#‘3.5.0’
library(reshape2);packageVersion("reshape2")#‘1.4.4’
library(microbiome);packageVersion("microbiome")#‘1.24.0’
library(ranacapa);packageVersion("ranacapa")#‘0.1.0’
library(S4Vectors);packageVersion("S4Vectors")#‘0.40.2’
library(Biobase);packageVersion("Biobase")#‘2.62.0’
library(BiocGenerics);packageVersion("BiocGenerics")#‘0.48.1’
library(parallel);packageVersion("parallel")#‘4.3.2’
library(ggpubr);packageVersion("ggpubr")#‘0.6.0’
#library(metagMisc)
library(decontam);packageVersion("decontam")#‘1.22.0’
library(RColorBrewer);packageVersion("RColorBrewer")#‘1.1.3’
library(ggpubr);packageVersion("ggpubr")#‘0.6.0’
library(multcompView);packageVersion("multcompView")#‘0.1.10’
library(metagenomeSeq);packageVersion("metagenomeSeq")#‘1.43.0’
#library(devtools)
library(lawstat);packageVersion("lawstat")#‘3.6’
library(DirichletMultinomial);packageVersion("DirichletMultinomial")#‘1.44.0’
library(lattice);packageVersion("lattice")# ‘0.22.5’
library(xtable);packageVersion("xtable")#‘1.8.4’
library(permute);packageVersion("permute")#‘0.9.7’
library(grid);packageVersion("grid")#‘4.3.2’
library("doParallel");packageVersion("doParallel")#‘1.0.17’
#library(igraph)
library(ade4);packageVersion("ade4")#‘1.7.22’
#library(rgl)
library(car);packageVersion("car")#‘3.1.2’
library(gridExtra);packageVersion("gridExtra")# ‘2.3’
library(tidyverse);packageVersion("tidyverse")#‘2.0.0’
library(phia);packageVersion("phia")#‘0.3.1’
library(lme4);packageVersion("lme4")#‘1.1.35.1’
library(broom);packageVersion("broom")#‘1.0.5’
library(AICcmodavg);packageVersion("AICcmodavg")#‘2.3.3’
library(multcomp);packageVersion("multcomp")#‘1.4.25’
library(rcompanion);packageVersion("rcompanion")#‘2.4.35’
library(hrbrthemes);packageVersion("hrbrthemes")#‘0.8.7’
library(viridis);packageVersion("viridis")#‘0.6.5’
library(multcompView);packageVersion("multcompView")#‘0.1.10’#stats
library(ape);packageVersion("ape")#‘5.7.1’#loading the tree
library(microbiomeutilities);packageVersion("microbiomeutilities")#‘1.0.17’#process taxonomy table
library(microViz);packageVersion("microViz")#‘0.12.1’#clean taxonomy table to be cited
library(GUniFrac);packageVersion("GUniFrac")#‘1.8’
library(ggtext) ;packageVersion("ggtext")# ‘0.1.2’for rotated labels on ord_plot() 
library(ggraph);packageVersion("ggraph")# ‘2.2.1’for taxatree_plots()
library(DT);packageVersion("DT")# ‘0.32’for tax_fix_interactive()
library(corncob);packageVersion("corncob")#‘0.4.1’ for beta binomial models in tax_model()
library(VennDiagram);packageVersion("VennDiagram")#‘1.7.3’
library(UpSetR);packageVersion("UpSetR")#‘1.4.0’
library(rstatix);packageVersion("rstatix")#‘0.7.2’
library(ALDEx2);packageVersion("ALDEx2")#[1] ‘1.34.0’
library(cowplot);packageVersion("cowplot")#‘1.1.3’
library(HMP);packageVersion("HMP")#HMP: Hypothesis Testing and Power Calculations for Comparing Metagenomic Samples from HMP #‘2.0.1’
library(dendextend);packageVersion("dendextend")#‘1.17.1’
#citation("rstatix ")
library(microbiome)
library(dplyr)
library(phyloseq)
library(hrbrthemes)
library(gcookbook)
library(tidyverse)
library(patchwork)
#library(MicrobiotaProcess)
library(pairwiseAdonis)
#load devtools
library(devtools)
phyloseq_transform_css <- function(physeq, norm = TRUE, log = TRUE, ...){
  # require(metagenomeSeq)
  MGS <- phyloseq::phyloseq_to_metagenomeSeq(physeq)
  # get the normalized count matrix
  otu_norm <- metagenomeSeq::MRcounts(MGS, norm = norm, log = log, ...)
  # exportMat(datt.norm, file = "tmp.txt")    # norm = TRUE, log = TRUE
  ## save sample statistics (sample scaling factor, quantile value, number of identified features and library size):
  # exportStats(dattM.sf, file = "tmp_stats.txt")
  # Substitue raw abundance to the css-normalized data
  physeq.tr <- physeq
  phyloseq::otu_table(physeq.tr) <- phyloseq::otu_table(otu_norm, taxa_are_rows = T)
  return(physeq.tr)
}

source("Rscript_parwise.adonis.txt")
source("aggregate_top_taxa.txt")
#source("PD.txt")
source("aggregate_rare.txt")
#source("plot_composition.txt")
source("RSCRIPT_parwise.adonis.txt")
#source("plot_hierarchy.R")
```

#installed.packages("rgl")

```{r,echo=FALSE,warning=FALSE}
mycolors=c("grey35","#00AFBB","goldenrod","#0dba2f","#392682","coral3","#8E9CA3","chartreuse4","grey85","#FF2700","bisque4","#0B775E","darkred","#F39B7FB2"
           ,"#FA41CA","steelblue","blue","#D16103", "#C3D7A4", "#52854C","deepskyblue","#293352","#FFDB6D","Yellow","#CC79A7","#392682", "#CAB2D6","maroon"
           ,"orchid1","khaki2","#86486F","#7E6148B2");mycolors

myco=c("black","grey50","orange","springgreen3","brown","deepskyblue","salmon","tan");myco

mycola=c("grey35","chartreuse4","#FE7F9D","darkturquoise", "#F0E442", "#009E73", "#FF0000",
         "Orange","blue","#850dba","green1","#FF6AD5", "#6A3D9A","Yellow","Blue",
         "deeppink","#1C0221","#00AFBB","steelblue","#D16103", "#C3D7A4", "#52854C",
         "#4E84C4","#293352","#FFDB6D","#CC79A7", "#C4961A", "#F4EDCA","dodgerblue2", "#E31A1C", # red
         "green4","#0dba2f", # purple"#FF7F00", # orange
         "black", "gold1", "skyblue2", "#FB9A99", # lt pink "palegreen2",
         "#CAB2D6", # lt purple "#FDBF6F", # lt orange "gray70", "khaki2","maroon", "orchid1", "deeppink1", 
         "blue1", "steelblue4", "Sky Blue",  "grey85","#3B9AB2", "goldenrod", "yellow3",
         "darkorange4", "brown","#999999", "#E69F00", "#56B4E9", "#009E73","#F0E442", 
         "#D55E00", "#E69F00", "#ba0d42",
         "yellow4", "#D55E00", "#CC79A7","darkred" ,
         "#56B4E9","#86486F", "#446455", "bisque4","#F7B1AB",
         "#807182","#E3DDDD", "#A45C3D","coral3","#000099","#E2C59F",
         "#B6C5CC", "#8E9CA3", "#556670","#771C19", "#AA3929", "#E25033", 
         "#F27314", "#F8A31B","#2D2D2D","#91D1C2B2","#DC0000B2", "#7E6148B2",
         "#E64B35B2", "#4DBBD5B2", "#00A087B2", "#3C5488B2","#F39B7FB2","#8491B4B2",
         "#FF2700","#62BBA5","#B58900","#80CDC1","#8073AC","#0072B2","#392682",
         "#FA41CA","#0B775E","#02401B","#F4B5BD","#B6854D","deepskyblue");mycola

mycol = c("bisque4", "deeppink", "chartreuse4","blue","coral3","goldenrod","deepskyblue","darkred","#0dba2f", "Orange","Yellow", "#850dba","Blue", "Sky Blue", "#ba0d42");mycol

theme_set(theme_bw())
pal = "Set1"
scale_colour_discrete <-  function(palname=pal, ...){
  scale_colour_brewer(palette=palname, ...)
}
scale_fill_discrete <-  function(palname=pal, ...){
  scale_fill_brewer(palette=palname, ...)
}
```

#Import data and comprise the phyloseq object

```{r,echo=FALSE,warning=FALSE}
setwd("/Users/expeditoolimi/Documents/Manuscript-AA-OE_SB/Microbiome_analysis")

## Imported otu table
Bacteria_otu_microcosm <- read.table("Bacteria-table.txt", header=TRUE,row.names="OTUID");Bacteria_otu_microcosm
Bacteria_otu_microcosm <- otu_table(Bacteria_otu_microcosm, taxa_are_rows = TRUE);Bacteria_otu_microcosm 

## Imported taxonomy
#Bacteria_tax_microcosm <- read.csv("Taxonomy.csv",header=TRUE,row.names="OTUID");Bacteria_tax_microcosm
Bacteria_tax_microcosm <- read.table("Taxon_Bacteria.txt", header=TRUE, sep="\t",row.names="OTUID");Bacteria_tax_microcosm
Bacteria_tax_microcosm <- as.matrix(Bacteria_tax_microcosm);Bacteria_tax_microcosm
Bacteria_tax_microcosm <- tax_table(Bacteria_tax_microcosm);Bacteria_tax_microcosm

## Imported metadata file
#metadata_Bacteria <- read.table("Mapping_File_16S_Combined.txt", header=TRUE, sep="\t",row.names="SampleID");metadata_Bacteria
metadata_Bacteria <- read.csv("Mapping_File_16S_Combined.csv",header=TRUE,row.names="SampleID");metadata_Bacteria

#rownames(metadata) <- metadata[,1]
metadata_Bacteria <- sample_data(metadata_Bacteria);metadata_Bacteria
#tree

all(rownames(metadata_Bacteria) %in% colnames(Bacteria_otu_microcosm))
#Convert to phyloseq object
Bacteria_mereged <- merge_phyloseq(Bacteria_otu_microcosm,Bacteria_tax_microcosm, metadata_Bacteria);Bacteria_mereged# [ 19705 taxa and 280 samples ]
microbiome::summarize_phyloseq(Bacteria_mereged)

#Filtering
Bacteria_mereged = subset_taxa(Bacteria_mereged, Kingdom=="Bacteria");Bacteria_mereged#   [ 18878 taxa and 280 samples ]
Bacteria_mereged = subset_taxa(Bacteria_mereged, Phylum!="CyanoBacteria");Bacteria_mereged#  [ 18878 taxa and 280 samples ]
Bacteria_mereged = subset_taxa(Bacteria_mereged, Order!="Chloroplast");Bacteria_mereged#  [ 18685 taxa and 280 samples ]
Bacteria_mereged = subset_taxa(Bacteria_mereged, Family!="Mitochondria");Bacteria_mereged# [ 18360 taxa and 280 samples ]
Bacteria_mereged <- prune_taxa(taxa_sums(Bacteria_mereged)>0, Bacteria_mereged);Bacteria_mereged# [ 18357 taxa and 280 samples ]

#Summarize 
microbiome::summarize_phyloseq(Bacteria_mereged)

taxonomy <- as.vector(phyloseq::tax_table(Bacteria_mereged))

sums_Bacteria_mereged <- sample_sums(Bacteria_mereged);sums_Bacteria_mereged
sums_Bacteria_mereged <- data.frame(sums_Bacteria_mereged);sums_Bacteria_mereged
sums_Bacteria_mereged <- as.matrix(sums_Bacteria_mereged);sums_Bacteria_mereged
colSums(sums_Bacteria_mereged) #  9964207      
max(sums_Bacteria_mereged) #177062
min(sums_Bacteria_mereged) # 1387
sums_Bacteria_mereged
save(Bacteria_mereged, file = "/Users/expeditoolimi/Documents/Manuscript-AA-OE_SB/Microbiome_analysis/Bacteria_mereged.RData")
```

#alpha Rarefaction
#https://genoweb.toulouse.inra.fr/~formation/15_FROGS/8-February2017/phyloseq_02_2017.R
#chrome-extension://efaidnbmnnnibpcajpcglclefindmkaj/https://www.gdc-docs.ethz.ch/MDA/handouts/MDA20_PhyloseqFormation_Mahendra_Mariadassou.pdf
#https://evomics.org/wp-content/uploads/2016/01/phyloseq-Lab-01-Answers.html
```{r,echo=FALSE,warning=FALSE}
setwd("/Users/expeditoolimi/Documents/Manuscript-AA-OE_SB/Microbiome_analysis")

Bacteria_mereged_noontrol= subset_samples(Bacteria_mereged, Sample_type %in% c("4_Phyllosphere","2_Soil","3_Rhizosphere"));Bacteria_mereged_noontrol#[ 15421 taxa and 1102 samples ]

p_rar <- ggrare(Bacteria_mereged_noontrol, step = 1387 , color = "Sample_type", se = FALSE)
p_rar <- p_rar + facet_wrap(~Sample_type, ncol = 3) + theme_bw()
p_rar <- p_rar + scale_color_manual(values = mycola)
plot(p_rar)
tiff("p_rar.tiff", units="in", width=8, height=4, res=300)
p_rar.tiff=p_rar+plot_layout(ncol =1 );p_rar.tiff
ggsave("p_rar.tiff", plot = p_rar.tiff)
dev.off()
```

#adopted from https://microbiome.github.io/tutorials/Composition.html
#Experiment (subset experiment one)

```{r,echo=FALSE,warning=FALSE}
setwd("/Users/expeditoolimi/Documents/Manuscript-AA-OE_SB/Microbiome_analysis")
load("/Users/expeditoolimi/Documents/Manuscript-AA-OE_SB/Microbiome_analysis/Bacteria_mereged.RData")
Bacteria_experiment= subset_samples(Bacteria_mereged, Experiment %in% c("One"));Bacteria_experiment#[ 15421 taxa and 1102 samples ]
Bacteria_experiment= subset_samples(Bacteria_experiment, Sample_type %in% c("4_Phyllosphere","3_Rhizosphere","2_Soil"));Bacteria_experiment#[ 15421 taxa and 1102 samples ]

####
Bacteria_experiment_table = as(otu_table(Bacteria_experiment), "matrix")
# Coerce to data.frame
Bacteria_experiment_table = as.data.frame(Bacteria_experiment_table)
write.csv(Bacteria_experiment_table ,file = "Bacteria_experiment_table.csv")

#Taxomic composition
library(ggplot2)
library(MicrobiotaProcess)
citation("MicrobiotaProcess")
classtaxa <- get_taxadf(obj=Bacteria_experiment, taxlevel=3);classtaxa
phylataxa <- get_taxadf(obj=Bacteria_experiment, taxlevel=2);phylataxa
ordertaxa <- get_taxadf(obj=Bacteria_experiment, taxlevel=4);ordertaxa
Famtaxa <- get_taxadf(obj=Bacteria_experiment, taxlevel=5);Famtaxa
Gentaxa <- get_taxadf(obj=Bacteria_experiment, taxlevel=6);Gentaxa

#https://yulab-smu.top/MicrobiotaProcessWorkshop/articles/MicrobiotaProcessWorkshop.html
#####################################################################################################
tiff("Bgen.tiff", units="in", width=12, height=9, res=300)
Bgen <- ggbartax(obj=Gentaxa, facetNames="Treatment_Sample", plotgroup=TRUE, topn=15) +
  xlab("Treatment") +
  ylab("relative abundance (%)") +scale_fill_manual(values=mycolors)+theme(axis.text.x = element_text(angle = 270)) +
  #scale_color_manual(values=c("chartreuse4","bisque4","deepskyblue","Sky Blue","goldenrod", "#ba0d42", "grey85","Orange","coral3","darkred","#0dba2f","#850dba","grey35","Blue","deeppink","blue","Yellow")) + 
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=3))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+ 
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=14),
        axis.title.y = element_text(face="bold", colour="gray33", size=14),
        axis.text.x = element_text(colour="gray33", size=11),
        axis.text.y  = element_text(colour="gray33", size=11))+ labs(tag = "A")+
  theme(legend.position = "bottom")+ 
  labs(fill='Phyla')+
  theme(legend.title = element_text(color = "gray33", size = 12),
        legend.text = element_text(color = "gray33", size = 12))+
  theme(text=element_text(family="sans"))+
  theme(text=element_text(family="sans"))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=12),
        axis.title.y = element_text(face="bold", colour="gray33", size=12),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=12));Bgen
#####################################################################################################
#order
tiff("Bgen.tiff", units="in", width=12, height=9, res=300)
Bgen.tiff=Bgen+plot_layout(ncol =1 );Bgen.tiff
ggsave("Bgen.tiff", plot = Bgen.tiff)

#####################################################################################################
Bord_category <- ggbartax(obj=ordertaxa, facetNames="Sample_type_Pool2", plotgroup=TRUE, topn=20) +
  xlab("Cultivars") +
  ylab("relative abundance (%)") +scale_fill_manual(values=mycolors)+theme(axis.text.x = element_text(angle = 270)) +
  #scale_color_manual(values=c("chartreuse4","bisque4","deepskyblue","Sky Blue","goldenrod", "#ba0d42", "grey85","Orange","coral3","darkred","#0dba2f","#850dba","grey35","Blue","deeppink","blue","Yellow")) + 
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=3))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=0.4))+ 
  theme(axis.title.x = element_text(face="bold", colour="black", size=14),
        axis.title.y = element_text(face="bold", colour="black", size=14),
        axis.text.x = element_text(colour="black", size=11),
        axis.text.y  = element_text(colour="black", size=11))+ labs(tag = "G")+
  theme(legend.position = "bottom")+ 
  labs(fill='Phyla')+
  theme(legend.title = element_text(color = "black", size = 12),
        legend.text = element_text(color = "black", size = 12))+
  theme(text=element_text(family="sans"))+
  theme(text=element_text(family="sans"))+
  theme(axis.title.x = element_text(face="bold", colour="black", size=12),
        axis.title.y = element_text(face="bold", colour="black", size=12),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="black", size=12));Bord_category

#Order
tiff("Bord_category.tiff", units="in", width=12, height=9, res=300)
Bord_category.tiff=Bord_category+plot_layout(ncol = 1);Bord_category.tiff
ggsave("Bfam.tiff", plot = Bord_category.tiff)
dev.off()
#####################################################################################################

```

#PCOA-Bacteria
```{r,echo=FALSE,warning=FALSE, }

setwd("/Users/expeditoolimi/Documents/Manuscript-AA-OE_SB/Microbiome_analysis")
load("/Users/expeditoolimi/Documents/Manuscript-AA-OE_SB/Microbiome_analysis/Bacteria_mereged.RData")

Bacteria_experiment= subset_samples(Bacteria_mereged, Experiment %in% c("One"));Bacteria_experiment#[ 15421 taxa and 1102 samples ]
Bacteria_experiment= subset_samples(Bacteria_experiment, Sample_type %in% c("4_Phyllosphere","3_Rhizosphere","2_Soil"));Bacteria_experiment#[ 15421 taxa and 1102 samples ]
pcoacares_Bacteria <- get_pcoa(obj=Bacteria_experiment, distmethod="bray", method="hellinger");pcoacares_Bacteria#pcoa
sample_data(Bacteria_experiment)
#PCA_pcoa_countries of origign
#Visulizing the result
pcoaplot3 <- ggordpoint(obj=pcoacares_Bacteria, biplot=F, speciesannot=F,
                        factorNames=c("Treatment"), ellipse=F) +
  scale_fill_manual(values=myco)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=2))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=14, face="bold", color = "black"),
        axis.text.y = element_text(size=14, face="bold", color = "black"))+
  theme(text=element_text(size=14,  family="sans"))+theme(legend.position = "")+
  theme(legend.title=element_blank())+labs(tag = "B")+ geom_point(size = 5);pcoaplot3


#PCA_pcoa_countries of origign
#Visulizing the result
#tiff("PCA_PCOA_Region.tiff", units="in", width=12, height=10, res=300)

pcoaplot4 <- ggordpoint(obj=pcoacares_Bacteria, biplot=F, speciesannot=F,
                        factorNames=c("Sample_type"), ellipse=F) +
  scale_fill_manual(values=myco)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=4))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=14, face="bold", color = "black"),
        axis.text.y = element_text(size=14, face="bold", color = "black"))+
  theme(text=element_text(size=14,  family="sans"))+theme(legend.position = "")+
  theme(legend.title=element_blank())+labs(tag = "")+ geom_point(size = 5);pcoaplot4

library(ggpubr)
tiff("pcoaplot3.tiff", units="in", width=10, height=8, res=300)
pcoaplot3.tiff=pcoaplot3+plot_layout(ncol = 1);pcoaplot3.tiff
ggsave("pcoaplot3.tiff", plot = pcoaplot3.tiff)
dev.off()

tiff("pcoaplot4.tiff", units="in", width=10, height=8, res=300)
pcoaplot4.tiff=pcoaplot4+plot_layout(ncol = 1);pcoaplot4.tiff
ggsave("pcoaplot4.tiff", plot = pcoaplot4.tiff)
dev.off()

#####
#subset into compartments
Bacteria_experiment= subset_samples(Bacteria_mereged, Experiment %in% c("One"));Bacteria_experiment#[ 15421 taxa and 1102 samples ]
Bacteria_experiment= subset_samples(Bacteria_experiment, Sample_type %in% c("4_Phyllosphere","3_Rhizosphere","2_Soil"));Bacteria_experiment#[ 15421 taxa and 1102 samples ]
#soil

Bacteria_experiment_soil= subset_samples(Bacteria_experiment, Sample_type %in% c("2_Soil"));Bacteria_experiment_soil#[ 15421 taxa and 1102 samples ]
pcoacares_Bacteria_soil <- get_pcoa(obj=Bacteria_experiment_soil, distmethod="bray", method="hellinger");pcoacares_Bacteria_soil#pcoa

#Visulizing the result
bpcoa_soil <- ggordpoint(obj=pcoacares_Bacteria_soil, biplot=F, speciesannot=F,
                         factorNames=c("Treatment"), ellipse=F) +
  scale_fill_manual(values=myco)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=2))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=12, face="bold", color = "black"),
        axis.text.y = element_text(size=12, face="bold", color = "black"))+
  theme(text=element_text(size=12,  family="sans"))+theme(legend.position = "")+
  theme(legend.title=element_blank())+labs(tag = "A")+ geom_point(size = 5);bpcoa_soil

############
Bacteria_experiment_3_Rhizosphere= subset_samples(Bacteria_experiment, Sample_type %in% c("3_Rhizosphere"));Bacteria_experiment_3_Rhizosphere#[ 15421 taxa and 1102 samples ]
pcoacares_Bacteria_3_Rhizosphere <- get_pcoa(obj=Bacteria_experiment_3_Rhizosphere, distmethod="bray", method="hellinger");pcoacares_Bacteria_3_Rhizosphere#pcoa

#Visulizing the result
bpcoa_3_Rhizosphere <- ggordpoint(obj=pcoacares_Bacteria_3_Rhizosphere, biplot=F, speciesannot=F,
                                factorNames=c("Treatment"), ellipse=F) +
  scale_fill_manual(values=myco)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=2))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=12, face="bold", color = "black"),
        axis.text.y = element_text(size=12, face="bold", color = "black"))+
  theme(text=element_text(size=12,  family="sans"))+theme(legend.position = "")+
  theme(legend.title=element_blank())+labs(tag = "B")+ geom_point(size = 5);bpcoa_3_Rhizosphere

############
Bacteria_experiment_Phyllosphere= subset_samples(Bacteria_experiment, Sample_type %in% c("4_Phyllosphere"));Bacteria_experiment_Phyllosphere#[ 15421 taxa and 1102 samples ]
pcoacares_Bacteria_Phyllosphere <- get_pcoa(obj=Bacteria_experiment_Phyllosphere, distmethod="bray", method="hellinger");pcoacares_Bacteria_Phyllosphere#pcoa

#Visulizing the result
bpcoa_Phyllosphere <- ggordpoint(obj=pcoacares_Bacteria_Phyllosphere, biplot=F, speciesannot=F,
                                 factorNames=c("Treatment"), ellipse=F) +
  scale_fill_manual(values=myco)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=2))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=12, face="bold", color = "black"),
        axis.text.y = element_text(size=12, face="bold", color = "black"))+
  theme(text=element_text(size=12,  family="sans"))+theme(legend.position = "")+
  theme(legend.title=element_blank())+labs(tag = "C")+ geom_point(size = 5);bpcoa_Phyllosphere
#####################################################################################################
#https://rdrr.io/github/adrientaudiere/MiscMetabar/src/R/plot_functions.R
#####################################################################################################
#https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/ 
#Agglomerate to phylum-level and rename
Bacteria_mereged_phylum <- phyloseq::tax_glom(Bacteria_mereged, "Phylum")#[ 38 taxa and 1197 samples ]
phyloseq::taxa_names(Bacteria_mereged_phylum) <- phyloseq::tax_table(Bacteria_mereged_phylum)[, "Phylum"]
phyloseq::otu_table(Bacteria_mereged_phylum)[1:5, 1:5]
sample_data(Bacteria_mereged_phylum)

#Melt and plot
tiff("Bacteria_mereged_phylum.tiff", units="in", width=20, height=15, res=300)
Bacteria_mereged_phylum.tiff=phyloseq::psmelt(Bacteria_mereged_phylum) %>%
  ggplot(data = ., aes(x =Sample_type, y = Abundance)) +
  geom_boxplot(outlier.shape  = NA) +
  geom_jitter(aes(color = OTU), height = 0, width = .2) +
  labs(x = "", y = "Abundance\n") +
  facet_wrap(~ OTU, scales = "free")+
  facet_wrap(~ OTU, scales = "free")+theme(axis.text.x = element_text(angle = 90))+
  scale_color_manual(values=mycola);Bacteria_mereged_phylum.tiff
Bacteria_mereged_phylum.tiff=Bacteria_mereged_phylum.tiff+plot_layout(ncol = );Bacteria_mereged_phylum.tiff
ggsave("Bacteria_mereged_phylum.tiff", plot = Bacteria_mereged_phylum.tiff)
dev.off()

#Melt and plot
tiff("Bacteria_mereged_phylum_treatcayeg.tiff", units="in", width=20, height=15, res=300)
Bacteria_mereged_phylum_treatcayeg.tiff=phyloseq::psmelt(Bacteria_mereged_phylum) %>%
  ggplot(data = ., aes(x =Sample_type_Treatment_type, y = Abundance)) +
  geom_boxplot(outlier.shape  = NA) +
  geom_jitter(aes(color = OTU), height = 0, width = .2) +
  labs(x = "", y = "Abundance\n") +
  facet_wrap(~ OTU, scales = "free")+
  facet_wrap(~ OTU, scales = "free")+theme(axis.text.x = element_text(angle = 90))+
  scale_color_manual(values=mycola);Bacteria_mereged_phylum_treatcayeg.tiff
Bacteria_mereged_phylum_treatcayeg.tiff=Bacteria_mereged_phylum_treatcayeg.tiff+plot_layout(ncol = );Bacteria_mereged_phylum_treatcayeg.tiff
ggsave("Bacteria_mereged_phylum_treatcayeg.tiff", plot = Bacteria_mereged_phylum_treatcayeg.tiff)
dev.off()
########################################################################################################################
#subseting the data for experiment one
Bacteria_microcosm_one<- subset_samples(Bacteria_mereged, Experiment %in% c("One"));Bacteria_microcosm_one#[ 15421 taxa and 1102 samples ]
#remove the control
Bacteria_microcosm_one<- subset_samples(Bacteria_microcosm_one, Sample_type %in% c("4_Phyllosphere","2_Soil","3_Rhizosphere"));Bacteria_microcosm_one#[ 15421 taxa and 1102 samples ]
#####################################################################################################################################################################################################

```


#Bacterial Betadiversity
```{r,echo=FALSE,warning=FALSE}
setwd("/Users/expeditoolimi/Documents/Manuscript-AA-OE_SB/Microbiome_analysis")
load("/Users/expeditoolimi/Documents/Manuscript-AA-OE_SB/Microbiome_analysis/Bacteria_mereged.RData")

#The function
phyloseq_transform_css <- function(physeq, norm = TRUE, log = TRUE, ...){
  # require(metagenomeSeq)
  MGS <- phyloseq::phyloseq_to_metagenomeSeq(physeq)
  # get the normalized count matrix
  otu_norm <- metagenomeSeq::MRcounts(MGS, norm = norm, log = log, ...)
  # exportMat(datt.norm, file = "tmp.txt")    # norm = TRUE, log = TRUE
  ## save sample statistics (sample scaling factor, quantile value, number of identified features and library size):
  # exportStats(dattM.sf, file = "tmp_stats.txt")
  # Substitue raw abundance to the css-normalized data
  physeq.tr <- physeq
  phyloseq::otu_table(physeq.tr) <- phyloseq::otu_table(otu_norm, taxa_are_rows = T)
  return(physeq.tr)
}

##Beta diversity using CSS
Bacteria_microcosm_one_CSS = phyloseq_transform_css(Bacteria_microcosm_one);Bacteria_microcosm_one_CSS#########
sample_data(Bacteria_microcosm_one_CSS)
#First get otu_table and transpose it:
dist.matrix_Bacteria_microcosm_one_CSS <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS)))
sampledf_Bacteria_microcosm_one_CSS <- data.frame(sample_data(Bacteria_microcosm_one_CSS))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS ~ Sample_type*Treatment_type*Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#                             Df SumOfSqs R2       F       Pr(>F)    
#Sample_type                  2   11.622 0.31570 36.2517  0.001 ***
#Treatment_type               2    0.655 0.01779  2.0429  0.012 *  
#Treatment                    5    1.758 0.04776  2.1939  0.001 ***
#Sample_type:Treatment_type   4    1.058 0.02875  1.6507  0.007 ** 
#Sample_type:Treatment       10    2.484 0.06749  1.5499  0.002 ** 
#Residual                   120   19.235 0.52251                   
#Total                      143   36.812 1.00000                   
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 
#############################################################################################################################
#PCOA
#visualize
matrixdistance_Bacteria_microcosm_one_CSS_PCoA <- ordinate(physeq = Bacteria_microcosm_one_CSS, method = "PCoA", distance = "bray", trymax = 1000)
#Stress:     0.2069558 

Fig2=plot_ordination(physeq = Bacteria_microcosm_one_CSS, ordination = matrixdistance_Bacteria_microcosm_one_CSS_PCoA, color = "Treatment",shape = "Sample_type") + 
theme(text = element_text(size = 12)) + scale_color_manual(values =myco)+theme(legend.title=element_blank())+  geom_point(aes(color =Treatment ), size = 7)+
  scale_shape_manual(values=c(19,17, 15,18))+ theme_gray(base_size = 14);Fig2
tiff("Fig2.tiff", units="in", width=11, height=6, res=300)
ggsave("Fig2.tiff", plot = Fig2)
dev.off()
#############################################################################################################################
#Bacteria
Bacteria_microcosm_one_CSS_bact_control= subset_samples(Bacteria_microcosm_one_CSS, Treatment_type %in% c("Bacteria","Control"));Bacteria_microcosm_one_CSS_bact_control#[ 15421 taxa and 1102 samples ]
#First get otu_table and transpose it:
dist.matrix_Bacteria_microcosm_one_CSS_bact_control <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_bact_control)))
sampledf_Bacteria_microcosm_one_CSS_bact_control <- data.frame(sample_data(Bacteria_microcosm_one_CSS_bact_control))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_bact_control <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_bact_control, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_bact_control
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_bact_control), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_bact_control)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_bact_control ~ Treatment_type, data=sampleda, permutation=999);adores#change from adonis to adonis2
#                Df SumOfSqs      R2      F Pr(>F)
#Treatment_type  1   0.2906 0.01582 1.1251  0.262
#Residual       70  18.0799 0.98418              
#Total          71  18.3705 1.00000 
#########################
#Volatiles
Volatile_microcosm_one_CSS_Volatile_control= subset_samples(Bacteria_microcosm_one_CSS, Treatment_type %in% c("Volatile","Control"));Volatile_microcosm_one_CSS_Volatile_control#[ 15421 taxa and 1102 samples ]
#First get otu_table and transpose it:
dist.matrix_Volatile_microcosm_one_CSS_Volatile_control <- t(data.frame(otu_table(Volatile_microcosm_one_CSS_Volatile_control)))
sampledf_Volatile_microcosm_one_CSS_Volatile_control <- data.frame(sample_data(Volatile_microcosm_one_CSS_Volatile_control))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Volatile_microcosm_one_CSS_Volatile_control <- vegdist(dist.matrix_Volatile_microcosm_one_CSS_Volatile_control, method = "bray"); phyloseq_bray_Volatile_microcosm_one_CSS_Volatile_control
sampleda <- data.frame(sample_data(Volatile_microcosm_one_CSS_Volatile_control), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Volatile_microcosm_one_CSS_Volatile_control)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Volatile_microcosm_one_CSS_Volatile_control ~ Treatment_type, data=sampleda, permutation=999);adores#change from adonis to adonis2
#Df SumOfSqs      R2      F Pr(>F)
#Treatment_type  1   0.2459 0.01072 0.9538  0.398
#Residual       88  22.6845 0.98928              
#Total          89  22.9304 1.00000  

#############################################################################################################################
#Subset the different compartments
#Phyllosphere
Bacteria_microcosm_one_CSS_Phyllosphere= subset_samples(Bacteria_microcosm_one_CSS, Sample_type %in% c("4_Phyllosphere"));Bacteria_microcosm_one_CSS_Phyllosphere#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Phyllosphere <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Phyllosphere)))
sampledf_Bacteria_microcosm_one_CSS_Phyllosphere <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Phyllosphere))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Phyllosphere, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Phyllosphere), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs     R2      F Pr(>F)    
#Treatment  7    3.345 0.2557 1.9631  0.001 ***
#Residual  40    9.737 0.7443                  
#Total     47   13.082 1.0000                  
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

#Bacteria vs control in the phyllosphere
Bacteria_microcosm_one_CSS_Phyllosphere_Bact_control= subset_samples(Bacteria_microcosm_one_CSS_Phyllosphere, Pool %in% c("Bacteria","Control"));Bacteria_microcosm_one_CSS_Phyllosphere_Bact_control#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Phyllosphere_Bact_control <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Phyllosphere_Bact_control)))
sampledf_Bacteria_microcosm_one_CSS_Phyllosphere_Bact_control <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Phyllosphere_Bact_control))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_Bact_control <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Phyllosphere_Bact_control, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_Bact_control
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Phyllosphere_Bact_control), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_Bact_control)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_Bact_control ~ Pool, data=sampleda, permutation=999);adores#change from adonis to adonis2
#         Df SumOfSqs      R2      F Pr(>F)  
#Pool      1   0.3858 0.06044 1.4153   0.03 *
#Residual 22   5.9974 0.93956                
#Total    23   6.3832 1.00000                

#Volatile vs control in the phyllosphere
Bacteria_microcosm_one_CSS_Phyllosphere_vola_control= subset_samples(Bacteria_microcosm_one_CSS_Phyllosphere, Pool %in% c("Volatiles","Control"));Bacteria_microcosm_one_CSS_Phyllosphere_vola_control#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Phyllosphere_vola_control <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Phyllosphere_vola_control)))
sampledf_Bacteria_microcosm_one_CSS_Phyllosphere_vola_control <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Phyllosphere_vola_control))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_vola_control <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Phyllosphere_vola_control, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_vola_control
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Phyllosphere_vola_control), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_vola_control)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_vola_control ~ Pool, data=sampleda, permutation=999);adores#change from adonis to adonis2
#         Df SumOfSqs      R2      F Pr(>F)  
#Pool      1   0.3309 0.04044 1.1801  0.098 .
#Residual 28   7.8524 0.95956                
#Total    29   8.1833 1.00000                
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1                

##########
# S. plymuthica HRO C48  vs control_phyllosphere
##########
Bacteria_microcosm_one_CSS_Phyllosphere_ser_cont= subset_samples(Bacteria_microcosm_one_CSS_Phyllosphere, Treatment %in% c("2_Serratia_plymuthica_HRO_C48","1_Control"));Bacteria_microcosm_one_CSS_Phyllosphere_ser_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Phyllosphere_ser_cont <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Phyllosphere_ser_cont)))
Bacteria_microcosm_one_CSS_Phyllosphere_ser_cont <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Phyllosphere_ser_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_ser_cont <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Phyllosphere_ser_cont, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_ser_cont
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Phyllosphere_ser_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_ser_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_ser_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2

#Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.59767 0.19964 2.4943  0.004 **
#Residual  10  2.39612 0.80036                 
#Total     11  2.99379 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
#load devtools

##########
# 3_Stenotrophomonas_rhizophilla_SPA_P69  vs control_phyllosphere
##########
Bacteria_microcosm_one_CSS_Phyllosphere_Steno_cont= subset_samples(Bacteria_microcosm_one_CSS_Phyllosphere, Treatment %in% c("3_Stenotrophomonas_rhizophilla_SPA_P69","1_Control"));Bacteria_microcosm_one_CSS_Phyllosphere_Steno_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Phyllosphere_Steno_cont <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Phyllosphere_Steno_cont)))
Bacteria_microcosm_one_CSS_Phyllosphere_Steno_cont <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Phyllosphere_Steno_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_Steno_cont <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Phyllosphere_Steno_cont, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_Steno_cont
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Phyllosphere_Steno_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_Steno_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_Steno_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#Df SumOfSqs      R2      F Pr(>F)  
#Treatment  1  0.36138 0.11931 1.3548  0.033 *
#Residual  10  2.66746 0.88069                
#Total     11  3.02884 1.00000                
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1


##########
# 4_Bacteria_Mixture  vs control_phyllosphere
##########
Bacteria_microcosm_one_CSS_Phyllosphere_consortium_cont= subset_samples(Bacteria_microcosm_one_CSS_Phyllosphere, Treatment %in% c("4_Bacteria_Mixture","1_Control"));Bacteria_microcosm_one_CSS_Phyllosphere_consortium_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Phyllosphere_consortium_cont <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Phyllosphere_consortium_cont)))
Bacteria_microcosm_one_CSS_Phyllosphere_consortium_cont <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Phyllosphere_consortium_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_consortium_cont <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Phyllosphere_consortium_cont, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_consortium_cont
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Phyllosphere_consortium_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_consortium_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_consortium_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1   0.4036 0.12502 1.4289  0.002 **
#Residual  10   2.8250 0.87498                 
#Total     11   3.2286 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

##########
# 5_3-Methyl-1-Butanol  vs control_phyllosphere
##########
Bacteria_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont= subset_samples(Bacteria_microcosm_one_CSS_Phyllosphere, Treatment %in% c("5_3-Methyl-1-Butanol","1_Control"));Bacteria_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont)))
Bacteria_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)
#Treatment  1   0.2938 0.09193 1.0124  0.465
#Residual  10   2.9022 0.90807              
#Total     11   3.1960 1.00000

##########
# 6_2-Butanone  vs control_phyllosphere
##########
Bacteria_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont= subset_samples(Bacteria_microcosm_one_CSS_Phyllosphere, Treatment %in% c("6_2-Butanone","1_Control"));Bacteria_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont)))
Bacteria_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.64831 0.20618 2.5973  0.003 **
#Residual  10  2.49604 0.79382                 
#Total     11  3.14435 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

##########
#7_2-Nonanone
##########
Bacteria_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont= subset_samples(Bacteria_microcosm_one_CSS_Phyllosphere, Treatment %in% c("7_2-Nonanone","1_Control"));Bacteria_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont)))
Bacteria_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)
#Treatment  1  0.28745 0.09146 1.0066  0.456
#Residual  10  2.85562 0.90854              
#Total     11  3.14306 1.00000 

##########
#8_Volatile_Mixture
##########
Bacteria_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont= subset_samples(Bacteria_microcosm_one_CSS_Phyllosphere, Treatment %in% c("8_Volatile_Mixture","1_Control"));Bacteria_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont)))
Bacteria_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)
#Treatment  1  0.29483 0.09483 1.0476  0.345
#Residual  10  2.81429 0.90517              
#Total     11  3.10912 1.00000

#####################################################################################################
#Phyllosphere deseq for the bacterial
#if (!require("BiocManager", quietly = TRUE))
#  install.packages("BiocManager")
#BiocManager::install("ALDEx2")
library(ALDEx2)

Bacteria_microcosm_one<- subset_samples(Bacteria_mereged, Experiment %in% c("One"));Bacteria_microcosm_one#[ 15421 taxa and 1102 samples ]
Bacteria_microcosm_one<- subset_samples(Bacteria_microcosm_one, Sample_type %in% c("4_Phyllosphere","2_Soil","3_Rhizosphere"));Bacteria_microcosm_one#[ 15421 taxa and 1102 samples ]
Bacteria_microcosm_one_Phyllosphere= subset_samples(Bacteria_microcosm_one, Sample_type %in% c("4_Phyllosphere"));Bacteria_microcosm_one_Phyllosphere#[ 15421 taxa and 1102 samples ]

#serratia vs control
Bacteria_microcosm_one_Phyllosphere_ser_cont= subset_samples(Bacteria_microcosm_one_Phyllosphere, Treatment %in% c("2_Serratia_plymuthica_HRO_C48","1_Control"));Bacteria_microcosm_one_Phyllosphere_ser_cont#[ 15421 taxa and 1102 samples ]
diagdds_ser = phyloseq_to_deseq2(Bacteria_microcosm_one_Phyllosphere_ser_cont, ~ Treatment);diagdds_ser
diagdds_ser = DESeq(diagdds_ser, test="Wald", fitType="parametric");diagdds_ser
res = results(diagdds_ser, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_ser = res[which(res$padj < alpha), ];sigtab_ser
sigtab_ser = cbind(as(sigtab_ser, "data.frame"), as(tax_table(Bacteria_microcosm_one_Phyllosphere_ser_cont)[rownames(sigtab_ser), ], "matrix"));sigtab_ser
head(sigtab_ser)
write.csv(sigtab_ser, "sigtab_ser.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}

sigtab_ser = sigtab_ser[order(sigtab_ser$"log2FoldChange", decreasing = TRUE),];sigtab_ser
sigtab_ser = sigtab_ser[1:25,];sigtab_ser
sigtab_ser$Genus = factor(sigtab_ser$Genus, levels = rev(sigtab_ser$Genus))
a = ggplot(sigtab_ser, aes(log2FoldChange, Genus, color = Genus ))+
  geom_point() +theme_bw() +
  geom_segment( aes(x=0, xend=log2FoldChange, y=Genus, yend=Genus, color = Genus)) +
  geom_vline(xintercept = 0, size=0.3) +xlab("log2FoldChange") +ylab(NULL);a# +


# Genus order
x = tapply(sigtab_ser$log2FoldChange, sigtab_ser$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser$Genus = factor(as.character(sigtab_ser$Genus), levels=names(x))
ggplot(sigtab_ser, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Gammaproteobacteria
#####################################################################################################
#steno vs control
Bacteria_microcosm_one_Phyllosphere_steno_cont= subset_samples(Bacteria_microcosm_one_Phyllosphere, Treatment %in% c("3_Stenotrophomonas_rhizophilla_SPA_P69","1_Control"));Bacteria_microcosm_one_Phyllosphere_steno_cont#[ 15421 taxa and 1102 samples ]
diagdds_Steno = phyloseq_to_deseq2(Bacteria_microcosm_one_Phyllosphere_steno_cont, ~ Treatment);diagdds_Steno
diagdds_Steno = DESeq(diagdds_Steno, test="Wald", fitType="parametric");diagdds_Steno
res = results(diagdds_Steno, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_Steno = res[which(res$padj < alpha), ];sigtab_Steno
sigtab_Steno = cbind(as(sigtab_Steno, "data.frame"), as(tax_table(Bacteria_microcosm_one_Phyllosphere_steno_cont)[rownames(sigtab_Steno), ], "matrix"));sigtab_Steno
head(sigtab_Steno)
write.csv(sigtab_Steno, "sigtab_Steno.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_Steno$log2FoldChange, sigtab_Steno$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_Steno$Phylum = factor(as.character(sigtab_Steno$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_Steno$log2FoldChange, sigtab_Steno$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_Steno$Genus = factor(as.character(sigtab_Steno$Genus), levels=names(x))
ggplot(sigtab_Steno, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Gammaproteobacteria and Actinobacteria
#####################################################################################################
#Consort vs control
Bacteria_microcosm_one_Phyllosphere_consortium_cont= subset_samples(Bacteria_microcosm_one_Phyllosphere, Treatment %in% c("4_Bacteria_Mixture","1_Control"));Bacteria_microcosm_one_Phyllosphere_consortium_cont#[ 15421 taxa and 1102 samples ]
diagdds_consortium = phyloseq_to_deseq2(Bacteria_microcosm_one_Phyllosphere_consortium_cont, ~ Treatment);diagdds_consortium
diagdds_consortium = DESeq(diagdds_consortium, test="Wald", fitType="parametric");diagdds_consortium
res = results(diagdds_consortium, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_consortium = res[which(res$padj < alpha), ];sigtab_consortium
sigtab_consortium = cbind(as(sigtab_consortium, "data.frame"), as(tax_table(Bacteria_microcosm_one_Phyllosphere_consortium_cont)[rownames(sigtab_consortium), ], "matrix"));sigtab_consortium
head(sigtab_consortium)
write.csv(sigtab_consortium, "sigtab_consortium.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_consortium$log2FoldChange, sigtab_consortium$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_consortium$Phylum = factor(as.character(sigtab_consortium$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_consortium$log2FoldChange, sigtab_consortium$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_consortium$Genus = factor(as.character(sigtab_consortium$Genus), levels=names(x))
ggplot(sigtab_consortium, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))

#Gammabnacteroidia
#####################################################################################################
#Consort vs control
Bacteria_microcosm_one_Phyllosphere_Methyl_cont= subset_samples(Bacteria_microcosm_one_Phyllosphere, Treatment %in% c("5_3-Methyl-1-Butanol","1_Control"));Bacteria_microcosm_one_Phyllosphere_Methyl_cont#[ 15421 taxa and 1102 samples ]
diagdds_Methyl = phyloseq_to_deseq2(Bacteria_microcosm_one_Phyllosphere_Methyl_cont, ~ Treatment);diagdds_Methyl
diagdds_Methyl = DESeq(diagdds_Methyl, test="Wald", fitType="parametric");diagdds_Methyl
res = results(diagdds_Methyl, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_Methyl = res[which(res$padj < alpha), ];sigtab_Methyl
sigtab_Methyl = cbind(as(sigtab_Methyl, "data.frame"), as(tax_table(Bacteria_microcosm_one_Phyllosphere_Methyl_cont)[rownames(sigtab_Methyl), ], "matrix"));sigtab_Methyl
head(sigtab_Methyl)
write.csv(sigtab_Methyl, "sigtab_Methyl.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_Methyl$log2FoldChange, sigtab_Methyl$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_Methyl$Phylum = factor(as.character(sigtab_Methyl$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_Methyl$log2FoldChange, sigtab_Methyl$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_Methyl$Genus = factor(as.character(sigtab_Methyl$Genus), levels=names(x))
ggplot(sigtab_Methyl, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Bacilli and Gammaproteobacteria
#####################################################################################################
#Butanone
#Consort vs control
Bacteria_microcosm_one_Phyllosphere_Butanone_cont= subset_samples(Bacteria_microcosm_one_Phyllosphere, Treatment %in% c("6_2-Butanone","1_Control"));Bacteria_microcosm_one_Phyllosphere_Butanone_cont#[ 15421 taxa and 1102 samples ]
diagdds_Butanone = phyloseq_to_deseq2(Bacteria_microcosm_one_Phyllosphere_Butanone_cont, ~ Treatment);diagdds_Butanone
diagdds_Butanone = DESeq(diagdds_Butanone, test="Wald", fitType="parametric");diagdds_Butanone
res = results(diagdds_Butanone, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_Butanone = res[which(res$padj < alpha), ];sigtab_Butanone
sigtab_Butanone = cbind(as(sigtab_Butanone, "data.frame"), as(tax_table(Bacteria_microcosm_one_Phyllosphere_Butanone_cont)[rownames(sigtab_Butanone), ], "matrix"));sigtab_Butanone
head(sigtab_Butanone)
write.csv(sigtab_Butanone, "sigtab_Butanone.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_Butanone$log2FoldChange, sigtab_Butanone$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_Butanone$Phylum = factor(as.character(sigtab_Butanone$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_Butanone$log2FoldChange, sigtab_Butanone$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_Butanone$Genus = factor(as.character(sigtab_Butanone$Genus), levels=names(x))
ggplot(sigtab_Butanone, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#10 ASVs assigned to Alphaproteobacteria, Gammaproteobacteria and Actinobacteria
#####################################################################################################
#7_2-Nonanone vs control
Bacteria_microcosm_one_Phyllosphere_7_2_Nonanone_cont= subset_samples(Bacteria_microcosm_one_Phyllosphere, Treatment %in% c("7_2-Nonanone","1_Control"));Bacteria_microcosm_one_Phyllosphere_7_2_Nonanone_cont#[ 15421 taxa and 1102 samples ]
diagdds_Nonanone = phyloseq_to_deseq2(Bacteria_microcosm_one_Phyllosphere_7_2_Nonanone_cont, ~ Treatment);diagdds_Nonanone
diagdds_Nonanone = DESeq(diagdds_Nonanone, test="Wald", fitType="parametric");diagdds_Nonanone
res = results(diagdds_Nonanone, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_Nonanone = res[which(res$padj < alpha), ];sigtab_Nonanone
sigtab_Nonanone = cbind(as(sigtab_Nonanone, "data.frame"), as(tax_table(Bacteria_microcosm_one_Phyllosphere_7_2_Nonanone_cont)[rownames(sigtab_Nonanone), ], "matrix"));sigtab_Nonanone
head(sigtab_Nonanone)
write.csv(sigtab_Nonanone, "sigtab_Nonanone.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_Nonanone$log2FoldChange, sigtab_Nonanone$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_Nonanone$Phylum = factor(as.character(sigtab_Nonanone$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_Nonanone$log2FoldChange, sigtab_Nonanone$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_Nonanone$Genus = factor(as.character(sigtab_Nonanone$Genus), levels=names(x))
ggplot(sigtab_Nonanone, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Actinobacteria
#####################################################################################################
#8_Volatile_Mixture
Bacteria_microcosm_one_Phyllosphere8_Volatile_Mixture_cont= subset_samples(Bacteria_microcosm_one_Phyllosphere, Treatment %in% c("8_Volatile_Mixture","1_Control"));Bacteria_microcosm_one_Phyllosphere8_Volatile_Mixture_cont#[ 15421 taxa and 1102 samples ]
diagdds_Volatile_mixture = phyloseq_to_deseq2(Bacteria_microcosm_one_Phyllosphere8_Volatile_Mixture_cont, ~ Treatment);diagdds_Volatile_mixture
diagdds_Volatile_mixture = DESeq(diagdds_Volatile_mixture, test="Wald", fitType="parametric");diagdds_Volatile_mixture
res = results(diagdds_Volatile_mixture, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_Volatile_mixture = res[which(res$padj < alpha), ];sigtab_Volatile_mixture
sigtab_Volatile_mixture = cbind(as(sigtab_Volatile_mixture, "data.frame"), as(tax_table(Bacteria_microcosm_one_Phyllosphere8_Volatile_Mixture_cont)[rownames(sigtab_Volatile_mixture), ], "matrix"));sigtab_Volatile_mixture
head(sigtab_Volatile_mixture)
write.csv(sigtab_Volatile_mixture, "sigtab_Volatile_mixture.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_Volatile_mixture$log2FoldChange, sigtab_Volatile_mixture$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_Volatile_mixture$Phylum = factor(as.character(sigtab_Volatile_mixture$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_Volatile_mixture$log2FoldChange, sigtab_Volatile_mixture$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_Volatile_mixture$Genus = factor(as.character(sigtab_Volatile_mixture$Genus), levels=names(x))
ggplot(sigtab_Volatile_mixture, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))

#Gammaproteobactria
#####################################################################################################
#####################################################################################################
#Subset the different compartments
#3_Rhizosphere
#Subset the different compartments
#3_Rhizosphere

Bacteria_microcosm_one_CSS_Rhizosphere= subset_samples(Bacteria_microcosm_one_CSS, Sample_type %in% c("3_Rhizosphere"));Bacteria_microcosm_one_CSS_Rhizosphere#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Rhizosphere <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Rhizosphere)))
sampledf_Bacteria_microcosm_one_CSS_Rhizosphere <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Rhizosphere))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Rhizosphere, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Rhizosphere), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)    
#Treatment  7   1.5490 0.23288 1.7347  0.001 ***
#Residual  40   5.1026 0.76712                  
#Total     47   6.6516 1.00000                  
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

############################################################
#Bacteria vs control in the 3_Rhizosphere
Bacteria_microcosm_one_CSS_Rhizosphere_Bact_control= subset_samples(Bacteria_microcosm_one_CSS_Rhizosphere, Pool %in% c("Bacteria","Control"));Bacteria_microcosm_one_CSS_Rhizosphere_Bact_control#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Rhizosphere_Bact_control <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Rhizosphere_Bact_control)))
sampledf_Bacteria_microcosm_one_CSS_Rhizosphere_Bact_control <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Rhizosphere_Bact_control))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_Bact_control <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Rhizosphere_Bact_control, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_Bact_control
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Rhizosphere_Bact_control), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_Bact_control)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_Bact_control ~ Pool, data=sampleda, permutation=999);adores#change from adonis to adonis2
#Df SumOfSqs      R2      F Pr(>F)   
#         Pool      1  0.22813 0.07963 1.9034  0.006 **
#Residual 22  2.63679 0.92037                 
#Total    23  2.86493 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

#Volatile vs control in the 3_Rhizosphere
Bacteria_microcosm_one_CSS_Rhizosphere_vola_control= subset_samples(Bacteria_microcosm_one_CSS_Rhizosphere, Pool %in% c("Volatiles","Control"));Bacteria_microcosm_one_CSS_Rhizosphere_vola_control#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Rhizosphere_vola_control <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Rhizosphere_vola_control)))
sampledf_Bacteria_microcosm_one_CSS_Rhizosphere_vola_control <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Rhizosphere_vola_control))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_vola_control <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Rhizosphere_vola_control, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_vola_control
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Rhizosphere_vola_control), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_vola_control)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_vola_control ~ Pool, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)
#Pool      1   0.1597 0.03836 1.1169  0.211
#Residual 28   4.0027 0.96164              
#Total    29   4.1623 1.00000               

############################################################
# S. plymuthica HRO C48  vs control_Rhizosphere
############################################################
Bacteria_microcosm_one_CSS_Rhizosphere_ser_cont= subset_samples(Bacteria_microcosm_one_CSS_Rhizosphere, Treatment %in% c("2_Serratia_plymuthica_HRO_C48","1_Control"));Bacteria_microcosm_one_CSS_Rhizosphere_ser_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Rhizosphere_ser_cont <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Rhizosphere_ser_cont)))
Bacteria_microcosm_one_CSS_Rhizosphere_ser_cont <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Rhizosphere_ser_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_ser_cont <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Rhizosphere_ser_cont, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_ser_cont
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Rhizosphere_ser_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_ser_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_ser_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2

#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.21439 0.15784 1.8742  0.006 **
#Residual  10  1.14391 0.84216                 
#Total     11  1.35830 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

####################################################################################################
#Differential analyses
Bacteria_microcosm_one<- subset_samples(Bacteria_mereged, Experiment %in% c("One"));Bacteria_microcosm_one#[ 15421 taxa and 1102 samples ]
Bacteria_microcosm_one<- subset_samples(Bacteria_microcosm_one, Sample_type %in% c("4_Phyllosphere","2_Soil","3_Rhizosphere"));Bacteria_microcosm_one#[ 15421 taxa and 1102 samples ]
Bacteria_microcosm_one_Rhizosphere= subset_samples(Bacteria_microcosm_one, Sample_type %in% c("3_Rhizosphere"));Bacteria_microcosm_one_Rhizosphere#[ 15421 taxa and 1102 samples ]
#serratia vs control
Bacteria_microcosm_one_Rhizosphere_ser_cont= subset_samples(Bacteria_microcosm_one_Rhizosphere, Treatment %in% c("2_Serratia_plymuthica_HRO_C48","1_Control"));Bacteria_microcosm_one_Rhizosphere_ser_cont#[ 15421 taxa and 1102 samples ]
diagdds_ser = phyloseq_to_deseq2(Bacteria_microcosm_one_Rhizosphere_ser_cont, ~ Treatment);diagdds_ser
diagdds_ser = DESeq(diagdds_ser, test="Wald", fitType="parametric");diagdds_ser
res = results(diagdds_ser, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_ser = res[which(res$padj < alpha), ];sigtab_ser
sigtab_ser = cbind(as(sigtab_ser, "data.frame"), as(tax_table(Bacteria_microcosm_one_Rhizosphere_ser_cont)[rownames(sigtab_ser), ], "matrix"));sigtab_ser
head(sigtab_ser)
write.csv(sigtab_ser, "sigtab_ser.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_ser$log2FoldChange, sigtab_ser$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser$Phylum = factor(as.character(sigtab_ser$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_ser$log2FoldChange, sigtab_ser$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser$Genus = factor(as.character(sigtab_ser$Genus), levels=names(x))
ggplot(sigtab_ser, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Gammaproteobacteria (increase),Deltaproteobacteria (decrease),Bacteroidia (decrease),Alphaproteobacteria (incre)
############################################################################################
############################################################################################
############################################################################################
############################################################################################

# 3_Stenotrophomonas_rhizophilla_SPA_P69  vs control_Rhizosphere
##########
Bacteria_microcosm_one_CSS_Rhizosphere_Steno_cont= subset_samples(Bacteria_microcosm_one_CSS_Rhizosphere, Treatment %in% c("3_Stenotrophomonas_rhizophilla_SPA_P69","1_Control"));Bacteria_microcosm_one_CSS_Rhizosphere_Steno_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Rhizosphere_Steno_cont <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Rhizosphere_Steno_cont)))
Bacteria_microcosm_one_CSS_Rhizosphere_Steno_cont <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Rhizosphere_Steno_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_Steno_cont <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Rhizosphere_Steno_cont, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_Steno_cont
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Rhizosphere_Steno_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_Steno_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_Steno_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)  
#Treatment  1  0.20628 0.13765 1.5962  0.024 *
#Residual  10  1.29230 0.86235                
#Total     11  1.49858 1.00000                
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
############################################################################################
############################################################################################
Bacteria_microcosm_one_Rhizosphere_Steno_cont= subset_samples(Bacteria_microcosm_one_Rhizosphere, Treatment %in% c("3_Stenotrophomonas_rhizophilla_SPA_P69","1_Control"));Bacteria_microcosm_one_Rhizosphere_Steno_cont#[ 15421 taxa and 1102 samples ]
diagdds_Steno = phyloseq_to_deseq2(Bacteria_microcosm_one_Rhizosphere_Steno_cont, ~ Treatment);diagdds_Steno
diagdds_Steno = DESeq(diagdds_Steno, test="Wald", fitType="parametric");diagdds_Steno
res = results(diagdds_Steno, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_Steno = res[which(res$padj < alpha), ];sigtab_Steno
sigtab_Steno = cbind(as(sigtab_Steno, "data.frame"), as(tax_table(Bacteria_microcosm_one_Rhizosphere_Steno_cont)[rownames(sigtab_Steno), ], "matrix"));sigtab_Steno
head(sigtab_Steno)
write.csv(sigtab_Steno, "sigtab_Steno.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_Steno$log2FoldChange, sigtab_Steno$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_Steno$Phylum = factor(as.character(sigtab_Steno$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_Steno$log2FoldChange, sigtab_Steno$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_Steno$Genus = factor(as.character(sigtab_Steno$Genus), levels=names(x))
ggplot(sigtab_Steno, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Gammaproteobacteria (increase),Deltaproteobacteria (increase),Bacteroidia (increase),Alphaproteobacteria (decrease), and Verrucomicrobia
############################################################################################
############################################################################################

##########
# 4_Bacteria_Mixture  vs control_Rhizosphere
##########
Bacteria_microcosm_one_CSS_Rhizosphere_consortium_cont= subset_samples(Bacteria_microcosm_one_CSS_Rhizosphere, Treatment %in% c("4_Bacteria_Mixture","1_Control"));Bacteria_microcosm_one_CSS_Rhizosphere_consortium_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Rhizosphere_consortium_cont <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Rhizosphere_consortium_cont)))
Bacteria_microcosm_one_CSS_Rhizosphere_consortium_cont <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Rhizosphere_consortium_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_consortium_cont <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Rhizosphere_consortium_cont, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_consortium_cont
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Rhizosphere_consortium_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_consortium_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_consortium_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)  
#Treatment  1  0.18329 0.14509 1.6971  0.014 *
#Residual  10  1.08006 0.85491                
#Total     11  1.26336 1.00000                
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
############################################################################################
############################################################################################
Bacteria_microcosm_one_Rhizosphere_consortium_cont= subset_samples(Bacteria_microcosm_one_Rhizosphere, Treatment %in% c("4_Bacteria_Mixture","1_Control"));Bacteria_microcosm_one_Rhizosphere_consortium_cont#[ 15421 taxa and 1102 samples ]
diagdds_consortium = phyloseq_to_deseq2(Bacteria_microcosm_one_Rhizosphere_consortium_cont, ~ Treatment);diagdds_consortium
diagdds_consortium = DESeq(diagdds_consortium, test="Wald", fitType="parametric");diagdds_consortium
res = results(diagdds_consortium, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_consortium = res[which(res$padj < alpha), ];sigtab_consortium
sigtab_consortium = cbind(as(sigtab_consortium, "data.frame"), as(tax_table(Bacteria_microcosm_one_Rhizosphere_consortium_cont)[rownames(sigtab_consortium), ], "matrix"));sigtab_consortium
head(sigtab_consortium)
write.csv(sigtab_consortium, "sigtab_consortium.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_consortium$log2FoldChange, sigtab_consortium$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_consortium$Phylum = factor(as.character(sigtab_consortium$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_consortium$log2FoldChange, sigtab_consortium$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_consortium$Genus = factor(as.character(sigtab_consortium$Genus), levels=names(x))
ggplot(sigtab_consortium, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Gammaproteobacteria (increase),Deltaproteobacteria (increase),Bacteroidia (decrease),Alphaproteobacteria (decrease)
############################################################################################
###
##########
# 5_3-Methyl-1-Butanol  vs control_Rhizosphere
##########
Bacteria_microcosm_one_CSS_Rhizosphere_5_3_Methyl_1_Butanol_cont= subset_samples(Bacteria_microcosm_one_CSS_Rhizosphere, Treatment %in% c("5_3-Methyl-1-Butanol","1_Control"));Bacteria_microcosm_one_CSS_Rhizosphere_5_3_Methyl_1_Butanol_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Rhizosphere_5_3_Methyl_1_Butanol_cont <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Rhizosphere_5_3_Methyl_1_Butanol_cont)))
Bacteria_microcosm_one_CSS_Rhizosphere_5_3_Methyl_1_Butanol_cont <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Rhizosphere_5_3_Methyl_1_Butanol_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_5_3_Methyl_1_Butanol_cont <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Rhizosphere_5_3_Methyl_1_Butanol_cont, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_5_3_Methyl_1_Butanol_cont
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Rhizosphere_5_3_Methyl_1_Butanol_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_5_3_Methyl_1_Butanol_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_5_3_Methyl_1_Butanol_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)  
#Treatment  1  0.15796 0.11682 1.3227  0.074 .
#Residual  10  1.19420 0.88318                
#Total     11  1.35215 1.00000                
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

##########
# 6_2-Butanone  vs control_Rhizosphere
##########
Bacteria_microcosm_one_CSS_Rhizosphere_6_2_Butanone_cont= subset_samples(Bacteria_microcosm_one_CSS_Rhizosphere, Treatment %in% c("6_2-Butanone","1_Control"));Bacteria_microcosm_one_CSS_Rhizosphere_6_2_Butanone_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Rhizosphere_6_2_Butanone_cont <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Rhizosphere_6_2_Butanone_cont)))
Bacteria_microcosm_one_CSS_Rhizosphere_6_2_Butanone_cont <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Rhizosphere_6_2_Butanone_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_6_2_Butanone_cont <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Rhizosphere_6_2_Butanone_cont, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_6_2_Butanone_cont
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Rhizosphere_6_2_Butanone_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_6_2_Butanone_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_6_2_Butanone_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2     F Pr(>F)
#Treatment  1  0.11518 0.09083 0.999  0.411
#Residual  10  1.15291 0.90917             
#Total     11  1.26809 1.00000             
##########
#7_2-Nonanone
##########
Bacteria_microcosm_one_CSS_Rhizosphere_7_2_Nonanone_cont= subset_samples(Bacteria_microcosm_one_CSS_Rhizosphere, Treatment %in% c("7_2-Nonanone","1_Control"));Bacteria_microcosm_one_CSS_Rhizosphere_7_2_Nonanone_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Rhizosphere_7_2_Nonanone_cont <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Rhizosphere_7_2_Nonanone_cont)))
Bacteria_microcosm_one_CSS_Rhizosphere_7_2_Nonanone_cont <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Rhizosphere_7_2_Nonanone_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_7_2_Nonanone_cont <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Rhizosphere_7_2_Nonanone_cont, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_7_2_Nonanone_cont
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Rhizosphere_7_2_Nonanone_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_7_2_Nonanone_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere_7_2_Nonanone_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.30013 0.17418 2.1091  0.007 **
#Residual  10  1.42302 0.82582                 
#Total     11  1.72315 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
############################################################################################
############################################################################################
Bacteria_microcosm_one_Rhizosphere_Nonanone_cont= subset_samples(Bacteria_microcosm_one_Rhizosphere, Treatment %in% c("7_2-Nonanone","1_Control"));Bacteria_microcosm_one_Rhizosphere_Nonanone_cont#[ 15421 taxa and 1102 samples ]
diagdds_Nonanone = phyloseq_to_deseq2(Bacteria_microcosm_one_Rhizosphere_Nonanone_cont, ~ Treatment);diagdds_Nonanone
diagdds_Nonanone = DESeq(diagdds_Nonanone, test="Wald", fitType="parametric");diagdds_Nonanone
res = results(diagdds_Nonanone, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_Nonanone = res[which(res$padj < alpha), ];sigtab_Nonanone
sigtab_Nonanone = cbind(as(sigtab_Nonanone, "data.frame"), as(tax_table(Bacteria_microcosm_one_Rhizosphere_Nonanone_cont)[rownames(sigtab_Nonanone), ], "matrix"));sigtab_Nonanone
head(sigtab_Nonanone)
write.csv(sigtab_Nonanone, "sigtab_Nonanone.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_Nonanone$log2FoldChange, sigtab_Nonanone$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_Nonanone$Phylum = factor(as.character(sigtab_Nonanone$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_Nonanone$log2FoldChange, sigtab_Nonanone$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_Nonanone$Genus = factor(as.character(sigtab_Nonanone$Genus), levels=names(x))
ggplot(sigtab_Nonanone, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Gammaproteobacteria (decrease),Deltaproteobacteria (increase),Bacteroidia (decrease)
############################################################################################
##########
#8_Volatile_Mixture
##########
Bacteria_microcosm_one_CSS_Rhizosphere8_Volatile_Mixture_cont= subset_samples(Bacteria_microcosm_one_CSS_Rhizosphere, Treatment %in% c("8_Volatile_Mixture","1_Control"));Bacteria_microcosm_one_CSS_Rhizosphere8_Volatile_Mixture_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Rhizosphere8_Volatile_Mixture_cont <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Rhizosphere8_Volatile_Mixture_cont)))
Bacteria_microcosm_one_CSS_Rhizosphere8_Volatile_Mixture_cont <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Rhizosphere8_Volatile_Mixture_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere8_Volatile_Mixture_cont <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Rhizosphere8_Volatile_Mixture_cont, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere8_Volatile_Mixture_cont
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Rhizosphere8_Volatile_Mixture_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere8_Volatile_Mixture_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Rhizosphere8_Volatile_Mixture_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)
#Treatment  1  0.15292 0.10237 1.1405  0.243
#Residual  10  1.34084 0.89763              
#Total     11  1.49375 1.00000  


#####################################################################################################
#Subset the different compartments
#Soil
#Subset the different compartments
#Soil
Bacteria_microcosm_one_CSS_Soil= subset_samples(Bacteria_microcosm_one_CSS, Sample_type %in% c("2_Soil"));Bacteria_microcosm_one_CSS_Soil#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Soil <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Soil)))
sampledf_Bacteria_microcosm_one_CSS_Soil <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Soil))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Soil <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Soil, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Soil
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Soil), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Soil)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Soil ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  7   1.0618 0.19458 1.3805  0.009 **
#Residual  40   4.3952 0.80542                 
#Total     47   5.4570 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
############################################################
############################################################
#Bacteria vs control in the Soil
Bacteria_microcosm_one_CSS_Soil_Bact_control= subset_samples(Bacteria_microcosm_one_CSS_Soil, Pool %in% c("Bacteria","Control"));Bacteria_microcosm_one_CSS_Soil_Bact_control#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Soil_Bact_control <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Soil_Bact_control)))
sampledf_Bacteria_microcosm_one_CSS_Soil_Bact_control <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Soil_Bact_control))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_Bact_control <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Soil_Bact_control, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_Bact_control
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Soil_Bact_control), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_Bact_control)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_Bact_control ~ Pool, data=sampleda, permutation=999);adores#change from adonis to adonis2
#          Df SumOfSqs     R2      F Pr(>F)  
#Pool      1  0.16395 0.0648 1.5244  0.072 .
#Residual 22  2.36606 0.9352                
#Total    23  2.53000 1.0000                
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

#Volatile vs control in the Soil
Bacteria_microcosm_one_CSS_Soil_vola_control= subset_samples(Bacteria_microcosm_one_CSS_Soil, Pool %in% c("Volatiles","Control"));Bacteria_microcosm_one_CSS_Soil_vola_control#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Soil_vola_control <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Soil_vola_control)))
sampledf_Bacteria_microcosm_one_CSS_Soil_vola_control <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Soil_vola_control))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_vola_control <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Soil_vola_control, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_vola_control
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Soil_vola_control), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_vola_control)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_vola_control ~ Pool, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)  
#Pool      1   0.1906 0.05443 1.6118  0.051 .
#Residual 28   3.3114 0.94557                
#Total    29   3.5020 1.00000                
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1              
############################################################
############################################################
# S. plymuthica HRO C48  vs control_Soil
##########
Bacteria_microcosm_one_CSS_Soil_ser_cont= subset_samples(Bacteria_microcosm_one_CSS_Soil, Treatment %in% c("2_Serratia_plymuthica_HRO_C48","1_Control"));Bacteria_microcosm_one_CSS_Soil_ser_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Soil_ser_cont <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Soil_ser_cont)))
Bacteria_microcosm_one_CSS_Soil_ser_cont <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Soil_ser_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_ser_cont <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Soil_ser_cont, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_ser_cont
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Soil_ser_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_ser_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_ser_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2

#           Df SumOfSqs      R2      F Pr(>F)  
#Treatment  1  0.17404 0.13987 1.6261  0.077 .
#Residual  10  1.07026 0.86013                
#Total     11  1.24430 1.00000                
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

##########
# 3_Stenotrophomonas_rhizophilla_SPA_P69  vs control_Soil
##########
Bacteria_microcosm_one_CSS_Soil_Steno_cont= subset_samples(Bacteria_microcosm_one_CSS_Soil, Treatment %in% c("3_Stenotrophomonas_rhizophilla_SPA_P69","1_Control"));Bacteria_microcosm_one_CSS_Soil_Steno_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Soil_Steno_cont <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Soil_Steno_cont)))
Bacteria_microcosm_one_CSS_Soil_Steno_cont <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Soil_Steno_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_Steno_cont <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Soil_Steno_cont, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_Steno_cont
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Soil_Steno_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_Steno_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_Steno_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)
#Treatment  1  0.14782 0.11912 1.3523  0.144
#Residual  10  1.09308 0.88088              
#Total     11  1.24090 1.00000  


##########
# 4_Bacteria_Mixture  vs control_Soil
##########
Bacteria_microcosm_one_CSS_Soil_consortium_cont= subset_samples(Bacteria_microcosm_one_CSS_Soil, Treatment %in% c("4_Bacteria_Mixture","1_Control"));Bacteria_microcosm_one_CSS_Soil_consortium_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Soil_consortium_cont <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Soil_consortium_cont)))
Bacteria_microcosm_one_CSS_Soil_consortium_cont <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Soil_consortium_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_consortium_cont <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Soil_consortium_cont, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_consortium_cont
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Soil_consortium_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_consortium_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_consortium_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)
#Treatment  1  0.13778 0.11408 1.2877  0.136
#Residual  10  1.06991 0.88592              
#Total     11  1.20769 1.00000 

##########
# 5_3-Methyl-1-Butanol  vs control_Soil
##########
Bacteria_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont= subset_samples(Bacteria_microcosm_one_CSS_Soil, Treatment %in% c("5_3-Methyl-1-Butanol","1_Control"));Bacteria_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont)))
Bacteria_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)  
#Treatment  1  0.16688 0.13663 1.5825  0.085 .
#Residual  10  1.05455 0.86337                
#Total     11  1.22143 1.00000                
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

##########
# 6_2-Butanone  vs control_Soil
##########
Bacteria_microcosm_one_CSS_Soil_6_2_Butanone_cont= subset_samples(Bacteria_microcosm_one_CSS_Soil, Treatment %in% c("6_2-Butanone","1_Control"));Bacteria_microcosm_one_CSS_Soil_6_2_Butanone_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Soil_6_2_Butanone_cont <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Soil_6_2_Butanone_cont)))
Bacteria_microcosm_one_CSS_Soil_6_2_Butanone_cont <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Soil_6_2_Butanone_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_6_2_Butanone_cont <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Soil_6_2_Butanone_cont, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_6_2_Butanone_cont
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Soil_6_2_Butanone_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_6_2_Butanone_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_6_2_Butanone_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)
#Treatment  1  0.16077 0.12643 1.4473  0.102
#Residual  10  1.11084 0.87357              
#Total     11  1.27160 1.00000            
##########
#7_2-Nonanone
##########
Bacteria_microcosm_one_CSS_Soil_7_2_Nonanone_cont= subset_samples(Bacteria_microcosm_one_CSS_Soil, Treatment %in% c("7_2-Nonanone","1_Control"));Bacteria_microcosm_one_CSS_Soil_7_2_Nonanone_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Soil_7_2_Nonanone_cont <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Soil_7_2_Nonanone_cont)))
Bacteria_microcosm_one_CSS_Soil_7_2_Nonanone_cont <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Soil_7_2_Nonanone_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_7_2_Nonanone_cont <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Soil_7_2_Nonanone_cont, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_7_2_Nonanone_cont
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Soil_7_2_Nonanone_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_7_2_Nonanone_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Soil_7_2_Nonanone_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs     R2      F Pr(>F)  
#Treatment  1  0.21456 0.1393 1.6184  0.072 .
#Residual  10  1.32573 0.8607                
#Total     11  1.54029 1.0000                
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

##########
#8_Volatile_Mixture
##########
Bacteria_microcosm_one_CSS_Soil8_Volatile_Mixture_cont= subset_samples(Bacteria_microcosm_one_CSS_Soil, Treatment %in% c("8_Volatile_Mixture","1_Control"));Bacteria_microcosm_one_CSS_Soil8_Volatile_Mixture_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_microcosm_one_CSS_Soil8_Volatile_Mixture_cont <- t(data.frame(otu_table(Bacteria_microcosm_one_CSS_Soil8_Volatile_Mixture_cont)))
Bacteria_microcosm_one_CSS_Soil8_Volatile_Mixture_cont <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Soil8_Volatile_Mixture_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_microcosm_one_CSS_Soil8_Volatile_Mixture_cont <- vegdist(dist.matrix_Bacteria_microcosm_one_CSS_Soil8_Volatile_Mixture_cont, method = "bray"); phyloseq_bray_Bacteria_microcosm_one_CSS_Soil8_Volatile_Mixture_cont
sampleda <- data.frame(sample_data(Bacteria_microcosm_one_CSS_Soil8_Volatile_Mixture_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_microcosm_one_CSS_Soil8_Volatile_Mixture_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_microcosm_one_CSS_Soil8_Volatile_Mixture_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)  
#Treatment  1  0.16105 0.13159 1.5153  0.097 .
#Residual  10  1.06283 0.86841                
#Total     11  1.22388 1.00000                
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1             
```



#FUNGI COMMUNITY #FUNGI COMMUNITY#FUNGI COMMUNITY#FUNGI COMMUNITY#FUNGI COMMUNITY
```{r,echo=FALSE,warning=FALSE}
setwd("/Users/expeditoolimi/Documents/Manuscript-AA-OE_SB/Microbiome_analysis")

## Imported otu table
Fungi_otu_microcosm <- read.table("Fungi-table.txt", header=TRUE,row.names="OTUID");Fungi_otu_microcosm
Fungi_otu_microcosm <- otu_table(Fungi_otu_microcosm, taxa_are_rows = TRUE);Fungi_otu_microcosm 

## Imported taxonomy
#Fungi_tax_microcosm <- read.csv("Taxonomy.csv",header=TRUE,row.names="OTUID");Fungi_tax_microcosm
Fungi_tax_microcosm <- read.table("Taxon_Fungi.txt", header=TRUE, sep="\t",row.names="OTUID");Fungi_tax_microcosm
Fungi_tax_microcosm <- as.matrix(Fungi_tax_microcosm);Fungi_tax_microcosm
Fungi_tax_microcosm <- tax_table(Fungi_tax_microcosm);Fungi_tax_microcosm

## Imported metadata file
#metadata_Fungi <- read.table("Mapping_File_16S_Combined.txt", header=TRUE, sep="\t",row.names="SampleID");metadata_Fungi
metadata_Fungi <- read.csv("Mapping_File_ITS_Combined.csv",header=TRUE,row.names="SampleID");metadata_Fungi

#rownames(metadata) <- metadata[,1]
metadata_Fungi <- sample_data(metadata_Fungi);metadata_Fungi
#tree
all(rownames(metadata_Fungi) %in% colnames(Fungi_otu_microcosm))
#Convert to phyloseq object
Fungi_mereged <- merge_phyloseq(Fungi_otu_microcosm,Fungi_tax_microcosm, metadata_Fungi);Fungi_mereged#  [ 9079 taxa and 281 samples ]
microbiome::summarize_phyloseq(Fungi_mereged)
#[1] "1] Min. number of reads = 0"
#[1] "2] Max. number of reads = 293267"
#[1] "3] Total number of reads = 11905194"
#[1] "4] Average number of reads = 42367.2384341637"
#[1] "5] Median number of reads = 26978"
#[1] "7] Sparsity = 0.972519587848694"
#[1] "6] Any OTU sum to 1 or less? YES"
#[1] "8] Number of singletons = 21"
#[1] "9] Percent of OTUs that are singletons \n        (i.e. exactly one read detected across all samples)0.23130300693909"
#[1] "10] Number of sample variables are: 14"

Fungi_mereged = subset_taxa(Fungi_mereged, Kingdom=="Fungi");Fungi_mereged#   [ 8450 taxa and 281 samples ]
Fungi_mereged = subset_taxa(Fungi_mereged, Phylum!="CyanoFungi");Fungi_mereged#   [ 8450 taxa and 281 samples ]
Fungi_mereged = subset_taxa(Fungi_mereged, Order!="Chloroplast");Fungi_mereged#   [ 8450 taxa and 281 samples ]
Fungi_mereged = subset_taxa(Fungi_mereged, Family!="Mitochondria");Fungi_mereged#  [ 8450 taxa and 281 samples ]
Fungi_mereged <- prune_taxa(taxa_sums(Fungi_mereged)>0, Fungi_mereged);Fungi_mereged# [ 8450 taxa and 281 samples ]
#OTU_table_Fungi_mereged = as(otu_table(Fungi_mereged), "matrix")
#########################################################################################################
#Summarize 
microbiome::summarize_phyloseq(Fungi_mereged)
#1] "1] Min. number of reads = 0"
#[1] "2] Max. number of reads = 293140"
#[1] "3] Total number of reads = 11774794"
#[1] "4] Average number of reads = 41903.1814946619"
#[1] "5] Median number of reads = 26558"
#1] "7] Sparsity = 0.971471709237929"
#[1] "6] Any OTU sum to 1 or less? YES"
#[1] "8] Number of singletons = 21"
#[1] "9] Percent of OTUs that are singletons \n        (i.e. exactly one read detected across all samples)0.248520710059172"

taxonomy <- as.vector(phyloseq::tax_table(Fungi_mereged))

sums_Fungi_mereged <- sample_sums(Fungi_mereged);sums_Fungi_mereged
sums_Fungi_mereged <- data.frame(sums_Fungi_mereged);sums_Fungi_mereged
sums_Fungi_mereged <- as.matrix(sums_Fungi_mereged);sums_Fungi_mereged
colSums(sums_Fungi_mereged) #  11774794       
max(sums_Fungi_mereged) #293140
min(sums_Fungi_mereged) # 0
sums_Fungi_mereged
##################
save(Fungi_mereged, file = "/Users/expeditoolimi/Documents/Manuscript-AA-OE_SB/Microbiome_analysis/Fungi_mereged.RData")

```


#Rarefaction

```{r,echo=FALSE,warning=FALSE}
setwd("/Users/expeditoolimi/Documents/Manuscript-AA-OE_SB/Microbiome_analysis")
load("/Users/expeditoolimi/Documents/Manuscript-AA-OE_SB/Microbiome_analysis/Fungi_mereged.RData")


Fungi_mereged_noontrol= subset_samples(Fungi_mereged, Sample_type %in% c("4_Phyllosphere","2_Soil","3_Rhizosphere"));Fungi_mereged_noontrol#[ 15421 taxa and 1102 samples ]
p_rarf <- ggrare(Fungi_mereged_noontrol, step = 2045 , color = "Sample_type", se = FALSE)
p_rarf <- p_rarf + facet_wrap(~Sample_type, ncol = 3) + theme_bw()
p_rarf <- p_rarf + scale_color_manual(values = mycola)
plot(p_rarf)
tiff("p_rarf.tiff", units="in", width=8, height=4, res=300)
p_rarf.tiff=p_rarf+plot_layout(ncol =1 );p_rarf.tiff
ggsave("p_rarf.tiff", plot = p_rarf.tiff)
dev.off()
```


#from https://microbiome.github.io/tutorials/Composition.html
```{r,echo=FALSE,warning=FALSE}
setwd("/Users/expeditoolimi/Documents/Manuscript-AA-OE_SB/Microbiome_analysis")
load("/Users/expeditoolimi/Documents/Manuscript-AA-OE_SB/Microbiome_analysis/Fungi_mereged.RData")

#Experiment (subset experiment one)
Fungi_experiment= subset_samples(Fungi_mereged, Experiment %in% c("One"));Fungi_experiment#[ 15421 taxa and 1102 samples ]
Fungi_experiment= subset_samples(Fungi_experiment, Sample_type %in% c("4_Phyllosphere","3_Rhizosphere","2_Soil"));Fungi_experiment#[ 15421 taxa and 1102 samples ]
################################################################################################################################################################################################
Fungi_experiment_table = as(otu_table(Fungi_experiment), "matrix")
# Coerce to data.frame
Fungi_experiment_table = as.data.frame(Fungi_experiment_table)
write.csv(Fungi_experiment_table ,file = "Fungi_experiment_table.csv")

###########################################################################################
library(ggplot2)
library(MicrobiotaProcess)
citation("MicrobiotaProcess")
fclasstaxa <- get_taxadf(obj=Fungi_experiment, taxlevel=3);fclasstaxa
fphylataxa <- get_taxadf(obj=Fungi_experiment, taxlevel=2);fphylataxa
fordertaxa <- get_taxadf(obj=Fungi_experiment, taxlevel=4);fordertaxa
fgentaxa <- get_taxadf(obj=Fungi_experiment, taxlevel=5);fgentaxa
fGentaxa <- get_taxadf(obj=Fungi_experiment, taxlevel=6);fGentaxa
#####################################################################################################
#https://yulab-smu.top/MicrobiotaProcessWorkshop/articles/MicrobiotaProcessWorkshop.html
##########################################################################################################################################################################################################
sample_data(fordertaxa)
#####################################################################################################
#https://yulab-smu.top/MicrobiotaProcessWorkshop/articles/MicrobiotaProcessWorkshop.html
#####################################################################################################
tiff("fgen.tiff", units="in", width=12, height=9, res=300)
fgen <- ggbartax(obj=fGentaxa, facetNames="Treatment_Sample", plotgroup=TRUE, topn=15) +
  xlab("Treatment") +
  ylab("relative abundance (%)") +scale_fill_manual(values=mycolors)+theme(axis.text.x = element_text(angle = 270)) +
  #scale_color_manual(values=c("chartreuse4","bisque4","deepskyblue","Sky Blue","goldenrod", "#ba0d42", "grey85","Orange","coral3","darkred","#0dba2f","#850dba","grey35","Blue","deeppink","blue","Yellow")) + 
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=3))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "gray33", size=0.4))+ 
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=14),
        axis.title.y = element_text(face="bold", colour="gray33", size=14),
        axis.text.x = element_text(colour="gray33", size=11),
        axis.text.y  = element_text(colour="gray33", size=11))+ labs(tag = "B")+
  theme(legend.position = "bottom")+ 
  labs(fill='Phyla')+
  theme(legend.title = element_text(color = "gray33", size = 12),
        legend.text = element_text(color = "gray33", size = 12))+
  theme(text=element_text(family="sans"))+
  theme(axis.title.x = element_text(face="bold", colour="gray33", size=12),
        axis.title.y = element_text(face="bold", colour="gray33", size=12),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="gray33", size=12));fgen
#####################################################################################################
#Combined_picture-order
tiff("Figure3.tiff", units="in", width=19, height=9, res=300)
Figure3.tiff=Bgen+fgen+plot_layout(ncol = 2);Figure3.tiff
ggsave("Figure3.tiff", plot = Figure3.tiff)
#####################################################################################################
Ford_category <- ggbartax(obj=fordertaxa, facetNames="Sample_type_pool2", plotgroup=TRUE, topn=20) +
  xlab("Cultivars") +
  ylab("relative abundance (%)") +scale_fill_manual(values=mycolors)+theme(axis.text.x = element_text(angle = 270)) +
  #scale_color_manual(values=c("chartreuse4","bisque4","deepskyblue","Sky Blue","goldenrod", "#ba0d42", "grey85","Orange","coral3","darkred","#0dba2f","#850dba","grey35","Blue","deeppink","blue","Yellow")) + 
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=3))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=0.4))+ 
  theme(axis.title.x = element_text(face="bold", colour="black", size=14),
        axis.title.y = element_text(face="bold", colour="black", size=14),
        axis.text.x = element_text(colour="black", size=11),
        axis.text.y  = element_text(colour="black", size=11))+ labs(tag = "B")+
  theme(legend.position = "bottom")+ 
  labs(fill='Phyla')+
  theme(legend.title = element_text(color = "black", size = 12),
        legend.text = element_text(color = "black", size = 12))+
  theme(text=element_text(family="sans"))+
  theme(text=element_text(family="sans"))+
  theme(axis.title.x = element_text(face="bold", colour="black", size=12),
        axis.title.y = element_text(face="bold", colour="black", size=12),
        axis.text.x = element_blank(),
        axis.text.y  = element_text( colour="black", size=12));Ford_category
#####################################################################################################
#Combined_picture-order
tiff("Bord_category_comb.tiff", units="in", width=17, height=8, res=300)
Bord_category_comb.tiff=Bord_category+Ford_category+plot_layout(ncol = 2);Bord_category_comb.tiff
ggsave("Bord_category_comb.tiff", plot = Bord_category_comb.tiff)
#####################################################################################################
```

#PCOA-Fungi
```{r,echo=FALSE,warning=FALSE}
setwd("/Users/expeditoolimi/Documents/Manuscript-AA-OE_SB/Microbiome_analysis")
load("/Users/expeditoolimi/Documents/Manuscript-AA-OE_SB/Microbiome_analysis/Fungi_mereged.RData")

pcoacares_fungi <- get_pcoa(obj=Fungi_experiment, distmethod="bray", method="hellinger");pcoacares_fungi#pcoa

#PCA_pcoa_countries of origign
#Visulizing the result
pcoaplot1 <- ggordpoint(obj=pcoacares_fungi, biplot=F, speciesannot=F,
                        factorNames=c("Treatment"), ellipse=F) +
  scale_fill_manual(values=myco)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=2))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=14, face="bold", color = "black"),
        axis.text.y = element_text(size=14, face="bold", color = "black"))+
  theme(text=element_text(size=14,  family="sans"))+theme(legend.position = "")+
  theme(legend.title=element_blank())+labs(tag = "C")+ geom_point(size = 5);pcoaplot1


#PCA_pcoa_countries of origign
#Visulizing the result
#tiff("PCA_PCOA_Region.tiff", units="in", width=12, height=10, res=300)

pcoaplot2 <- ggordpoint(obj=pcoacares_fungi, biplot=F, speciesannot=F,
                        factorNames=c("Sample_type"), ellipse=F) +
  scale_fill_manual(values=myco)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=4))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=14, face="bold", color = "black"),
        axis.text.y = element_text(size=14, face="bold", color = "black"))+
  theme(text=element_text(size=14,  family="sans"))+theme(legend.position = "")+
  theme(legend.title=element_blank())+labs(tag = "C")+ geom_point(size = 5);pcoaplot2

library(ggpubr)
tiff("pcoaplot1.tiff", units="in", width=10, height=8, res=300)
pcoaplot1.tiff=pcoaplot1+plot_layout(ncol = 1);pcoaplot1.tiff
ggsave("pcoaplot1.tiff", plot = pcoaplot1.tiff)
dev.off()

tiff("pcoaplot2.tiff", units="in", width=10, height=8, res=300)
pcoaplot2.tiff=pcoaplot2+plot_layout(ncol = 1);pcoaplot2.tiff
ggsave("pcoaplot2.tiff", plot = pcoaplot2.tiff)
dev.off()
##########################################################################################################################################################################################################
#subset into compartments
Fungi_experiment= subset_samples(Fungi_mereged, Experiment %in% c("One"));Fungi_experiment#[ 15421 taxa and 1102 samples ]
Fungi_experiment= subset_samples(Fungi_experiment, Sample_type %in% c("4_Phyllosphere","3_Rhizosphere","2_Soil"));Fungi_experiment#[ 15421 taxa and 1102 samples ]
Fungi_experiment_soil= subset_samples(Fungi_experiment, Sample_type %in% c("2_Soil"));Fungi_experiment_soil#[ 15421 taxa and 1102 samples ]
pcoacares_fungi_soil <- get_pcoa(obj=Fungi_experiment_soil, distmethod="bray", method="hellinger");pcoacares_fungi_soil#pcoa

#PCA_pcoa_countries of origign
#Visulizing the result
fpcoa_soil <- ggordpoint(obj=pcoacares_fungi_soil, biplot=F, speciesannot=F,
                        factorNames=c("Treatment"), ellipse=F) +
  scale_fill_manual(values=myco)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=2))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=12, face="bold", color = "black"),
        axis.text.y = element_text(size=12, face="bold", color = "black"))+
  theme(text=element_text(size=12,  family="sans"))+theme(legend.position = "")+
  theme(legend.title=element_blank())+labs(tag = "D")+ geom_point(size = 5);fpcoa_soil


############
Fungi_experiment_3_Rhizosphere= subset_samples(Fungi_experiment, Sample_type %in% c("3_Rhizosphere"));Fungi_experiment_3_Rhizosphere#[ 15421 taxa and 1102 samples ]
pcoacares_fungi_3_Rhizosphere <- get_pcoa(obj=Fungi_experiment_3_Rhizosphere, distmethod="bray", method="hellinger");pcoacares_fungi#pcoa

#PCA_pcoa_countries of origign
#Visulizing the result
fpcoa_3_Rhizosphere <- ggordpoint(obj=pcoacares_fungi_3_Rhizosphere, biplot=F, speciesannot=F,
                         factorNames=c("Treatment"), ellipse=F) +
  scale_fill_manual(values=myco)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=2))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=12, face="bold", color = "black"),
        axis.text.y = element_text(size=12, face="bold", color = "black"))+
  theme(text=element_text(size=12,  family="sans"))+theme(legend.position = "")+
  theme(legend.title=element_blank())+labs(tag = "E")+ geom_point(size = 5);fpcoa_3_Rhizosphere

############
Fungi_experiment_Phyllosphere= subset_samples(Fungi_experiment, Sample_type %in% c("4_Phyllosphere"));Fungi_experiment_Phyllosphere#[ 15421 taxa and 1102 samples ]
pcoacares_fungi_Phyllosphere <- get_pcoa(obj=Fungi_experiment_Phyllosphere, distmethod="bray", method="hellinger");pcoacares_fungi#pcoa

#PCA_pcoa_countries of origign
#Visulizing the result
tiff("fpcoa_Phyllosphere.tiff", units="in", width=16, height=8, res=300)

fpcoa_Phyllosphere <- ggordpoint(obj=pcoacares_fungi_Phyllosphere, biplot=F, speciesannot=F,
                                factorNames=c("Treatment"), ellipse=F) +
  scale_fill_manual(values=myco)+
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=1))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1))+
  theme(axis.text.x = element_text(size=12, face="bold", color = "black"),
        axis.text.y = element_text(size=12, face="bold", color = "black"))+
  theme(text=element_text(size=12,  family="sans"))+theme(legend.position = "bottom")+
  theme(legend.title=element_blank())+labs(tag = "F")+ geom_point(size = 5);fpcoa_Phyllosphere

tiff("fpcoa_Phyllosphere.tiff", units="in", width=16, height=8, res=300)
fpcoa_Phyllosphere.tiff=fpcoa_Phyllosphere+plot_layout(nrow = 2);fpcoa_Phyllosphere.tiff
ggsave("fpcoa_Phyllosphere.tiff", plot = fpcoa_Phyllosphere.tiff)

#####################################################################################################
#Combined_picture-class
#tiff("class_combined.tiff", units="in", width=16, height=8, res=300)
class_combined.tiff=Bclass+fclass+plot_layout(ncol = 2);class_combined.tiff
ggsave("class_combined.tiff", plot = class_combined.tiff)

#Combined_picture-families
tiff("fam_combined.tiff", units="in", width=18, height=8, res=300)
fam_combined.tiff=Bfam+fgen+plot_layout(ncol = 2);fam_combined.tiff
ggsave("fam_combined.tiff", plot = fam_combined.tiff)
##########################################################################################################################################################################################################
#Building figure2

#Combined_picture-families
tiff("Figure2.tiff", units="in", width=17, height=13, res=300)
Figure2.tiff=pcoaplot3+pcoaplot1+Bclass+fclass+plot_layout(ncol = 2);Figure2.tiff
ggsave("Figure2.tiff", plot = Figure2.tiff)

#Treatments picture
tiff("pcoa_Figure2.tiff", units="in", width=8, height=10, res=300)
pcoa_Figure2.tiff=pcoaplot3+pcoaplot1+plot_layout(nrow = 2);pcoa_Figure2.tiff
ggsave("pcoa_Figure2.tiff", plot = pcoa_Figure2.tiff)

#Compartmnent picture (S4)
tiff("pcoa_FigureS4.tiff", units="in", width=13, height=6, res=300)
pcoa_FigureS4.tiff=pcoaplot4+pcoaplot2+plot_layout(ncol = 2);pcoa_FigureS4.tiff
ggsave("pcoa_FigureS4.tiff", plot = pcoa_FigureS4.tiff)

#####################################################################################################
#Figure S4-compartment-specific
tiff("pcoacomp_spec_FigureS4.tiff", units="in", width=11, height=6, res=300)
pcoacomp_spec_FigureS4.tiff=bpcoa_soil+bpcoa_3_Rhizosphere+bpcoa_Phyllosphere+fpcoa_soil+fpcoa_3_Rhizosphere+fpcoa_Phyllosphere+plot_layout(ncol = 3);pcoacomp_spec_FigureS4.tiff
ggsave("pcoacomp_spec_FigureS4.tiff", plot = pcoacomp_spec_FigureS4.tiff)

#####################################################################################################
#https://rdrr.io/github/adrientaudiere/MiscMetabar/src/R/plot_functions.R
#https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/ 
#Agglomerate to phylum-level and rename
Fungi_mereged_phylum <- phyloseq::tax_glom(Fungi_mereged, "Phylum")#[ 38 taxa and 1197 samples ]
phyloseq::taxa_names(Fungi_mereged_phylum) <- phyloseq::tax_table(Fungi_mereged_phylum)[, "Phylum"]
phyloseq::otu_table(Fungi_mereged_phylum)[1:5, 1:5]
sample_data(Fungi_mereged_phylum)

#Melt and plot
tiff("Fungi_mereged_phylum.tiff", units="in", width=12, height=9, res=300)
Fungi_mereged_phylum.tiff=phyloseq::psmelt(Fungi_mereged_phylum) %>%
  ggplot(data = ., aes(x =Sample_type, y = Abundance)) +
  geom_boxplot(outlier.shape  = NA) +
  geom_jitter(aes(color = OTU), height = 0, width = .2) +
  labs(x = "", y = "Abundance\n") +
  facet_wrap(~ OTU, scales = "free")+theme(axis.text.x = element_text(angle = 90))+
  scale_color_manual(values=mycola);Fungi_mereged_phylum.tiff
Fungi_mereged_phylum.tiff=Fungi_mereged_phylum.tiff+plot_layout(ncol =1 );Fungi_mereged_phylum.tiff
ggsave("Fungi_mereged_phylum.tiff", plot = Fungi_mereged_phylum.tiff)
dev.off()

########################################################################################################################
#subseting the data for experiment one
Fungi_microcosm_one<- subset_samples(Fungi_mereged, Experiment %in% c("One"));Fungi_microcosm_one#[ 15421 taxa and 1102 samples ]
#remove the control
Fungi_microcosm_one<- subset_samples(Fungi_microcosm_one, Sample_type %in% c("4_Phyllosphere","2_Soil","3_Rhizosphere"));Fungi_microcosm_one#[ 15421 taxa and 1102 samples ]
#####################################################################################################################################################################################################
#The function
#The function
phyloseq_transform_css <- function(physeq, norm = TRUE, log = TRUE, ...){
  # require(metagenomeSeq)
  MGS <- phyloseq::phyloseq_to_metagenomeSeq(physeq)
  # get the normalized count matrix
  otu_norm <- metagenomeSeq::MRcounts(MGS, norm = norm, log = log, ...)
  # exportMat(datt.norm, file = "tmp.txt")    # norm = TRUE, log = TRUE
  ## save sample statistics (sample scaling factor, quantile value, number of identified features and library size):
  # exportStats(dattM.sf, file = "tmp_stats.txt")
  # Substitue raw abundance to the css-normalized data
  physeq.tr <- physeq
  phyloseq::otu_table(physeq.tr) <- phyloseq::otu_table(otu_norm, taxa_are_rows = T)
  return(physeq.tr)
}

##Beta diversity using CSS
Fungi_microcosm_one_CSS = phyloseq_transform_css(Fungi_microcosm_one);Fungi_microcosm_one_CSS#########
sample_data(Fungi_microcosm_one_CSS)
#First get otu_table and transpose it:
dist.matrix_Fungi_microcosm_one_CSS <- t(data.frame(otu_table(Fungi_microcosm_one_CSS)))
sampledf_Fungi_microcosm_one_CSS <- data.frame(sample_data(Fungi_microcosm_one_CSS))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS <- vegdist(dist.matrix_Fungi_microcosm_one_CSS, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS ~ Sample_type*Treatment_type*Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#                             Df SumOfSqs  R2      F       Pr(>F)    
#Sample_type                  2   7.9098 0.30478 36.6891  0.001 ***
#Treatment_type               2   0.9575 0.03689  4.4413  0.001 ***
#Treatment                    5   1.7769 0.06847  3.2968  0.001 ***
#Sample_type:Treatment_type   4   0.8662 0.03337  2.0088  0.002 ** 
#Sample_type:Treatment       10   1.5068 0.05806  1.3978  0.018 *  
#Residual                   120  12.9353 0.49842                   
#Total                      143  25.9524 1.00000                   
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
###########################################################################
###########################################################################

#PCOA
#visualize
matrixdistance_Fungi_microcosm_one_CSS_PCoA <- ordinate(physeq = Fungi_microcosm_one_CSS, method = "PCoA", distance = "bray", trymax = 1000)
Fig2_1=plot_ordination(physeq = Fungi_microcosm_one_CSS, ordination = matrixdistance_Fungi_microcosm_one_CSS_PCoA, color = "Treatment",shape = "Sample_type") + 
  theme(text = element_text(size = 12)) + scale_color_manual(values =myco)+theme(legend.title=element_blank())+  geom_point(aes(color =Treatment ), size = 7)+
  scale_shape_manual(values=c(19,17, 15,18))+ theme_gray(base_size = 14);Fig2_1
tiff("Fig2_1.tiff", units="in", width=11, height=6, res=300)
ggsave("Fig2_1.tiff", plot = Fig2_1)
dev.off()
###########################################################################
Fungi_microcosm_one_CSS_bact_control= subset_samples(Fungi_microcosm_one_CSS, Treatment_type %in% c("Bacteria","Control"));Fungi_microcosm_one_CSS_bact_control#[ 15421 taxa and 1102 samples ]
#First get otu_table and transpose it:
dist.matrix_Fungi_microcosm_one_CSS_bact_control <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_bact_control)))
sampledf_Fungi_microcosm_one_CSS_bact_control <- data.frame(sample_data(Fungi_microcosm_one_CSS_bact_control))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_bact_control <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_bact_control, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_bact_control
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_bact_control), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_bact_control)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_bact_control ~ Treatment_type, data=sampleda, permutation=999);adores#change from adonis to adonis2
#                Df SumOfSqs      R2      F Pr(>F)   
#Treatment_type  1   0.5707 0.04538 3.3278  0.003 **
#Residual       70  12.0045 0.95462                 
#Total          71  12.5752 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
###########################################################################
###########################################################################

#Volatiles
Volatile_microcosm_one_CSS_Volatile_control= subset_samples(Fungi_microcosm_one_CSS, Treatment_type %in% c("Volatile","Control"));Volatile_microcosm_one_CSS_Volatile_control#[ 15421 taxa and 1102 samples ]
#First get otu_table and transpose it:
dist.matrix_Volatile_microcosm_one_CSS_Volatile_control <- t(data.frame(otu_table(Volatile_microcosm_one_CSS_Volatile_control)))
sampledf_Volatile_microcosm_one_CSS_Volatile_control <- data.frame(sample_data(Volatile_microcosm_one_CSS_Volatile_control))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Volatile_microcosm_one_CSS_Volatile_control <- vegdist(dist.matrix_Volatile_microcosm_one_CSS_Volatile_control, method = "bray"); phyloseq_bray_Volatile_microcosm_one_CSS_Volatile_control
sampleda <- data.frame(sample_data(Volatile_microcosm_one_CSS_Volatile_control), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Volatile_microcosm_one_CSS_Volatile_control)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Volatile_microcosm_one_CSS_Volatile_control ~ Treatment_type, data=sampleda, permutation=999);adores#change from adonis to adonis2
#               Df  SumOfSqs    R2    F Pr(>F)  
#Treatment_type  1   0.4574 0.02845 2.5768  0.019 *
#Residual       88  15.6198 0.97155                
#Total          89  16.0771 1.00000                
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1  
#####################################################################################################

#Subset the different compartments
#Phyllosphere
Fungi_microcosm_one_CSS_Phyllosphere= subset_samples(Fungi_microcosm_one_CSS, Sample_type %in% c("4_Phyllosphere"));Fungi_microcosm_one_CSS_Phyllosphere#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_Phyllosphere <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_Phyllosphere)))
sampledf_Fungi_microcosm_one_CSS_Phyllosphere <- data.frame(sample_data(Fungi_microcosm_one_CSS_Phyllosphere))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_Phyllosphere, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_Phyllosphere), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2     F Pr(>F)    
#Treatment  7   2.3107 0.33004 2.815  0.001 ***
#Residual  40   4.6906 0.66996                 
#Total     47   7.0013 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1


############################################################
############################################################
#Fungi vs control in the Phyllosphere
Fungi_microcosm_one_CSS_Phyllosphere_Bact_control= subset_samples(Fungi_microcosm_one_CSS_Phyllosphere, Pool %in% c("Bacteria","Control"));Fungi_microcosm_one_CSS_Phyllosphere_Bact_control#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_Phyllosphere_Bact_control <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_Phyllosphere_Bact_control)))
sampledf_Fungi_microcosm_one_CSS_Phyllosphere_Bact_control <- data.frame(sample_data(Fungi_microcosm_one_CSS_Phyllosphere_Bact_control))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_Bact_control <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_Phyllosphere_Bact_control, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_Bact_control
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_Phyllosphere_Bact_control), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_Bact_control)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_Bact_control ~ Pool, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Pool      1   0.2782 0.07796 1.8602  0.009 **
#Residual 22   3.2901 0.92204                 
#Total    23   3.5682 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

####################################
####################################

#Volatile vs control in the Phyllosphere
Fungi_microcosm_one_CSS_Phyllosphere_vola_control= subset_samples(Fungi_microcosm_one_CSS_Phyllosphere, Pool %in% c("Volatiles","Control"));Fungi_microcosm_one_CSS_Phyllosphere_vola_control#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_Phyllosphere_vola_control <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_Phyllosphere_vola_control)))
sampledf_Fungi_microcosm_one_CSS_Phyllosphere_vola_control <- data.frame(sample_data(Fungi_microcosm_one_CSS_Phyllosphere_vola_control))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_vola_control <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_Phyllosphere_vola_control, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_vola_control
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_Phyllosphere_vola_control), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_vola_control)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_vola_control ~ Pool, data=sampleda, permutation=999);adores#change from adonis to adonis2
#         Df SumOfSqs     R2      F Pr(>F)   
#Pool      1   0.2651 0.0665 1.9946  0.005 **
#Residual 28   3.7218 0.9335                 
#Total    29   3.9870 1.0000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1              
############################################################
############################################################

#No significantly different taxa
# S. plymuthica HRO C48  vs control_phyllosphere
##########
Fungi_microcosm_one_CSS_Phyllosphere_ser_cont= subset_samples(Fungi_microcosm_one_CSS_Phyllosphere, Treatment %in% c("2_Serratia_plymuthica_HRO_C48","1_Control"));Fungi_microcosm_one_CSS_Phyllosphere_ser_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_Phyllosphere_ser_cont <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_Phyllosphere_ser_cont)))
Fungi_microcosm_one_CSS_Phyllosphere_ser_cont <- data.frame(sample_data(Fungi_microcosm_one_CSS_Phyllosphere_ser_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_ser_cont <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_Phyllosphere_ser_cont, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_ser_cont
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_Phyllosphere_ser_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_ser_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_ser_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#            Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.37914 0.23855 3.1328  0.002 **
#Residual  10  1.21023 0.76145                 
#Total     11  1.58937 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

##########
# 3_Stenotrophomonas_rhizophilla_SPA_P69  vs control_phyllosphere
##########
Fungi_microcosm_one_CSS_Phyllosphere_Steno_cont= subset_samples(Fungi_microcosm_one_CSS_Phyllosphere, Treatment %in% c("3_Stenotrophomonas_rhizophilla_SPA_P69","1_Control"));Fungi_microcosm_one_CSS_Phyllosphere_Steno_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_Phyllosphere_Steno_cont <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_Phyllosphere_Steno_cont)))
Fungi_microcosm_one_CSS_Phyllosphere_Steno_cont <- data.frame(sample_data(Fungi_microcosm_one_CSS_Phyllosphere_Steno_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_Steno_cont <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_Phyllosphere_Steno_cont, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_Steno_cont
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_Phyllosphere_Steno_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_Steno_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_Steno_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.26895 0.17075 2.0591  0.004 **
#Residual  10  1.30615 0.82925                 
#Total     11  1.57509 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1


##########
# 4_Fungi_Mixture  vs control_phyllosphere
##########
Fungi_microcosm_one_CSS_Phyllosphere_consortium_cont= subset_samples(Fungi_microcosm_one_CSS_Phyllosphere, Treatment %in% c("4_Bacteria_Mixture","1_Control"));Fungi_microcosm_one_CSS_Phyllosphere_consortium_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_Phyllosphere_consortium_cont <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_Phyllosphere_consortium_cont)))
Fungi_microcosm_one_CSS_Phyllosphere_consortium_cont <- data.frame(sample_data(Fungi_microcosm_one_CSS_Phyllosphere_consortium_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_consortium_cont <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_Phyllosphere_consortium_cont, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_consortium_cont
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_Phyllosphere_consortium_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_consortium_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_consortium_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.25481 0.15683 1.8599  0.005 **
#Residual  10  1.37001 0.84317                 
#Total     11  1.62482 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

##########
# 5_3-Methyl-1-Butanol  vs control_phyllosphere
##########
Fungi_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont= subset_samples(Fungi_microcosm_one_CSS_Phyllosphere, Treatment %in% c("5_3-Methyl-1-Butanol","1_Control"));Fungi_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont)))
Fungi_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont <- data.frame(sample_data(Fungi_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_5_3_Methyl_1_Butanol_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs     R2     F Pr(>F)   
#Treatment  1  0.29926 0.1975 2.461  0.003 **
#Residual  10  1.21599 0.8025                
#Total     11  1.51525 1.0000                
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
##########
# 6_2-Butanone  vs control_phyllosphere
##########
Fungi_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont= subset_samples(Fungi_microcosm_one_CSS_Phyllosphere, Treatment %in% c("6_2-Butanone","1_Control"));Fungi_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont)))
Fungi_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont <- data.frame(sample_data(Fungi_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_6_2_Butanone_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.27336 0.20235 2.5368  0.002 **
#Residual  10  1.07757 0.79765                 
#Total     11  1.35092 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

##########
#7_2-Nonanone
##########
Fungi_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont= subset_samples(Fungi_microcosm_one_CSS_Phyllosphere, Treatment %in% c("7_2-Nonanone","1_Control"));Fungi_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont)))
Fungi_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont <- data.frame(sample_data(Fungi_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere_7_2_Nonanone_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.31112 0.20102 2.5159  0.005 **
#Residual  10  1.23659 0.79898                 
#Total     11  1.54771 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

##########
#8_Volatile_Mixture
##########
Fungi_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont= subset_samples(Fungi_microcosm_one_CSS_Phyllosphere, Treatment %in% c("8_Volatile_Mixture","1_Control"));Fungi_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont)))
Fungi_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont <- data.frame(sample_data(Fungi_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_Phyllosphere8_Volatile_Mixture_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2

#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.27085 0.19167 2.3712  0.002 **
#Residual  10  1.14221 0.80833                 
#Total     11  1.41305 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

#####################################################################################################
Fungi_microcosm_one<- subset_samples(Fungi_mereged, Experiment %in% c("One"));Fungi_microcosm_one#[ 15421 taxa and 1102 samples ]
Fungi_microcosm_one<- subset_samples(Fungi_microcosm_one, Sample_type %in% c("4_Phyllosphere","2_Soil","3_Rhizosphere"));Fungi_microcosm_one#[ 15421 taxa and 1102 samples ]
Fungi_microcosm_one_Phyllosphere= subset_samples(Fungi_microcosm_one, Sample_type %in% c("4_Phyllosphere"));Fungi_microcosm_one_Phyllosphere#[ 15421 taxa and 1102 samples ]

#serratia vs control
Fungi_microcosm_one_Phyllosphere_ser_cont= subset_samples(Fungi_microcosm_one_Phyllosphere, Treatment %in% c("2_Serratia_plymuthica_HRO_C48","1_Control"));Fungi_microcosm_one_Phyllosphere_ser_cont#[ 15421 taxa and 1102 samples ]
diagdds_ser = phyloseq_to_deseq2(Fungi_microcosm_one_Phyllosphere_ser_cont, ~ Treatment);diagdds_ser
diagdds_ser = DESeq(diagdds_ser, test="Wald", fitType="parametric");diagdds_ser
res = results(diagdds_ser, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_ser = res[which(res$padj < alpha), ];sigtab_ser
sigtab_ser = cbind(as(sigtab_ser, "data.frame"), as(tax_table(Fungi_microcosm_one_Phyllosphere_ser_cont)[rownames(sigtab_ser), ], "matrix"));sigtab_ser
head(sigtab_ser)
write.csv(sigtab_ser, "sigtab_ser.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_ser$log2FoldChange, sigtab_ser$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser$Phylum = factor(as.character(sigtab_ser$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_ser$log2FoldChange, sigtab_ser$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser$Genus = factor(as.character(sigtab_ser$Genus), levels=names(x))
ggplot(sigtab_ser, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Eurotiomycetes,Unidentified_fungi
#####################################################################################################
#steno vs control
Fungi_microcosm_one_Phyllosphere_steno_cont= subset_samples(Fungi_microcosm_one_Phyllosphere, Treatment %in% c("3_Stenotrophomonas_rhizophilla_SPA_P69","1_Control"));Fungi_microcosm_one_Phyllosphere_steno_cont#[ 15421 taxa and 1102 samples ]
diagdds_Steno = phyloseq_to_deseq2(Fungi_microcosm_one_Phyllosphere_steno_cont, ~ Treatment);diagdds_Steno
diagdds_Steno = DESeq(diagdds_Steno, test="Wald", fitType="parametric");diagdds_Steno
res = results(diagdds_Steno, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_Steno = res[which(res$padj < alpha), ];sigtab_Steno
sigtab_Steno = cbind(as(sigtab_Steno, "data.frame"), as(tax_table(Fungi_microcosm_one_Phyllosphere_steno_cont)[rownames(sigtab_Steno), ], "matrix"));sigtab_Steno
head(sigtab_Steno)
write.csv(sigtab_Steno, "sigtab_Steno.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_Steno$log2FoldChange, sigtab_Steno$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_Steno$Phylum = factor(as.character(sigtab_Steno$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_Steno$log2FoldChange, sigtab_Steno$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_Steno$Genus = factor(as.character(sigtab_Steno$Genus), levels=names(x))
ggplot(sigtab_Steno, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Agaricomycetes,Dothideomycetes, Endogomycetes, Eurotiomycetes, Microbotryomycetes, and Sordariomycetes
#####################################################################################################
#Consort vs control
Fungi_microcosm_one_Phyllosphere_consortium_cont= subset_samples(Fungi_microcosm_one_Phyllosphere, Treatment %in% c("4_Bacteria_Mixture","1_Control"));Fungi_microcosm_one_Phyllosphere_consortium_cont#[ 15421 taxa and 1102 samples ]
diagdds_Consortium = phyloseq_to_deseq2(Fungi_microcosm_one_Phyllosphere_consortium_cont, ~ Treatment);diagdds_Consortium
diagdds_Consortium = DESeq(diagdds_Consortium, test="Wald", fitType="parametric");diagdds_Consortium
res = results(diagdds_Consortium, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_Consortium = res[which(res$padj < alpha), ];sigtab_Consortium
sigtab_Consortium = cbind(as(sigtab_Consortium, "data.frame"), as(tax_table(Fungi_microcosm_one_Phyllosphere_consortium_cont)[rownames(sigtab_Consortium), ], "matrix"));sigtab_Consortium
head(sigtab_Consortium)
write.csv(sigtab_Consortium, "sigtab_Consortium.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_Consortium$log2FoldChange, sigtab_Consortium$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_Consortium$Phylum = factor(as.character(sigtab_Consortium$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_Consortium$log2FoldChange, sigtab_Consortium$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_Consortium$Genus = factor(as.character(sigtab_Consortium$Genus), levels=names(x))
ggplot(sigtab_Consortium, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))

# Eurotiomycetes (Aspergillaceae Penicillium Unidentified_Penicilliu)
#####################################################################################################
#Consort vs control
Fungi_microcosm_one_Phyllosphere_Methyl_cont= subset_samples(Fungi_microcosm_one_Phyllosphere, Treatment %in% c("5_3-Methyl-1-Butanol","1_Control"));Fungi_microcosm_one_Phyllosphere_Methyl_cont#[ 15421 taxa and 1102 samples ]
diagdds_Methyl = phyloseq_to_deseq2(Fungi_microcosm_one_Phyllosphere_Methyl_cont, ~ Treatment);diagdds_Methyl
diagdds_Methyl = DESeq(diagdds_Methyl, test="Wald", fitType="parametric");diagdds_Methyl
res = results(diagdds_Methyl, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_Methyl = res[which(res$padj < alpha), ];sigtab_Methyl
sigtab_Methyl = cbind(as(sigtab_Methyl, "data.frame"), as(tax_table(Fungi_microcosm_one_Phyllosphere_Methyl_cont)[rownames(sigtab_Methyl), ], "matrix"));sigtab_Methyl
head(sigtab_Methyl)
write.csv(sigtab_Methyl, "sigtab_Methyl.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_Methyl$log2FoldChange, sigtab_Methyl$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_Methyl$Phylum = factor(as.character(sigtab_Methyl$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_Methyl$log2FoldChange, sigtab_Methyl$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_Methyl$Genus = factor(as.character(sigtab_Methyl$Genus), levels=names(x))
ggplot(sigtab_Methyl, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Agaricomycetes, Eurotiomycetes
#####################################################################################################
#Butanone
#Consort vs control
Fungi_microcosm_one_Phyllosphere_Butanone_cont= subset_samples(Fungi_microcosm_one_Phyllosphere, Treatment %in% c("6_2-Butanone","1_Control"));Fungi_microcosm_one_Phyllosphere_Butanone_cont#[ 15421 taxa and 1102 samples ]
diagdds_Butanone = phyloseq_to_deseq2(Fungi_microcosm_one_Phyllosphere_Butanone_cont, ~ Treatment);diagdds_Butanone
diagdds_Butanone = DESeq(diagdds_Butanone, test="Wald", fitType="parametric");diagdds_Butanone
res = results(diagdds_Butanone, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_Butanone = res[which(res$padj < alpha), ];sigtab_Butanone
sigtab_Butanone = cbind(as(sigtab_Butanone, "data.frame"), as(tax_table(Fungi_microcosm_one_Phyllosphere_Butanone_cont)[rownames(sigtab_Butanone), ], "matrix"));sigtab_Butanone
head(sigtab_Butanone)
write.csv(sigtab_Butanone, "sigtab_Butanone.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_Butanone$log2FoldChange, sigtab_Butanone$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_Butanone$Phylum = factor(as.character(sigtab_Butanone$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_Butanone$log2FoldChange, sigtab_Butanone$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_Butanone$Genus = factor(as.character(sigtab_Butanone$Genus), levels=names(x))
ggplot(sigtab_Butanone, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Agaricomycetes, Endogonomycetes, and Eurotiomycetes
#####################################################################################################
#7_2-Nonanone vs control
Fungi_microcosm_one_Phyllosphere_7_2_Nonanone_cont= subset_samples(Fungi_microcosm_one_Phyllosphere, Treatment %in% c("7_2-Nonanone","1_Control"));Fungi_microcosm_one_Phyllosphere_7_2_Nonanone_cont#[ 15421 taxa and 1102 samples ]
diagdds_Nonanone = phyloseq_to_deseq2(Fungi_microcosm_one_Phyllosphere_7_2_Nonanone_cont, ~ Treatment);diagdds_Nonanone
diagdds_Nonanone = DESeq(diagdds_Nonanone, test="Wald", fitType="parametric");diagdds_Nonanone
res = results(diagdds_Nonanone, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_Nonanone = res[which(res$padj < alpha), ];sigtab_Nonanone
sigtab_Nonanone = cbind(as(sigtab_Nonanone, "data.frame"), as(tax_table(Fungi_microcosm_one_Phyllosphere_7_2_Nonanone_cont)[rownames(sigtab_Nonanone), ], "matrix"));sigtab_Nonanone
head(sigtab_Nonanone)
write.csv(sigtab_Nonanone, "sigtab_Nonanone.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_Nonanone$log2FoldChange, sigtab_Nonanone$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_Nonanone$Phylum = factor(as.character(sigtab_Nonanone$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_Nonanone$log2FoldChange, sigtab_Nonanone$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_Nonanone$Genus = factor(as.character(sigtab_Nonanone$Genus), levels=names(x))
ggplot(sigtab_Nonanone, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#ndogonomycetes, and Sordariomycetes
#####################################################################################################
#8_Volatile_Mixture
Fungi_microcosm_one_Phyllosphere8_Volatile_Mixture_cont= subset_samples(Fungi_microcosm_one_Phyllosphere, Treatment %in% c("8_Volatile_Mixture","1_Control"));Fungi_microcosm_one_Phyllosphere8_Volatile_Mixture_cont#[ 15421 taxa and 1102 samples ]
diagdds_Volatile_mixture = phyloseq_to_deseq2(Fungi_microcosm_one_Phyllosphere8_Volatile_Mixture_cont, ~ Treatment);diagdds_Volatile_mixture
diagdds_Volatile_mixture = DESeq(diagdds_Volatile_mixture, test="Wald", fitType="parametric");diagdds_Volatile_mixture
res = results(diagdds_Volatile_mixture, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_Volatile_mixture = res[which(res$padj < alpha), ];sigtab_Volatile_mixture
sigtab_Volatile_mixture = cbind(as(sigtab_Volatile_mixture, "data.frame"), as(tax_table(Fungi_microcosm_one_Phyllosphere8_Volatile_Mixture_cont)[rownames(sigtab_Volatile_mixture), ], "matrix"));sigtab_Volatile_mixture
head(sigtab_Volatile_mixture)
write.csv(sigtab_Volatile_mixture, "sigtab_Volatile_mixture.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_Volatile_mixture$log2FoldChange, sigtab_Volatile_mixture$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_Volatile_mixture$Phylum = factor(as.character(sigtab_Volatile_mixture$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_Volatile_mixture$log2FoldChange, sigtab_Volatile_mixture$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_Volatile_mixture$Genus = factor(as.character(sigtab_Volatile_mixture$Genus), levels=names(x))
ggplot(sigtab_Volatile_mixture, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Eurotiomycetes

#####################################################################################################
#Subset the different compartments
#3_Rhizosphere
#Subset the different compartments
#3_Rhizosphere
Fungi_microcosm_one_CSS_3_Rhizosphere= subset_samples(Fungi_microcosm_one_CSS, Sample_type %in% c("3_Rhizosphere"));Fungi_microcosm_one_CSS_3_Rhizosphere#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_3_Rhizosphere <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_3_Rhizosphere)))
sampledf_Fungi_microcosm_one_CSS_3_Rhizosphere <- data.frame(sample_data(Fungi_microcosm_one_CSS_3_Rhizosphere))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_3_Rhizosphere, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_3_Rhizosphere), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)    
#Treatment  7   1.5227 0.21815 1.5944  0.001 ***
#Residual  40   5.4573 0.78185                  
#Total     47   6.9800 1.00000                  
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

############################################################
############################################################
#Fungi vs control in the 3_Rhizosphere
Fungi_microcosm_one_CSS_3_Rhizosphere_Bact_control= subset_samples(Fungi_microcosm_one_CSS_3_Rhizosphere, Pool %in% c("Bacteria","Control"));Fungi_microcosm_one_CSS_3_Rhizosphere_Bact_control#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_3_Rhizosphere_Bact_control <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_3_Rhizosphere_Bact_control)))
sampledf_Fungi_microcosm_one_CSS_3_Rhizosphere_Bact_control <- data.frame(sample_data(Fungi_microcosm_one_CSS_3_Rhizosphere_Bact_control))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_Bact_control <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_3_Rhizosphere_Bact_control, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_Bact_control
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_3_Rhizosphere_Bact_control), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_Bact_control)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_Bact_control ~ Pool, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)    
#Pool      1   0.3164 0.09937 2.4273  0.001 ***
#Residual 22   2.8678 0.90063                  
#Total    23   3.1842 1.00000                  
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

#Volatile vs control in the 3_Rhizosphere
Fungi_microcosm_one_CSS_3_Rhizosphere_vola_control= subset_samples(Fungi_microcosm_one_CSS_3_Rhizosphere, Pool %in% c("Volatiles","Control"));Fungi_microcosm_one_CSS_3_Rhizosphere_vola_control#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_3_Rhizosphere_vola_control <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_3_Rhizosphere_vola_control)))
sampledf_Fungi_microcosm_one_CSS_3_Rhizosphere_vola_control <- data.frame(sample_data(Fungi_microcosm_one_CSS_3_Rhizosphere_vola_control))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_vola_control <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_3_Rhizosphere_vola_control, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_vola_control
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_3_Rhizosphere_vola_control), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_vola_control)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_vola_control ~ Pool, data=sampleda, permutation=999);adores#change from adonis to adonis2
#         Df SumOfSqs      R2      F Pr(>F)
#Pool      1   0.1612 0.03735 1.0864  0.246
#Residual 28   4.1549 0.96265              
#Total    29   4.3161 1.00000              
############################################################
############################################################
# S. plymuthica HRO C48  vs control_3_Rhizosphere
##########
Fungi_microcosm_one_CSS_3_Rhizosphere_ser_cont= subset_samples(Fungi_microcosm_one_CSS_3_Rhizosphere, Treatment %in% c("2_Serratia_plymuthica_HRO_C48","1_Control"));Fungi_microcosm_one_CSS_3_Rhizosphere_ser_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_3_Rhizosphere_ser_cont <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_3_Rhizosphere_ser_cont)))
Fungi_microcosm_one_CSS_3_Rhizosphere_ser_cont <- data.frame(sample_data(Fungi_microcosm_one_CSS_3_Rhizosphere_ser_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_ser_cont <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_3_Rhizosphere_ser_cont, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_ser_cont
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_3_Rhizosphere_ser_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_ser_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_ser_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.33285 0.21347 2.7141  0.003 **
#Residual  10  1.22637 0.78653                 
#Total     11  1.55922 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

##########
# 3_Stenotrophomonas_rhizophilla_SPA_P69  vs control_3_Rhizosphere
##########
Fungi_microcosm_one_CSS_3_Rhizosphere_Steno_cont= subset_samples(Fungi_microcosm_one_CSS_3_Rhizosphere, Treatment %in% c("3_Stenotrophomonas_rhizophilla_SPA_P69","1_Control"));Fungi_microcosm_one_CSS_3_Rhizosphere_Steno_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_3_Rhizosphere_Steno_cont <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_3_Rhizosphere_Steno_cont)))
Fungi_microcosm_one_CSS_3_Rhizosphere_Steno_cont <- data.frame(sample_data(Fungi_microcosm_one_CSS_3_Rhizosphere_Steno_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_Steno_cont <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_3_Rhizosphere_Steno_cont, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_Steno_cont
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_3_Rhizosphere_Steno_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_Steno_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_Steno_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.22195 0.14445 1.6884  0.007 **
#Residual  10  1.31456 0.85555                 
#Total     11  1.53650 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

##########
# 4_Fungi_Mixture  vs control_3_Rhizosphere
##########
Fungi_microcosm_one_CSS_3_Rhizosphere_consortium_cont= subset_samples(Fungi_microcosm_one_CSS_3_Rhizosphere, Treatment %in% c("4_Bacteria_Mixture","1_Control"));Fungi_microcosm_one_CSS_3_Rhizosphere_consortium_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_3_Rhizosphere_consortium_cont <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_3_Rhizosphere_consortium_cont)))
Fungi_microcosm_one_CSS_3_Rhizosphere_consortium_cont <- data.frame(sample_data(Fungi_microcosm_one_CSS_3_Rhizosphere_consortium_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_consortium_cont <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_3_Rhizosphere_consortium_cont, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_consortium_cont
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_3_Rhizosphere_consortium_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_consortium_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_consortium_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.23338 0.14677 1.7202  0.009 **
#Residual  10  1.35674 0.85323                 
#Total     11  1.59012 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
#####################################################################################################
Fungi_microcosm_one<- subset_samples(Fungi_mereged, Experiment %in% c("One"));Fungi_microcosm_one#[ 15421 taxa and 1102 samples ]
Fungi_microcosm_one<- subset_samples(Fungi_microcosm_one, Sample_type %in% c("4_Phyllosphere","2_Soil","3_Rhizosphere"));Fungi_microcosm_one#[ 15421 taxa and 1102 samples ]
Fungi_microcosm_one_3_Rhizosphere= subset_samples(Fungi_microcosm_one, Sample_type %in% c("3_Rhizosphere"));Fungi_microcosm_one_3_Rhizosphere#[ 15421 taxa and 1102 samples ]


#serratia vs control
Fungi_microcosm_one_3_Rhizosphere_ser_cont= subset_samples(Fungi_microcosm_one_3_Rhizosphere, Treatment %in% c("2_Serratia_plymuthica_HRO_C48","1_Control"));Fungi_microcosm_one_3_Rhizosphere_ser_cont#[ 15421 taxa and 1102 samples ]
diagdds_ser = phyloseq_to_deseq2(Fungi_microcosm_one_3_Rhizosphere_ser_cont, ~ Treatment);diagdds_ser
diagdds_ser = DESeq(diagdds_ser, test="Wald", fitType="parametric");diagdds_ser
res = results(diagdds_ser, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_ser = res[which(res$padj < alpha), ];sigtab_ser
sigtab_ser = cbind(as(sigtab_ser, "data.frame"), as(tax_table(Fungi_microcosm_one_3_Rhizosphere_ser_cont)[rownames(sigtab_ser), ], "matrix"));sigtab_ser
head(sigtab_ser)
write.csv(sigtab_ser, "sigtab_ser.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_ser$log2FoldChange, sigtab_ser$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser$Phylum = factor(as.character(sigtab_ser$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_ser$log2FoldChange, sigtab_ser$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser$Genus = factor(as.character(sigtab_ser$Genus), levels=names(x))
ggplot(sigtab_ser, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Dothideomycetes, Endogomycetes, Leotiomycetes, Mortierellomycetes, and Sordariomycetes
#####################################################################################################
#steno vs control
Fungi_microcosm_one_3_Rhizosphere_steno_cont= subset_samples(Fungi_microcosm_one_3_Rhizosphere, Treatment %in% c("3_Stenotrophomonas_rhizophilla_SPA_P69","1_Control"));Fungi_microcosm_one_3_Rhizosphere_steno_cont#[ 15421 taxa and 1102 samples ]
diagdds_Steno = phyloseq_to_deseq2(Fungi_microcosm_one_3_Rhizosphere_steno_cont, ~ Treatment);diagdds_Steno
diagdds_Steno = DESeq(diagdds_Steno, test="Wald", fitType="parametric");diagdds_Steno
res = results(diagdds_Steno, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_Steno = res[which(res$padj < alpha), ];sigtab_Steno
sigtab_Steno = cbind(as(sigtab_Steno, "data.frame"), as(tax_table(Fungi_microcosm_one_3_Rhizosphere_steno_cont)[rownames(sigtab_Steno), ], "matrix"));sigtab_Steno
head(sigtab_Steno)
write.csv(sigtab_Steno, "sigtab_Steno.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_Steno$log2FoldChange, sigtab_Steno$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_Steno$Phylum = factor(as.character(sigtab_Steno$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_Steno$log2FoldChange, sigtab_Steno$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_Steno$Genus = factor(as.character(sigtab_Steno$Genus), levels=names(x))
ggplot(sigtab_Steno, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
##Agaricomycetes,Dothideomycetes, Endogomycetes, Leotiomycetes, and Sordariomycetes
#####################################################################################################
#Consort vs control
Fungi_microcosm_one_3_Rhizosphere_consortium_cont= subset_samples(Fungi_microcosm_one_3_Rhizosphere, Treatment %in% c("4_Bacteria_Mixture","1_Control"));Fungi_microcosm_one_3_Rhizosphere_consortium_cont#[ 15421 taxa and 1102 samples ]
diagdds_Consortium = phyloseq_to_deseq2(Fungi_microcosm_one_3_Rhizosphere_consortium_cont, ~ Treatment);diagdds_Consortium
diagdds_Consortium = DESeq(diagdds_Consortium, test="Wald", fitType="parametric");diagdds_Consortium
res = results(diagdds_Consortium, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_Consortium = res[which(res$padj < alpha), ];sigtab_Consortium
sigtab_Consortium = cbind(as(sigtab_Consortium, "data.frame"), as(tax_table(Fungi_microcosm_one_3_Rhizosphere_consortium_cont)[rownames(sigtab_Consortium), ], "matrix"));sigtab_Consortium
head(sigtab_Consortium)
write.csv(sigtab_Consortium, "sigtab_Consortium.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_Consortium$log2FoldChange, sigtab_Consortium$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_Consortium$Phylum = factor(as.character(sigtab_Consortium$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_Consortium$log2FoldChange, sigtab_Consortium$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_Consortium$Genus = factor(as.character(sigtab_Consortium$Genus), levels=names(x))
ggplot(sigtab_Consortium, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))

##Agaricomycetes,Cystobasidiomycetes, Endogomycetes, Eurotiomycetes, Mortierellomycetes, and Sordariomycetes

#####################################################################################################

Fungi_microcosm_one_CSS_3_Rhizosphere_5_3_Methyl_1_Butanol_cont= subset_samples(Fungi_microcosm_one_CSS_3_Rhizosphere, Treatment %in% c("5_3-Methyl-1-Butanol","1_Control"));Fungi_microcosm_one_CSS_3_Rhizosphere_5_3_Methyl_1_Butanol_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_3_Rhizosphere_5_3_Methyl_1_Butanol_cont <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_3_Rhizosphere_5_3_Methyl_1_Butanol_cont)))
Fungi_microcosm_one_CSS_3_Rhizosphere_5_3_Methyl_1_Butanol_cont <- data.frame(sample_data(Fungi_microcosm_one_CSS_3_Rhizosphere_5_3_Methyl_1_Butanol_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_5_3_Methyl_1_Butanol_cont <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_3_Rhizosphere_5_3_Methyl_1_Butanol_cont, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_5_3_Methyl_1_Butanol_cont
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_3_Rhizosphere_5_3_Methyl_1_Butanol_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_5_3_Methyl_1_Butanol_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_5_3_Methyl_1_Butanol_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)
#Treatment  1  0.17648 0.11654 1.3191   0.13
#Residual  10  1.33791 0.88346              
#Total     11  1.51440 1.00000

##########
# 6_2-Butanone  vs control_3_Rhizosphere
##########
Fungi_microcosm_one_CSS_3_Rhizosphere_6_2_Butanone_cont= subset_samples(Fungi_microcosm_one_CSS_3_Rhizosphere, Treatment %in% c("6_2-Butanone","1_Control"));Fungi_microcosm_one_CSS_3_Rhizosphere_6_2_Butanone_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_3_Rhizosphere_6_2_Butanone_cont <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_3_Rhizosphere_6_2_Butanone_cont)))
Fungi_microcosm_one_CSS_3_Rhizosphere_6_2_Butanone_cont <- data.frame(sample_data(Fungi_microcosm_one_CSS_3_Rhizosphere_6_2_Butanone_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_6_2_Butanone_cont <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_3_Rhizosphere_6_2_Butanone_cont, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_6_2_Butanone_cont
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_3_Rhizosphere_6_2_Butanone_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_6_2_Butanone_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_6_2_Butanone_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)
#Treatment  1  0.13955 0.09619 1.0642  0.274
#Residual  10  1.31131 0.90381              
#Total     11  1.45087 1.00000              
##########
#7_2-Nonanone
##########
Fungi_microcosm_one_CSS_3_Rhizosphere_7_2_Nonanone_cont= subset_samples(Fungi_microcosm_one_CSS_3_Rhizosphere, Treatment %in% c("7_2-Nonanone","1_Control"));Fungi_microcosm_one_CSS_3_Rhizosphere_7_2_Nonanone_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_3_Rhizosphere_7_2_Nonanone_cont <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_3_Rhizosphere_7_2_Nonanone_cont)))
Fungi_microcosm_one_CSS_3_Rhizosphere_7_2_Nonanone_cont <- data.frame(sample_data(Fungi_microcosm_one_CSS_3_Rhizosphere_7_2_Nonanone_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_7_2_Nonanone_cont <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_3_Rhizosphere_7_2_Nonanone_cont, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_7_2_Nonanone_cont
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_3_Rhizosphere_7_2_Nonanone_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_7_2_Nonanone_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere_7_2_Nonanone_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)
#Treatment  1  0.21425 0.12827 1.4715  0.101
#Residual  10  1.45602 0.87173              
#Total     11  1.67027 1.00000

##########
#8_Volatile_Mixture
##########
Fungi_microcosm_one_CSS_3_Rhizosphere8_Volatile_Mixture_cont= subset_samples(Fungi_microcosm_one_CSS_3_Rhizosphere, Treatment %in% c("8_Volatile_Mixture","1_Control"));Fungi_microcosm_one_CSS_3_Rhizosphere8_Volatile_Mixture_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_3_Rhizosphere8_Volatile_Mixture_cont <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_3_Rhizosphere8_Volatile_Mixture_cont)))
Fungi_microcosm_one_CSS_3_Rhizosphere8_Volatile_Mixture_cont <- data.frame(sample_data(Fungi_microcosm_one_CSS_3_Rhizosphere8_Volatile_Mixture_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere8_Volatile_Mixture_cont <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_3_Rhizosphere8_Volatile_Mixture_cont, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere8_Volatile_Mixture_cont
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_3_Rhizosphere8_Volatile_Mixture_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere8_Volatile_Mixture_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_3_Rhizosphere8_Volatile_Mixture_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)
#Treatment  1  0.16498 0.10053 1.1177   0.26
#Residual  10  1.47605 0.89947              
#Total     11  1.64102 1.00000 

#####################################################################################################

#Subset the different compartments
#Soil
Fungi_microcosm_one_CSS_Soil= subset_samples(Fungi_microcosm_one_CSS, Sample_type %in% c("2_Soil"));Fungi_microcosm_one_CSS_Soil#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_Soil <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_Soil)))
sampledf_Fungi_microcosm_one_CSS_Soil <- data.frame(sample_data(Fungi_microcosm_one_CSS_Soil))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_Soil <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_Soil, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_Soil
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_Soil), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_Soil)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_Soil ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)    
#Treatment  7   1.2740 0.31369 2.6118  0.001 ***
#Residual  40   2.7874 0.68631                  
#Total     47   4.0614 1.00000                  
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
############################################################
############################################################
#Fungi vs control in the Soil
Fungi_microcosm_one_CSS_Soil_Bact_control= subset_samples(Fungi_microcosm_one_CSS_Soil, Pool %in% c("Bacteria","Control"));Fungi_microcosm_one_CSS_Soil_Bact_control#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_Soil_Bact_control <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_Soil_Bact_control)))
sampledf_Fungi_microcosm_one_CSS_Soil_Bact_control <- data.frame(sample_data(Fungi_microcosm_one_CSS_Soil_Bact_control))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_Soil_Bact_control <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_Soil_Bact_control, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_Soil_Bact_control
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_Soil_Bact_control), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_Soil_Bact_control)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_Soil_Bact_control ~ Pool, data=sampleda, permutation=999);adores#change from adonis to adonis2
#         Df SumOfSqs      R2      F Pr(>F)    
#Pool      1  0.28294 0.12993 3.2853  0.001 ***
#Residual 22  1.89470 0.87007                  
#Total    23  2.17765 1.00000                  
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
#Volatile vs control in the Soil
Fungi_microcosm_one_CSS_Soil_vola_control= subset_samples(Fungi_microcosm_one_CSS_Soil, Pool %in% c("Volatiles","Control"));Fungi_microcosm_one_CSS_Soil_vola_control#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_Soil_vola_control <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_Soil_vola_control)))
sampledf_Fungi_microcosm_one_CSS_Soil_vola_control <- data.frame(sample_data(Fungi_microcosm_one_CSS_Soil_vola_control))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_Soil_vola_control <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_Soil_vola_control, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_Soil_vola_control
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_Soil_vola_control), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_Soil_vola_control)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_Soil_vola_control ~ Pool, data=sampleda, permutation=999);adores#change from adonis to adonis2
#         Df SumOfSqs      R2      F Pr(>F)    
#Pool      1  0.43397 0.17382 5.8909  0.001 ***
#Residual 28  2.06268 0.82618                  
#Total    29  2.49665 1.00000                  
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1             
############################################################
############################################################
############################################################
############################################################# 
#S. plymuthica HRO C48  vs control_Soil
##########
Fungi_microcosm_one_CSS_Soil_ser_cont= subset_samples(Fungi_microcosm_one_CSS_Soil, Treatment %in% c("2_Serratia_plymuthica_HRO_C48","1_Control"));Fungi_microcosm_one_CSS_Soil_ser_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_Soil_ser_cont <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_Soil_ser_cont)))
Fungi_microcosm_one_CSS_Soil_ser_cont <- data.frame(sample_data(Fungi_microcosm_one_CSS_Soil_ser_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_Soil_ser_cont <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_Soil_ser_cont, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_Soil_ser_cont
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_Soil_ser_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_Soil_ser_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_Soil_ser_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.40295 0.34135 5.1825  0.004 **
#Residual  10  0.77753 0.65865                 
#Total     11  1.18048 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

##########
# 3_Stenotrophomonas_rhizophilla_SPA_P69  vs control_Soil
##########
Fungi_microcosm_one_CSS_Soil_Steno_cont= subset_samples(Fungi_microcosm_one_CSS_Soil, Treatment %in% c("3_Stenotrophomonas_rhizophilla_SPA_P69","1_Control"));Fungi_microcosm_one_CSS_Soil_Steno_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_Soil_Steno_cont <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_Soil_Steno_cont)))
Fungi_microcosm_one_CSS_Soil_Steno_cont <- data.frame(sample_data(Fungi_microcosm_one_CSS_Soil_Steno_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_Soil_Steno_cont <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_Soil_Steno_cont, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_Soil_Steno_cont
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_Soil_Steno_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_Soil_Steno_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_Soil_Steno_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.22971 0.21967 2.8151  0.003 **
#Residual  10  0.81601 0.78033                 
#Total     11  1.04572 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

##########
# 4_Fungi_Mixture  vs control_Soil
##########
Fungi_microcosm_one_CSS_Soil_consortium_cont= subset_samples(Fungi_microcosm_one_CSS_Soil, Treatment %in% c("4_Bacteria_Mixture","1_Control"));Fungi_microcosm_one_CSS_Soil_consortium_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_Soil_consortium_cont <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_Soil_consortium_cont)))
Fungi_microcosm_one_CSS_Soil_consortium_cont <- data.frame(sample_data(Fungi_microcosm_one_CSS_Soil_consortium_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_Soil_consortium_cont <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_Soil_consortium_cont, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_Soil_consortium_cont
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_Soil_consortium_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_Soil_consortium_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_Soil_consortium_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)  
#Treatment  1  0.12512 0.13053 1.5012  0.029 *
#Residual  10  0.83348 0.86947                
#Total     11  0.95861 1.00000                
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

##########
# 5_3-Methyl-1-Butanol  vs control_Soil
##########
Fungi_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont= subset_samples(Fungi_microcosm_one_CSS_Soil, Treatment %in% c("5_3-Methyl-1-Butanol","1_Control"));Fungi_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont)))
Fungi_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont <- data.frame(sample_data(Fungi_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_Soil_5_3_Methyl_1_Butanol_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.34826 0.30678 4.4255  0.007 **
#Residual  10  0.78693 0.69322                 
#Total     11  1.13519 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

##########
# 6_2-Butanone  vs control_Soil
##########
Fungi_microcosm_one_CSS_Soil_6_2_Butanone_cont= subset_samples(Fungi_microcosm_one_CSS_Soil, Treatment %in% c("6_2-Butanone","1_Control"));Fungi_microcosm_one_CSS_Soil_6_2_Butanone_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_Soil_6_2_Butanone_cont <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_Soil_6_2_Butanone_cont)))
Fungi_microcosm_one_CSS_Soil_6_2_Butanone_cont <- data.frame(sample_data(Fungi_microcosm_one_CSS_Soil_6_2_Butanone_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_Soil_6_2_Butanone_cont <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_Soil_6_2_Butanone_cont, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_Soil_6_2_Butanone_cont
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_Soil_6_2_Butanone_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_Soil_6_2_Butanone_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_Soil_6_2_Butanone_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.39254 0.34768 5.3298  0.003 **
#Residual  10  0.73651 0.65232                 
#Total     11  1.12905 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1           
##########
#7_2-Nonanone
##########
Fungi_microcosm_one_CSS_Soil_7_2_Nonanone_cont= subset_samples(Fungi_microcosm_one_CSS_Soil, Treatment %in% c("7_2-Nonanone","1_Control"));Fungi_microcosm_one_CSS_Soil_7_2_Nonanone_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_Soil_7_2_Nonanone_cont <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_Soil_7_2_Nonanone_cont)))
Fungi_microcosm_one_CSS_Soil_7_2_Nonanone_cont <- data.frame(sample_data(Fungi_microcosm_one_CSS_Soil_7_2_Nonanone_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_Soil_7_2_Nonanone_cont <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_Soil_7_2_Nonanone_cont, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_Soil_7_2_Nonanone_cont
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_Soil_7_2_Nonanone_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_Soil_7_2_Nonanone_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_Soil_7_2_Nonanone_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.22619 0.23503 3.0724  0.002 **
#Residual  10  0.73622 0.76497                 
#Total     11  0.96241 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

##########
#8_Volatile_Mixture
##########
Fungi_microcosm_one_CSS_Soil8_Volatile_Mixture_cont= subset_samples(Fungi_microcosm_one_CSS_Soil, Treatment %in% c("8_Volatile_Mixture","1_Control"));Fungi_microcosm_one_CSS_Soil8_Volatile_Mixture_cont#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_microcosm_one_CSS_Soil8_Volatile_Mixture_cont <- t(data.frame(otu_table(Fungi_microcosm_one_CSS_Soil8_Volatile_Mixture_cont)))
Fungi_microcosm_one_CSS_Soil8_Volatile_Mixture_cont <- data.frame(sample_data(Fungi_microcosm_one_CSS_Soil8_Volatile_Mixture_cont))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_microcosm_one_CSS_Soil8_Volatile_Mixture_cont <- vegdist(dist.matrix_Fungi_microcosm_one_CSS_Soil8_Volatile_Mixture_cont, method = "bray"); phyloseq_bray_Fungi_microcosm_one_CSS_Soil8_Volatile_Mixture_cont
sampleda <- data.frame(sample_data(Fungi_microcosm_one_CSS_Soil8_Volatile_Mixture_cont), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_microcosm_one_CSS_Soil8_Volatile_Mixture_cont)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_microcosm_one_CSS_Soil8_Volatile_Mixture_cont ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.28199 0.24932 3.3212  0.003 **
#Residual  10  0.84907 0.75068                 
#Total     11  1.13106 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1           
#####################################################################################################
#Significance table
colors=c("gray33");colors
stacked_bar= read.table ("Significance.txt", check.names = FALSE, header = TRUE, dec = ".", sep = "\t");stacked_bar
library(reshape2)
#use melt() to convert data frame from wide to long format
long_df <- melt(stacked_bar, id='group');long_df
write.csv(long_df,"long_df.csv")
percentage= read.table ("Percentage.txt", check.names = FALSE, header = TRUE, dec = ".", sep = "\t");percentage

#Filter_bacteria

percentage_bacteria=filter(percentage,Organism %in% c("Bacteria" ));percentage_bacteria

tiff("stacked_sign1.tiff", units="in", width=16, height=6, res=300)
library(ggplot2)
stacked_sign1.tiff=ggplot(percentage, aes(x = group, y = Percentage, fill = category)) + 
  geom_bar(stat = "identity") + scale_fill_manual(values=colors)+ theme(axis.text.x = element_text(angle = 90))+  theme(text=element_text(family="sans"))+
  theme(axis.title.x = element_text(face="bold", colour="black", size=12),
        axis.title.y = element_text(face="bold", colour="black", size=12),
        axis.text.x = element_text( colour="black", size=12),
        axis.text.y  = element_text( colour="black", size=12))++theme(text=element_text(size=12,  family="sans"))+ labs(tag = "E")+theme(legend.position = "top");stacked_sign1.tiff
#stacked_sign1.tiff=stacked_plot3+plot_layout(ncol = 1);stacked_sign.tiff
ggsave("stacked_sign1.tiff", plot = stacked_sign1.tiff)
dev.off()

library(ggplot2)
stacked_plot1=ggplot(percentage_bacteria, aes(x = group, y = Percentage, fill = category)) + 
  geom_bar(stat = "identity") + scale_fill_manual(values=colors)+ theme(axis.text.x = element_text(angle = 90))+  theme(text=element_text(family="sans"))+ 
  theme(axis.title.x = element_text(face="bold", colour="black", size=12),
        axis.title.y = element_text(face="bold", colour="black", size=12),
        axis.text.x = element_text( colour="black", size=12),
        axis.text.y  = element_text( colour="black", size=12))+theme(text=element_text(size=12,  family="sans"))+ labs(tag = "A")+theme(legend.position = "bottom");stacked_plot1

#Filter_Fungi
percentage_Fungi=filter(percentage,Organism %in% c("Fungi" ));percentage_Fungi

library(ggplot2)
stacked_plot2=ggplot(percentage_Fungi, aes(x = group, y = Percentage, fill = category)) + 
  geom_bar(stat = "identity") + scale_fill_manual(values=colors)+ theme(axis.text.x = element_text(angle = 90))+  theme(text=element_text(family="sans"))+
  theme(axis.title.x = element_text(face="bold", colour="black", size=12),
        axis.title.y = element_text(face="bold", colour="black", size=12),
        axis.text.x = element_text( colour="black", size=12),
        axis.text.y  = element_text( colour="black", size=12))+theme(text=element_text(size=12,  family="sans"))+ labs(tag = "B")+theme(legend.position = "bottom");stacked_plot2

tiff("stacked_sign.tiff", units="in", width=15, height=8, res=300)
stacked_sign.tiff=stacked_plot1+stacked_plot2+plot_layout(ncol = 2);stacked_sign.tiff
ggsave("stacked_sign.tiff", plot = stacked_sign.tiff)
dev.off()
#####################################################################################################
Fungi_microcosm_one<- subset_samples(Fungi_mereged, Experiment %in% c("One"));Fungi_microcosm_one#[ 15421 taxa and 1102 samples ]
Fungi_microcosm_one<- subset_samples(Fungi_microcosm_one, Sample_type %in% c("4_Phyllosphere","2_Soil","3_Rhizosphere"));Fungi_microcosm_one#[ 15421 taxa and 1102 samples ]
Fungi_microcosm_one_Soil= subset_samples(Fungi_microcosm_one, Sample_type %in% c("2_Soil"));Fungi_microcosm_one_Soil#[ 15421 taxa and 1102 samples ]


#serratia vs control
Fungi_microcosm_one_Soil_ser_cont= subset_samples(Fungi_microcosm_one_Soil, Treatment %in% c("2_Serratia_plymuthica_HRO_C48","1_Control"));Fungi_microcosm_one_Soil_ser_cont#[ 15421 taxa and 1102 samples ]
diagdds_ser = phyloseq_to_deseq2(Fungi_microcosm_one_Soil_ser_cont, ~ Treatment);diagdds_ser
diagdds_ser = DESeq(diagdds_ser, test="Wald", fitType="parametric");diagdds_ser
res = results(diagdds_ser, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_ser = res[which(res$padj < alpha), ];sigtab_ser
sigtab_ser = cbind(as(sigtab_ser, "data.frame"), as(tax_table(Fungi_microcosm_one_Soil_ser_cont)[rownames(sigtab_ser), ], "matrix"));sigtab_ser
head(sigtab_ser)
write.csv(sigtab_ser, "sigtab_ser.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_ser$log2FoldChange, sigtab_ser$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser$Phylum = factor(as.character(sigtab_ser$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_ser$log2FoldChange, sigtab_ser$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser$Genus = factor(as.character(sigtab_ser$Genus), levels=names(x))
ggplot(sigtab_ser, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Agaricomycetes,Dothideomycetes, Endogomycetes, Leotiomycetes, Mortierellomycetes, and Dordariomycetes
#####################################################################################################
#steno vs control
Fungi_microcosm_one_Soil_steno_cont= subset_samples(Fungi_microcosm_one_Soil, Treatment %in% c("3_Stenotrophomonas_rhizophilla_SPA_P69","1_Control"));Fungi_microcosm_one_Soil_steno_cont#[ 15421 taxa and 1102 samples ]
diagdds_Steno = phyloseq_to_deseq2(Fungi_microcosm_one_Soil_steno_cont, ~ Treatment);diagdds_Steno
diagdds_Steno = DESeq(diagdds_Steno, test="Wald", fitType="parametric");diagdds_Steno
res = results(diagdds_Steno, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_Steno = res[which(res$padj < alpha), ];sigtab_Steno
sigtab_Steno = cbind(as(sigtab_Steno, "data.frame"), as(tax_table(Fungi_microcosm_one_Soil_steno_cont)[rownames(sigtab_Steno), ], "matrix"));sigtab_Steno
head(sigtab_Steno)
write.csv(sigtab_Steno, "sigtab_Steno.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_Steno$log2FoldChange, sigtab_Steno$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_Steno$Phylum = factor(as.character(sigtab_Steno$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_Steno$log2FoldChange, sigtab_Steno$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_Steno$Genus = factor(as.character(sigtab_Steno$Genus), levels=names(x))
ggplot(sigtab_Steno, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
##Agaricomycetes,Dothideomycetes, Endogomycetes, Leotiomycetes, Mortierellomycetes, and Sordariomycetes
#####################################################################################################
#Consort vs control
Fungi_microcosm_one_Soil_consortium_cont= subset_samples(Fungi_microcosm_one_Soil, Treatment %in% c("4_Bacteria_Mixture","1_Control"));Fungi_microcosm_one_Soil_consortium_cont#[ 15421 taxa and 1102 samples ]
diagdds_Consortium = phyloseq_to_deseq2(Fungi_microcosm_one_Soil_consortium_cont, ~ Treatment);diagdds_Consortium
diagdds_Consortium = DESeq(diagdds_Consortium, test="Wald", fitType="parametric");diagdds_Consortium
res = results(diagdds_Consortium, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_Consortium = res[which(res$padj < alpha), ];sigtab_Consortium
sigtab_Consortium = cbind(as(sigtab_Consortium, "data.frame"), as(tax_table(Fungi_microcosm_one_Soil_consortium_cont)[rownames(sigtab_Consortium), ], "matrix"));sigtab_Consortium
head(sigtab_Consortium)
write.csv(sigtab_Consortium, "sigtab_Consortium.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_Consortium$log2FoldChange, sigtab_Consortium$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_Consortium$Phylum = factor(as.character(sigtab_Consortium$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_Consortium$log2FoldChange, sigtab_Consortium$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_Consortium$Genus = factor(as.character(sigtab_Consortium$Genus), levels=names(x))
ggplot(sigtab_Consortium, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))

##Agaricomycetes,Dothideomycetes, Endogomycetes, Leotiomycetes, Mortierellomycetes, and Sordariomycetes
#####################################################################################################
#Consort vs control
Fungi_microcosm_one_Soil_Methyl_cont= subset_samples(Fungi_microcosm_one_Soil, Treatment %in% c("5_3-Methyl-1-Butanol","1_Control"));Fungi_microcosm_one_Soil_Methyl_cont#[ 15421 taxa and 1102 samples ]
diagdds_Methyl = phyloseq_to_deseq2(Fungi_microcosm_one_Soil_Methyl_cont, ~ Treatment);diagdds_Methyl
diagdds_Methyl = DESeq(diagdds_Methyl, test="Wald", fitType="parametric");diagdds_Methyl
res = results(diagdds_Methyl, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_Methyl = res[which(res$padj < alpha), ];sigtab_Methyl
sigtab_Methyl = cbind(as(sigtab_Methyl, "data.frame"), as(tax_table(Fungi_microcosm_one_Soil_Methyl_cont)[rownames(sigtab_Methyl), ], "matrix"));sigtab_Methyl
head(sigtab_Methyl)
write.csv(sigtab_Methyl, "sigtab_Methyl.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_Methyl$log2FoldChange, sigtab_Methyl$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_Methyl$Phylum = factor(as.character(sigtab_Methyl$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_Methyl$log2FoldChange, sigtab_Methyl$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_Methyl$Genus = factor(as.character(sigtab_Methyl$Genus), levels=names(x))
ggplot(sigtab_Methyl, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
###Agaricomycetes,Dothideomycetes, Endogomycetes, Leotiomycetes, Mortierellomycetes, and Sordariomycetes
#####################################################################################################
#Butanone
#Consort vs control
Fungi_microcosm_one_Soil_Butanone_cont= subset_samples(Fungi_microcosm_one_Soil, Treatment %in% c("6_2-Butanone","1_Control"));Fungi_microcosm_one_Soil_Butanone_cont#[ 15421 taxa and 1102 samples ]
diagdds_Butanone = phyloseq_to_deseq2(Fungi_microcosm_one_Soil_Butanone_cont, ~ Treatment);diagdds_Butanone
diagdds_Butanone = DESeq(diagdds_Butanone, test="Wald", fitType="parametric");diagdds_Butanone
res = results(diagdds_Butanone, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_Butanone = res[which(res$padj < alpha), ];sigtab_Butanone
sigtab_Butanone = cbind(as(sigtab_Butanone, "data.frame"), as(tax_table(Fungi_microcosm_one_Soil_Butanone_cont)[rownames(sigtab_Butanone), ], "matrix"));sigtab_Butanone
head(sigtab_Butanone)
write.csv(sigtab_Butanone, "sigtab_Butanone.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_Butanone$log2FoldChange, sigtab_Butanone$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_Butanone$Phylum = factor(as.character(sigtab_Butanone$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_Butanone$log2FoldChange, sigtab_Butanone$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_Butanone$Genus = factor(as.character(sigtab_Butanone$Genus), levels=names(x))
ggplot(sigtab_Butanone, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
###Agaricomycetes,Dothideomycetes, Endogomycetes, Eurotiomycetes,Leotiomycetes, Mortierellomycetes, and Sordariomycetes

#####################################################################################################
#7_2-Nonanone vs control
Fungi_microcosm_one_Soil_7_2_Nonanone_cont= subset_samples(Fungi_microcosm_one_Soil, Treatment %in% c("7_2-Nonanone","1_Control"));Fungi_microcosm_one_Soil_7_2_Nonanone_cont#[ 15421 taxa and 1102 samples ]
diagdds_Nonanone = phyloseq_to_deseq2(Fungi_microcosm_one_Soil_7_2_Nonanone_cont, ~ Treatment);diagdds_Nonanone
diagdds_Nonanone = DESeq(diagdds_Nonanone, test="Wald", fitType="parametric");diagdds_Nonanone
res = results(diagdds_Nonanone, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_Nonanone = res[which(res$padj < alpha), ];sigtab_Nonanone
sigtab_Nonanone = cbind(as(sigtab_Nonanone, "data.frame"), as(tax_table(Fungi_microcosm_one_Soil_7_2_Nonanone_cont)[rownames(sigtab_Nonanone), ], "matrix"));sigtab_Nonanone
head(sigtab_Nonanone)
write.csv(sigtab_Nonanone, "sigtab_Nonanone.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_Nonanone$log2FoldChange, sigtab_Nonanone$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_Nonanone$Phylum = factor(as.character(sigtab_Nonanone$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_Nonanone$log2FoldChange, sigtab_Nonanone$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_Nonanone$Genus = factor(as.character(sigtab_Nonanone$Genus), levels=names(x))
ggplot(sigtab_Nonanone, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
###Agaricomycetes,Dothideomycetes, Endogomycetes, Leotiomycetes, and Sordariomycetes
#####################################################################################################
#8_Volatile_Mixture
Fungi_microcosm_one_Soil8_Volatile_Mixture_cont= subset_samples(Fungi_microcosm_one_Soil, Treatment %in% c("8_Volatile_Mixture","1_Control"));Fungi_microcosm_one_Soil8_Volatile_Mixture_cont#[ 15421 taxa and 1102 samples ]
diagdds_Volatile_mixture = phyloseq_to_deseq2(Fungi_microcosm_one_Soil8_Volatile_Mixture_cont, ~ Treatment);diagdds_Volatile_mixture
diagdds_Volatile_mixture = DESeq(diagdds_Volatile_mixture, test="Wald", fitType="parametric");diagdds_Volatile_mixture
res = results(diagdds_Volatile_mixture, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_Volatile_mixture = res[which(res$padj < alpha), ];sigtab_Volatile_mixture
sigtab_Volatile_mixture = cbind(as(sigtab_Volatile_mixture, "data.frame"), as(tax_table(Fungi_microcosm_one_Soil8_Volatile_Mixture_cont)[rownames(sigtab_Volatile_mixture), ], "matrix"));sigtab_Volatile_mixture
head(sigtab_Volatile_mixture)
write.csv(sigtab_Volatile_mixture, "sigtab_Volatile_mixture.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_Volatile_mixture$log2FoldChange, sigtab_Volatile_mixture$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_Volatile_mixture$Phylum = factor(as.character(sigtab_Volatile_mixture$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_Volatile_mixture$log2FoldChange, sigtab_Volatile_mixture$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_Volatile_mixture$Genus = factor(as.character(sigtab_Volatile_mixture$Genus), levels=names(x))
ggplot(sigtab_Volatile_mixture, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
###Dothideomycetes, Endogomycetes, Leotiomycetes, and Sordariomycetes
#####################################################################################################
#Consider the potting soil bacteria
Bacteria_experiment_soils= subset_samples(Bacteria_mereged, Experiment %in% c("One_one","Two_env","Two_pot"));Bacteria_experiment_soils#[ 15421 taxa and 1102 samples ]
#####################################################################################################
library(ggplot2)
library(MicrobiotaProcess)
citation("MicrobiotaProcess")
BFamtaxa_experiment_soils <- get_taxadf(obj=Bacteria_experiment_soils, taxlevel=5);BFamtaxa_experiment_soils
#####################################################################################################
tiff("fgen_soils_Bacteria.tiff", units="in", width=14, height=7, res=300)
fgen_soils_Bacteria <- ggbartax(obj=BFamtaxa_experiment_soils, facetNames="Soil_type_Treatment_Sample", plotgroup=TRUE, topn=20) +
  xlab("") +
  ylab("relative abundance (%)") +scale_fill_manual(values=mycolors)+theme(axis.text.x = element_text(angle = 270)) +
  #scale_color_manual(values=c("chartreuse4","bisque4","deepskyblue","Sky Blue","goldenrod", "#ba0d42", "grey85","Orange","coral3","darkred","#0dba2f","#850dba","grey35","Blue","deeppink","blue","Yellow")) + 
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=1))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=0.4))+ 
  theme(axis.title.x = element_text(face="bold", colour="black", size=14),
        axis.title.y = element_text(face="bold", colour="black", size=14),
        axis.text.x = element_text(colour="black", size=11),
        axis.text.y  = element_text(colour="black", size=11))+ labs(tag = "B")+
  theme(legend.position = "right")+ 
  labs(fill='Phyla')+
  theme(legend.title = element_text(color = "black", size = 12),
        legend.text = element_text(color = "black", size = 12))+
  theme(text=element_text(family="sans"));fgen_soils_Bacteria
dev.off()

#####################################################################################################
#Consider the potting soil fungi
Fungi_experiment_soils= subset_samples(Fungi_mereged, Experiment %in% c("One_one","Two_env","Two_pot"));Fungi_experiment_soils#[ 15421 taxa and 1102 samples ]
#####################################################################################################
library(ggplot2)
library(MicrobiotaProcess)
fgentaxa_experiment_soils <- get_taxadf(obj=Fungi_experiment_soils, taxlevel=5);fgentaxa_experiment_soils
#####################################################################################################
#https://yulab-smu.top/MicrobiotaProcessWorkshop/articles/MicrobiotaProcessWorkshop.html
tiff("fgen_soils_fungi.tiff", units="in", width=14, height=7, res=300)
fgen_soils_fungi <- ggbartax(obj=fgentaxa_experiment_soils, facetNames="Experiment", plotgroup=TRUE, topn=20) +
  xlab("") +
  ylab("relative abundance (%)") +scale_fill_manual(values=mycolors)+theme(axis.text.x = element_text(angle = 270)) +
  #scale_color_manual(values=c("chartreuse4","bisque4","deepskyblue","Sky Blue","goldenrod", "#ba0d42", "grey85","Orange","coral3","darkred","#0dba2f","#850dba","grey35","Blue","deeppink","blue","Yellow")) + 
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=1))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=0.4))+ 
  theme(axis.title.x = element_text(face="bold", colour="black", size=14),
        axis.title.y = element_text(face="bold", colour="black", size=14),
        axis.text.x = element_text(colour="black", size=11),
        axis.text.y  = element_text(colour="black", size=11))+ labs(tag = "D")+
  theme(legend.position = "right")+ 
  labs(fill='Phyla')+
  theme(legend.title = element_text(color = "black", size = 12),
        legend.text = element_text(color = "black", size = 12))+
  theme(text=element_text(family="sans"));fgen_soils_fungi
#####################################################################################################
#Figure S7 SOILS
tiff("FigureS7.tiff", units="in", width=9, height=12, res=300)
FigureS7.tiff=fgen_soils_Bacteria+fgen_soils_fungi+plot_layout(ncol = 1);FigureS7.tiff
ggsave("FigureS7.tiff", plot = FigureS7.tiff)
#####################################################################################################
```

#consider experiment2

```{r,echo=FALSE,warning=FALSE}
setwd("/Users/expeditoolimi/Documents/Manuscript-AA-OE_SB/Microbiome_analysis")
load("/Users/expeditoolimi/Documents/Manuscript-AA-OE_SB/Microbiome_analysis/Bacteria_mereged.RData")

Bacteria_experiment_two= subset_samples(Bacteria_mereged, Experiment %in% c("Two"));Bacteria_experiment_two#[ 15421 taxa and 1102 samples ]
Bacteria_experiment_two= subset_samples(Bacteria_experiment_two, Treatment %in% c("2_Serratia_plymuthica_HRO_C48","1_Control"));Bacteria_experiment_two#[ 15421 taxa and 1102 samples ]
#####################################################################################################
Bacteria_experiment_two_table = as(otu_table(Bacteria_experiment_two), "matrix")
# Coerce to data.frame
Bacteria_experiment_two_table = as.data.frame(Bacteria_experiment_two_table)
write.csv(Bacteria_experiment_two_table ,file = "Bacteria_experiment_two_table.csv")
#####################################################################################################

#Environmetal soil
Bacteria_experiment_two_Env= subset_samples(Bacteria_experiment_two, Soil_type %in% c("Environmental_soil"));Bacteria_experiment_two_Env#[ 15421 taxa and 1102 samples ]

#phyllosphere
Bacteria_experiment_two_Phyllosphere_env= subset_samples(Bacteria_experiment_two_Env, Sample_type %in% c("4_Phyllosphere"));Bacteria_experiment_two_Phyllosphere_env#[ 15421 taxa and 1102 samples ]
diagdds_ser_control_phyllosphere_env = phyloseq_to_deseq2(Bacteria_experiment_two_Phyllosphere_env, ~ Treatment);diagdds_ser_control_phyllosphere_env
diagdds_ser_control_phyllosphere_env = DESeq(diagdds_ser_control_phyllosphere_env, test="Wald", fitType="parametric");diagdds_ser_control_phyllosphere_env
res = results(diagdds_ser_control_phyllosphere_env, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_ser_control_phyllosphere_env = res[which(res$padj < alpha), ];sigtab_ser_control_phyllosphere_env
sigtab_ser_control_phyllosphere_env = cbind(as(sigtab_ser_control_phyllosphere_env, "data.frame"), as(tax_table(Bacteria_experiment_two_Phyllosphere_env)[rownames(sigtab_ser_control_phyllosphere_env), ], "matrix"));sigtab_ser_control_phyllosphere_env
head(sigtab_ser_control_phyllosphere_env)
write.csv(sigtab_ser_control_phyllosphere_env, "sigtab_ser_control_phyllosphere_env.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_ser_control_phyllosphere_env$log2FoldChange, sigtab_ser_control_phyllosphere_env$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_phyllosphere_env$Phylum = factor(as.character(sigtab_ser_control_phyllosphere_env$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_ser_control_phyllosphere_env$log2FoldChange, sigtab_ser_control_phyllosphere_env$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_phyllosphere_env$Genus = factor(as.character(sigtab_ser_control_phyllosphere_env$Genus), levels=names(x))
ggplot(sigtab_ser_control_phyllosphere_env, aes(x=Genus, y=log2FoldChange, color=Family)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Beijerinckiaceae,Burkholderiaceae,Flavobacteriaceae, Methylophilaceae, Pseudomonadaceae, Rhizobiaceae, Sphingomonadaceae, Spirosomaceae
#####################################################################################################
Bacteria_experiment_two_3_Rhizosphere_env= subset_samples(Bacteria_experiment_two_Env, Sample_type %in% c("3_Rhizosphere "));Bacteria_experiment_two_3_Rhizosphere_env#[ 15421 taxa and 1102 samples ]
diagdds_ser_control_3_Rhizosphere_env = phyloseq_to_deseq2(Bacteria_experiment_two_3_Rhizosphere_env, ~ Treatment);diagdds_ser_control_3_Rhizosphere_env
diagdds_ser_control_3_Rhizosphere_env = DESeq(diagdds_ser_control_3_Rhizosphere_env, test="Wald", fitType="parametric");diagdds_ser_control_3_Rhizosphere_env
res = results(diagdds_ser_control_3_Rhizosphere_env, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_ser_control_3_Rhizosphere_env = res[which(res$padj < alpha), ];sigtab_ser_control_3_Rhizosphere_env
sigtab_ser_control_3_Rhizosphere_env = cbind(as(sigtab_ser_control_3_Rhizosphere_env, "data.frame"), as(tax_table(Bacteria_experiment_two_3_Rhizosphere_env)[rownames(sigtab_ser_control_3_Rhizosphere_env), ], "matrix"));sigtab_ser_control_3_Rhizosphere_env
head(sigtab_ser_control_3_Rhizosphere_env)
write.csv(sigtab_ser_control_3_Rhizosphere_env, "sigtab_ser_control_3_Rhizosphere_env.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_ser_control_3_Rhizosphere_env$log2FoldChange, sigtab_ser_control_3_Rhizosphere_env$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_3_Rhizosphere_env$Phylum = factor(as.character(sigtab_ser_control_3_Rhizosphere_env$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_ser_control_3_Rhizosphere_env$log2FoldChange, sigtab_ser_control_3_Rhizosphere_env$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_3_Rhizosphere_env$Genus = factor(as.character(sigtab_ser_control_3_Rhizosphere_env$Genus), levels=names(x))
ggplot(sigtab_ser_control_3_Rhizosphere_env, aes(x=Genus, y=log2FoldChange, color=Family)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#####################################################################################################
Bacteria_experiment_two_Soil_env= subset_samples(Bacteria_experiment_two_Env, Sample_type %in% c("2_Soil"));Bacteria_experiment_two_Soil_env#[ 15421 taxa and 1102 samples ]
diagdds_ser_control_Soil_env = phyloseq_to_deseq2(Bacteria_experiment_two_Soil_env, ~ Treatment);diagdds_ser_control_Soil_env
diagdds_ser_control_Soil_env = DESeq(diagdds_ser_control_Soil_env, test="Wald", fitType="parametric");diagdds_ser_control_Soil_env
res = results(diagdds_ser_control_Soil_env, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_ser_control_Soil_env = res[which(res$padj < alpha), ];sigtab_ser_control_Soil_env
sigtab_ser_control_Soil_env = cbind(as(sigtab_ser_control_Soil_env, "data.frame"), as(tax_table(Bacteria_experiment_two_Soil_env)[rownames(sigtab_ser_control_Soil_env), ], "matrix"));sigtab_ser_control_Soil_env
head(sigtab_ser_control_Soil_env)
write.csv(sigtab_ser_control_Soil_env, "sigtab_ser_control_Soil_env.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_ser_control_Soil_env$log2FoldChange, sigtab_ser_control_Soil_env$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_Soil_env$Phylum = factor(as.character(sigtab_ser_control_Soil_env$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_ser_control_Soil_env$log2FoldChange, sigtab_ser_control_Soil_env$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_Soil_env$Genus = factor(as.character(sigtab_ser_control_Soil_env$Genus), levels=names(x))
ggplot(sigtab_ser_control_Soil_env, aes(x=Genus, y=log2FoldChange, color=Family)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Haliangiaceae, Xanthomonadaceae
#####################################################################################################
#Pottironmetal soil
Bacteria_experiment_two_Pott= subset_samples(Bacteria_experiment_two, Soil_type %in% c("Potting_soil"));Bacteria_experiment_two_Pott#[ 15421 taxa and 1102 samples ]

#phyllosphere
Bacteria_experiment_two_Phyllosphere_Pott= subset_samples(Bacteria_experiment_two_Pott, Sample_type %in% c("4_Phyllosphere"));Bacteria_experiment_two_Phyllosphere_Pott#[ 15421 taxa and 1102 samples ]
diagdds_ser_control_phyllosphere_Pott = phyloseq_to_deseq2(Bacteria_experiment_two_Phyllosphere_Pott, ~ Treatment);diagdds_ser_control_phyllosphere_Pott
diagdds_ser_control_phyllosphere_Pott = DESeq(diagdds_ser_control_phyllosphere_Pott, test="Wald", fitType="parametric");diagdds_ser_control_phyllosphere_Pott
res = results(diagdds_ser_control_phyllosphere_Pott, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_ser_control_phyllosphere_Pott = res[which(res$padj < alpha), ];sigtab_ser_control_phyllosphere_Pott
sigtab_ser_control_phyllosphere_Pott = cbind(as(sigtab_ser_control_phyllosphere_Pott, "data.frame"), as(tax_table(Bacteria_experiment_two_Phyllosphere_Pott)[rownames(sigtab_ser_control_phyllosphere_Pott), ], "matrix"));sigtab_ser_control_phyllosphere_Pott
head(sigtab_ser_control_phyllosphere_Pott)
write.csv(sigtab_ser_control_phyllosphere_Pott, "sigtab_ser_control_phyllosphere_Pott.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_ser_control_phyllosphere_Pott$log2FoldChange, sigtab_ser_control_phyllosphere_Pott$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_phyllosphere_Pott$Phylum = factor(as.character(sigtab_ser_control_phyllosphere_Pott$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_ser_control_phyllosphere_Pott$log2FoldChange, sigtab_ser_control_phyllosphere_Pott$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_phyllosphere_Pott$Genus = factor(as.character(sigtab_ser_control_phyllosphere_Pott$Genus), levels=names(x))
ggplot(sigtab_ser_control_phyllosphere_Pott, aes(x=Genus, y=log2FoldChange, color=Family)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Bacillaceae, Beijerinckiaceae,Burkholderiaceae,Chitinophagaceae, Nocardiaceae,Pseudomonadaceae,Solimonadaceae, Spirosomaceae
#####################################################################################################
Bacteria_experiment_two_3_Rhizosphere_Pott= subset_samples(Bacteria_experiment_two_Pott, Sample_type %in% c("3_Rhizosphere "));Bacteria_experiment_two_3_Rhizosphere_Pott#[ 15421 taxa and 1102 samples ]
diagdds_ser_control_3_Rhizosphere_Pott = phyloseq_to_deseq2(Bacteria_experiment_two_3_Rhizosphere_Pott, ~ Treatment);diagdds_ser_control_3_Rhizosphere_Pott
diagdds_ser_control_3_Rhizosphere_Pott = DESeq(diagdds_ser_control_3_Rhizosphere_Pott, test="Wald", fitType="parametric");diagdds_ser_control_3_Rhizosphere_Pott
res = results(diagdds_ser_control_3_Rhizosphere_Pott, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_ser_control_3_Rhizosphere_Pott = res[which(res$padj < alpha), ];sigtab_ser_control_3_Rhizosphere_Pott
sigtab_ser_control_3_Rhizosphere_Pott = cbind(as(sigtab_ser_control_3_Rhizosphere_Pott, "data.frame"), as(tax_table(Bacteria_experiment_two_3_Rhizosphere_Pott)[rownames(sigtab_ser_control_3_Rhizosphere_Pott), ], "matrix"));sigtab_ser_control_3_Rhizosphere_Pott
head(sigtab_ser_control_3_Rhizosphere_Pott)
write.csv(sigtab_ser_control_3_Rhizosphere_Pott, "sigtab_ser_control_3_Rhizosphere_Pott.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_ser_control_3_Rhizosphere_Pott$log2FoldChange, sigtab_ser_control_3_Rhizosphere_Pott$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_3_Rhizosphere_Pott$Phylum = factor(as.character(sigtab_ser_control_3_Rhizosphere_Pott$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_ser_control_3_Rhizosphere_Pott$log2FoldChange, sigtab_ser_control_3_Rhizosphere_Pott$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_3_Rhizosphere_Pott$Genus = factor(as.character(sigtab_ser_control_3_Rhizosphere_Pott$Genus), levels=names(x))
ggplot(sigtab_ser_control_3_Rhizosphere_Pott, aes(x=Genus, y=log2FoldChange, color=Family)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#####################################################################################################
Bacteria_experiment_two_Soil_Pott= subset_samples(Bacteria_experiment_two_Pott, Sample_type %in% c("2_Soil"));Bacteria_experiment_two_Soil_Pott#[ 15421 taxa and 1102 samples ]
diagdds_ser_control_Soil_Pott = phyloseq_to_deseq2(Bacteria_experiment_two_Soil_Pott, ~ Treatment);diagdds_ser_control_Soil_Pott
diagdds_ser_control_Soil_Pott = DESeq(diagdds_ser_control_Soil_Pott, test="Wald", fitType="parametric");diagdds_ser_control_Soil_Pott
res = results(diagdds_ser_control_Soil_Pott, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_ser_control_Soil_Pott = res[which(res$padj < alpha), ];sigtab_ser_control_Soil_Pott
sigtab_ser_control_Soil_Pott = cbind(as(sigtab_ser_control_Soil_Pott, "data.frame"), as(tax_table(Bacteria_experiment_two_Soil_Pott)[rownames(sigtab_ser_control_Soil_Pott), ], "matrix"));sigtab_ser_control_Soil_Pott
head(sigtab_ser_control_Soil_Pott)
write.csv(sigtab_ser_control_Soil_Pott, "sigtab_ser_control_Soil_Pott.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_ser_control_Soil_Pott$log2FoldChange, sigtab_ser_control_Soil_Pott$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_Soil_Pott$Phylum = factor(as.character(sigtab_ser_control_Soil_Pott$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_ser_control_Soil_Pott$log2FoldChange, sigtab_ser_control_Soil_Pott$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_Soil_Pott$Genus = factor(as.character(sigtab_ser_control_Soil_Pott$Genus), levels=names(x))
ggplot(sigtab_ser_control_Soil_Pott, aes(x=Genus, y=log2FoldChange, color=Family)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Haliangiaceae, Microbacteriaceae,Polyangiaceae, Pseudomonadaceae, Sphingomonadaceae
#####################################################################################################
library(ggplot2)
library(MicrobiotaProcess)
citation("MicrobiotaProcess")
BFamtaxa_EXPT2 <- get_taxadf(obj=Bacteria_experiment_two, taxlevel=5);BFamtaxa_EXPT2
sample_data(BFamtaxa_EXPT2)
#####################################################################################################

bfam_EXPT2 <- ggbartax(obj=BFamtaxa_EXPT2, facetNames="Soil_type_Sample_type_Treatment", plotgroup=TRUE, topn=20) +
  xlab("") +
  ylab("relative abundance (%)") +scale_fill_manual(values=mycolors)+theme(axis.text.x = element_text(angle = 270)) +
  #scale_color_manual(values=c("chartreuse4","bisque4","deepskyblue","Sky Blue","goldenrod", "#ba0d42", "grey85","Orange","coral3","darkred","#0dba2f","#850dba","grey35","Blue","deeppink","blue","Yellow")) + 
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=3))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=0.4))+ 
  theme(axis.title.x = element_text(face="bold", colour="black", size=14),
        axis.title.y = element_text(face="bold", colour="black", size=14),
        axis.text.x = element_blank(),
        axis.text.y  = element_text(colour="black", size=11))+ labs(tag = "C")+
  theme(legend.position = "bottom")+ 
  labs(fill='Phyla')+
  theme(legend.title = element_text(color = "black", size = 12),
        legend.text = element_text(color = "black", size = 12))+
  theme(text=element_text(family="sans"));bfam_EXPT2
#####################################################################################################

#PERMANOVA 
phyloseq_transform_css <- function(physeq, norm = TRUE, log = TRUE, ...){
  # require(metagenomeSeq)
  MGS <- phyloseq::phyloseq_to_metagenomeSeq(physeq)
  # get the normalized count matrix
  otu_norm <- metagenomeSeq::MRcounts(MGS, norm = norm, log = log, ...)
  # exportMat(datt.norm, file = "tmp.txt")    # norm = TRUE, log = TRUE
  ## save sample statistics (sample scaling factor, quantile value, number of identified features and library size):
  # exportStats(dattM.sf, file = "tmp_stats.txt")
  # Substitue raw abundance to the css-normalized data
  physeq.tr <- physeq
  phyloseq::otu_table(physeq.tr) <- phyloseq::otu_table(otu_norm, taxa_are_rows = T)
  return(physeq.tr)
}

##Beta diversity using CSS
Bacteria_experiment_two_CSS = phyloseq_transform_css(Bacteria_experiment_two);Bacteria_experiment_two_CSS#########
#First get otu_table and transpose it:
dist.matrix_Bacteria_experiment_two_CSS <- t(data.frame(otu_table(Bacteria_experiment_two_CSS)))
sampledf_Bacteria_experiment_two_CSS <- data.frame(sample_data(Bacteria_experiment_two_CSS))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_experiment_two_CSS <- vegdist(dist.matrix_Bacteria_experiment_two_CSS, method = "bray"); phyloseq_bray_Bacteria_experiment_two_CSS
sampleda <- data.frame(sample_data(Bacteria_experiment_two_CSS), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_experiment_two_CSS)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_experiment_two_CSS ~ Soil_type*Sample_type*Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#                                 Df SumOfSqs      R2       F Pr(>F)    
#Soil_type                        1   7.6906 0.36527 63.6085  0.001 ***
#Sample_type                      2   2.5392 0.12060 10.5006  0.001 ***
#Treatment                        1   0.5478 0.02602  4.5307  0.004 ** 
#Soil_type:Sample_type            2   1.6282 0.07733  6.7335  0.001 ***
#Soil_type:Treatment              1   0.4658 0.02212  3.8528  0.011 *  
#Sample_type:Treatment            2   0.4269 0.02027  1.7653  0.053 .  
#Soil_type:Sample_type:Treatment  2   0.5020 0.02384  2.0759  0.022 *  
#Residual                        60   7.2543 0.34454                   
#Total                           71  21.0548 1.00000                   
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

####################################################################################################
#Phyllosphere
Bacteria_experiment_two_CSS_phyllosphere= subset_samples(Bacteria_experiment_two_CSS, Sample_type %in% c("4_Phyllosphere"));Bacteria_experiment_two_CSS_phyllosphere#[ 15421 taxa and 1102 samples ]
#visualize
matrixdistance_Bacteria_experiment_two_CSS_phyllosphere_PCoA <- ordinate(physeq = Bacteria_experiment_two_CSS_phyllosphere, method = "PCoA", distance = "bray", trymax = 1000)
#Stress:     0.2069558 

Fig2_serr_cont_phy_bac=plot_ordination(physeq = Bacteria_experiment_two_CSS_phyllosphere, ordination = matrixdistance_Bacteria_experiment_two_CSS_phyllosphere_PCoA, color = "Treatment",shape = "Soil_type") + 
  theme(text = element_text(size = 12)) + scale_color_manual(values =myco)+theme(legend.title=element_blank())+  geom_point(aes(color =Treatment ), size = 4)+
  scale_shape_manual(values=c(19,17, 15,18))+ theme_gray(base_size = 14)+ labs(tag = "C")+
theme(legend.position="none");Fig2_serr_cont_phy_bac
tiff("Fig2_serr_cont_phy_bac.tiff", units="in", width=5, height=4, res=30)
ggsave("Fig2_serr_cont_phy_bac.tiff", plot = Fig2_serr_cont_phy_bac)
dev.off()
####################################################################################################
#3_Rhizosphere
Bacteria_experiment_two_CSS_3_Rhizosphere= subset_samples(Bacteria_experiment_two_CSS, Sample_type %in% c("3_Rhizosphere "));Bacteria_experiment_two_CSS_3_Rhizosphere#[ 15421 taxa and 1102 samples ]
#visualize
matrixdistance_Bacteria_experiment_two_CSS_3_Rhizosphere_PCoA <- ordinate(physeq = Bacteria_experiment_two_CSS_3_Rhizosphere, method = "PCoA", distance = "bray", trymax = 1000)
#Stress:     0.2069558 

Fig2_serr_cont_rhi_bac=plot_ordination(physeq = Bacteria_experiment_two_CSS_3_Rhizosphere, ordination = matrixdistance_Bacteria_experiment_two_CSS_3_Rhizosphere_PCoA, color = "Treatment",shape = "Soil_type") + 
  theme(text = element_text(size = 12)) + scale_color_manual(values =myco)+theme(legend.title=element_blank())+  geom_point(aes(color =Treatment ), size = 4)+
  scale_shape_manual(values=c(19,17, 15,18))+ theme_gray(base_size = 14)+ labs(tag = "B")+
theme(legend.position="none");Fig2_serr_cont_rhi_bac
tiff("Fig2_serr_cont_rhi_bac.tiff", units="in", width=5, height=4, res=300)
ggsave("Fig2_serr_cont_rhi_bac.tiff", plot = Fig2_serr_cont_rhi_bac)
dev.off()
####################################################################################################
####################################################################################################
#Soil
Bacteria_experiment_two_CSS_Soil= subset_samples(Bacteria_experiment_two_CSS, Sample_type %in% c("2_Soil"));Bacteria_experiment_two_CSS_Soil#[ 15421 taxa and 1102 samples ]
#visualize
matrixdistance_Bacteria_experiment_two_CSS_Soil_PCoA <- ordinate(physeq = Bacteria_experiment_two_CSS_Soil, method = "PCoA", distance = "bray", trymax = 1000)
#Stress:     0.2069558 

Fig2_serr_cont_soil_bac=plot_ordination(physeq = Bacteria_experiment_two_CSS_Soil, ordination = matrixdistance_Bacteria_experiment_two_CSS_Soil_PCoA, color = "Treatment",shape = "Soil_type") + 
  theme(text = element_text(size = 12)) + scale_color_manual(values =myco)+  geom_point(aes(color =Treatment ), size = 4)+
  scale_shape_manual(values=c(19,17, 15,18))+ theme_gray(base_size = 14)+ labs(tag = "A")+
theme(legend.position="none");Fig2_serr_cont_soil_bac
tiff("Fig2_serr_cont_soil_bac.tiff", units="in", width=5, height=4, res=300)
ggsave("Fig2_serr_cont_soil_bac.tiff", plot = Fig2_serr_cont_soil_bac)
dev.off()
####################################################################################################
####################################################################################################

#Environmental soil
Bacteria_experiment_two_CSS_Env= subset_samples(Bacteria_experiment_two_CSS, Soil_type %in% c("Environmental_soil"));Bacteria_experiment_two_CSS_Env#[ 15421 taxa and 1102 samples ]

#First get otu_table and transpose it:
dist.matrix_Bacteria_experiment_two_CSS_Env <- t(data.frame(otu_table(Bacteria_experiment_two_CSS_Env)))
sampledf_Bacteria_experiment_two_CSS_Env <- data.frame(sample_data(Bacteria_experiment_two_CSS_Env))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_experiment_two_CSS_Env <- vegdist(dist.matrix_Bacteria_experiment_two_CSS_Env, method = "bray"); phyloseq_bray_Bacteria_experiment_two_CSS_Env
sampleda <- data.frame(sample_data(Bacteria_experiment_two_CSS_Env), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_experiment_two_CSS_Env)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_experiment_two_CSS_Env ~Sample_type*Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#                       Df SumOfSqs      R2      F Pr(>F)    
#Sample_type            2   1.7788 0.21533 5.3920  0.001 ***
#Treatment              1   0.8423 0.10197 5.1067  0.001 ***
#Sample_type:Treatment  2   0.6913 0.08369 2.0957  0.006 ** 
#Residual              30   4.9483 0.59902                  
#Total                 35   8.2607 1.00000                  
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

#Subset the different compartments
#Phyllosphere
Bacteria_experiment_two_CSS_Phyllosphere= subset_samples(Bacteria_experiment_two_CSS_Env, Sample_type %in% c("Phyllosphere"));Bacteria_experiment_two_CSS_Phyllosphere#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_experiment_two_CSS_Phyllosphere <- t(data.frame(otu_table(Bacteria_experiment_two_CSS_Phyllosphere)))
sampledf_Bacteria_experiment_two_CSS_Phyllosphere <- data.frame(sample_data(Bacteria_experiment_two_CSS_Phyllosphere))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_experiment_two_CSS_Phyllosphere <- vegdist(dist.matrix_Bacteria_experiment_two_CSS_Phyllosphere, method = "bray"); phyloseq_bray_Bacteria_experiment_two_CSS_Phyllosphere
sampleda <- data.frame(sample_data(Bacteria_experiment_two_CSS_Phyllosphere), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_experiment_two_CSS_Phyllosphere)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_experiment_two_CSS_Phyllosphere ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.63624 0.23941 3.1477  0.008 **
#Residual  10  2.02131 0.76059                 
#Total     11  2.65755 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1


################################################################################################################################################################
#Subset the different compartments
#3_Rhizosphere
Bacteria_experiment_two_CSS_3_Rhizosphere= subset_samples(Bacteria_experiment_two_CSS_Env, Sample_type %in% c("3_Rhizosphere"));Bacteria_experiment_two_CSS_3_Rhizosphere#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_experiment_two_CSS_3_Rhizosphere <- t(data.frame(otu_table(Bacteria_experiment_two_CSS_3_Rhizosphere)))
sampledf_Bacteria_experiment_two_CSS_3_Rhizosphere <- data.frame(sample_data(Bacteria_experiment_two_CSS_3_Rhizosphere))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_experiment_two_CSS_3_Rhizosphere <- vegdist(dist.matrix_Bacteria_experiment_two_CSS_3_Rhizosphere, method = "bray"); phyloseq_bray_Bacteria_experiment_two_CSS_3_Rhizosphere
sampleda <- data.frame(sample_data(Bacteria_experiment_two_CSS_3_Rhizosphere), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_experiment_two_CSS_3_Rhizosphere)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_experiment_two_CSS_3_Rhizosphere ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)
#Treatment  1  0.23068 0.11675 1.3218  0.134
#Residual  10  1.74523 0.88325              
#Total     11  1.97591 1.00000   
#####################################################################################################
#Subset the different compartments
#Soil
Bacteria_experiment_two_CSS_Soil= subset_samples(Bacteria_experiment_two_CSS_Env, Sample_type %in% c("Soil"));Bacteria_experiment_two_CSS_Soil#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_experiment_two_CSS_Soil <- t(data.frame(otu_table(Bacteria_experiment_two_CSS_Soil)))
sampledf_Bacteria_experiment_two_CSS_Soil <- data.frame(sample_data(Bacteria_experiment_two_CSS_Soil))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_experiment_two_CSS_Soil <- vegdist(dist.matrix_Bacteria_experiment_two_CSS_Soil, method = "bray"); phyloseq_bray_Bacteria_experiment_two_CSS_Soil
sampleda <- data.frame(sample_data(Bacteria_experiment_two_CSS_Soil), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_experiment_two_CSS_Soil)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_experiment_two_CSS_Soil ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.66673 0.36069 5.6418  0.002 **
#Residual  10  1.18178 0.63931                 
#Total     11  1.84851 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
#####################################################################################################
#Potting soil
Bacteria_experiment_two_CSS_Pott= subset_samples(Bacteria_experiment_two_CSS, Soil_type %in% c("Potting_soil"));Bacteria_experiment_two_CSS_Pott#[ 15421 taxa and 1102 samples ]

#First get otu_table and transpose it:
dist.matrix_Bacteria_experiment_two_CSS_Pott <- t(data.frame(otu_table(Bacteria_experiment_two_CSS_Pott)))
sampledf_Bacteria_experiment_two_CSS_Pott <- data.frame(sample_data(Bacteria_experiment_two_CSS_Pott))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_experiment_two_CSS_Pott <- vegdist(dist.matrix_Bacteria_experiment_two_CSS_Pott, method = "bray"); phyloseq_bray_Bacteria_experiment_two_CSS_Pott
sampleda <- data.frame(sample_data(Bacteria_experiment_two_CSS_Pott), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_experiment_two_CSS_Pott)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_experiment_two_CSS_Pott ~Sample_type*Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#                       Df SumOfSqs      R2       F Pr(>F)    
#Sample_type            2   2.3886 0.46804 15.5374  0.001 ***
#Treatment              1   0.1713 0.03356  2.2284  0.055 .  
#Sample_type:Treatment  2   0.2375 0.04654  1.5449  0.108    
#Residual              30   2.3060 0.45185                   
#Total                 35   5.1035 1.00000                   
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

#Subset the different compartments
#Phyllosphere
Bacteria_experiment_two_CSS_Phyllosphere= subset_samples(Bacteria_experiment_two_CSS_Pott, Sample_type %in% c("Phyllosphere"));Bacteria_experiment_two_CSS_Phyllosphere#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_experiment_two_CSS_Phyllosphere <- t(data.frame(otu_table(Bacteria_experiment_two_CSS_Phyllosphere)))
sampledf_Bacteria_experiment_two_CSS_Phyllosphere <- data.frame(sample_data(Bacteria_experiment_two_CSS_Phyllosphere))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_experiment_two_CSS_Phyllosphere <- vegdist(dist.matrix_Bacteria_experiment_two_CSS_Phyllosphere, method = "bray"); phyloseq_bray_Bacteria_experiment_two_CSS_Phyllosphere
sampleda <- data.frame(sample_data(Bacteria_experiment_two_CSS_Phyllosphere), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_experiment_two_CSS_Phyllosphere)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_experiment_two_CSS_Phyllosphere ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)  
#Treatment  1  0.22205 0.13081 1.5049  0.019 *
#Residual  10  1.47545 0.86919                
#Total     11  1.69750 1.00000                
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

################################################################################################################################################################
#Subset the different compartments
#3_Rhizosphere
Bacteria_experiment_two_CSS_3_Rhizosphere= subset_samples(Bacteria_experiment_two_CSS_Pott, Sample_type %in% c("3_Rhizosphere"));Bacteria_experiment_two_CSS_3_Rhizosphere#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_experiment_two_CSS_3_Rhizosphere <- t(data.frame(otu_table(Bacteria_experiment_two_CSS_3_Rhizosphere)))
sampledf_Bacteria_experiment_two_CSS_3_Rhizosphere <- data.frame(sample_data(Bacteria_experiment_two_CSS_3_Rhizosphere))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_experiment_two_CSS_3_Rhizosphere <- vegdist(dist.matrix_Bacteria_experiment_two_CSS_3_Rhizosphere, method = "bray"); phyloseq_bray_Bacteria_experiment_two_CSS_3_Rhizosphere
sampleda <- data.frame(sample_data(Bacteria_experiment_two_CSS_3_Rhizosphere), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_experiment_two_CSS_3_Rhizosphere)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_experiment_two_CSS_3_Rhizosphere ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)
#Treatment  1  0.05445 0.12125 1.3798   0.13
#Residual  10  0.39463 0.87875              
#Total     11  0.44908 1.00000   
#####################################################################################################
#Subset the different compartments
#Soil
Bacteria_experiment_two_CSS_Soil= subset_samples(Bacteria_experiment_two_CSS_Pott, Sample_type %in% c("Soil"));Bacteria_experiment_two_CSS_Soil#[ 15421 taxa and 1102 samples ]
dist.matrix_Bacteria_experiment_two_CSS_Soil <- t(data.frame(otu_table(Bacteria_experiment_two_CSS_Soil)))
sampledf_Bacteria_experiment_two_CSS_Soil <- data.frame(sample_data(Bacteria_experiment_two_CSS_Soil))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Bacteria_experiment_two_CSS_Soil <- vegdist(dist.matrix_Bacteria_experiment_two_CSS_Soil, method = "bray"); phyloseq_bray_Bacteria_experiment_two_CSS_Soil
sampleda <- data.frame(sample_data(Bacteria_experiment_two_CSS_Soil), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Bacteria_experiment_two_CSS_Soil)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Bacteria_experiment_two_CSS_Soil ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)  
#Treatment  1  0.13230 0.23283 3.0349   0.02 *
#Residual  10  0.43594 0.76717                
#Total     11  0.56824 1.00000                
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
```

#cosider experiment2

```{r,echo=FALSE,warning=FALSE}
setwd("/Users/expeditoolimi/Documents/Manuscript-AA-OE_SB/Microbiome_analysis")
load("/Users/expeditoolimi/Documents/Manuscript-AA-OE_SB/Microbiome_analysis/Fungi_mereged.RData")

Fungi_experiment_two= subset_samples(Fungi_mereged, Experiment %in% c("Two"));Fungi_experiment_two#[ 15421 taxa and 1102 samples ]
Fungi_experiment_two= subset_samples(Fungi_experiment_two, Sample_type %in% c("4_Phyllosphere","3_Rhizosphere","2_Soil"));Fungi_experiment_two#[ 15421 taxa and 1102 samples ]
Fungi_experiment_two= subset_samples(Fungi_experiment_two, Treatment %in% c("2_Serratia_plymuthica_HRO_C48","1_Control"));Fungi_experiment_two#[ 15421 taxa and 1102 samples ]
#####################################################################################################
Fungi_experiment_two_table = as(otu_table(Fungi_experiment_two), "matrix")
# Coerce to data.frame
Fungi_experiment_two_table = as.data.frame(Fungi_experiment_two_table)
write.csv(Fungi_experiment_two_table ,file = "Fungi_experiment_two_table.csv")
#####################################################################################################
#Environmetal soil
Fungi_experiment_two_env_fungi= subset_samples(Fungi_experiment_two, Soil_type %in% c("Environmental_soil"));Fungi_experiment_two_env_fungi#[ 15421 taxa and 1102 samples ]
#phyllosphere
Fungi_experiment_two_Phyllosphere_env_fungi= subset_samples(Fungi_experiment_two_env_fungi, Sample_type %in% c("4_Phyllosphere"));Fungi_experiment_two_Phyllosphere_env_fungi#[ 15421 taxa and 1102 samples ]
diagdds_ser_control_phyllosphere_env_fungi = phyloseq_to_deseq2(Fungi_experiment_two_Phyllosphere_env_fungi, ~ Treatment);diagdds_ser_control_phyllosphere_env_fungi
diagdds_ser_control_phyllosphere_env_fungi = DESeq(diagdds_ser_control_phyllosphere_env_fungi, test="Wald", fitType="parametric");diagdds_ser_control_phyllosphere_env_fungi
res = results(diagdds_ser_control_phyllosphere_env_fungi, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_ser_control_phyllosphere_env_fungi = res[which(res$padj < alpha), ];sigtab_ser_control_phyllosphere_env_fungi
sigtab_ser_control_phyllosphere_env_fungi = cbind(as(sigtab_ser_control_phyllosphere_env_fungi, "data.frame"), as(tax_table(Fungi_experiment_two_Phyllosphere_env_fungi)[rownames(sigtab_ser_control_phyllosphere_env_fungi), ], "matrix"));sigtab_ser_control_phyllosphere_env_fungi
head(sigtab_ser_control_phyllosphere_env_fungi)
write.csv(sigtab_ser_control_phyllosphere_env_fungi, "sigtab_ser_control_phyllosphere_env_fungi.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_ser_control_phyllosphere_env_fungi$log2FoldChange, sigtab_ser_control_phyllosphere_env_fungi$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_phyllosphere_env_fungi$Phylum = factor(as.character(sigtab_ser_control_phyllosphere_env_fungi$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_ser_control_phyllosphere_env_fungi$log2FoldChange, sigtab_ser_control_phyllosphere_env_fungi$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_phyllosphere_env_fungi$Genus = factor(as.character(sigtab_ser_control_phyllosphere_env_fungi$Genus), levels=names(x))
ggplot(sigtab_ser_control_phyllosphere_env_fungi, aes(x=Genus, y=log2FoldChange, color=Family)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Amphisphaeriaceae,Aspergillaceae,Bondarzewiaceae, Chaetomiaceae, Cladosporiaceae, Clavicipitaceae, Cystobasidiaceae, Didymellaceae,Filobasidiaceae,
#Leptosphaeriaceae,Leucosporidiaceae,Nectriaceae,Pleosporaceae,Pseudeurotiaceae,Sclerotiniaceae,Trichocomaceae
#####################################################################################################
Fungi_experiment_two_3_Rhizosphere_env_fungi= subset_samples(Fungi_experiment_two_env_fungi, Sample_type %in% c("3_Rhizosphere"));Fungi_experiment_two_3_Rhizosphere_env_fungi#[ 15421 taxa and 1102 samples ]
diagdds_ser_control_3_Rhizosphere_env_fungi = phyloseq_to_deseq2(Fungi_experiment_two_3_Rhizosphere_env_fungi, ~ Treatment);diagdds_ser_control_3_Rhizosphere_env_fungi
diagdds_ser_control_3_Rhizosphere_env_fungi = DESeq(diagdds_ser_control_3_Rhizosphere_env_fungi, test="Wald", fitType="parametric");diagdds_ser_control_3_Rhizosphere_env_fungi
res = results(diagdds_ser_control_3_Rhizosphere_env_fungi, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_ser_control_3_Rhizosphere_env_fungi = res[which(res$padj < alpha), ];sigtab_ser_control_3_Rhizosphere_env_fungi
sigtab_ser_control_3_Rhizosphere_env_fungi = cbind(as(sigtab_ser_control_3_Rhizosphere_env_fungi, "data.frame"), as(tax_table(Fungi_experiment_two_3_Rhizosphere_env_fungi)[rownames(sigtab_ser_control_3_Rhizosphere_env_fungi), ], "matrix"));sigtab_ser_control_3_Rhizosphere_env_fungi
head(sigtab_ser_control_3_Rhizosphere_env_fungi)
write.csv(sigtab_ser_control_3_Rhizosphere_env_fungi, "sigtab_ser_control_3_Rhizosphere_env_fungi.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_ser_control_3_Rhizosphere_env_fungi$log2FoldChange, sigtab_ser_control_3_Rhizosphere_env_fungi$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_3_Rhizosphere_env_fungi$Phylum = factor(as.character(sigtab_ser_control_3_Rhizosphere_env_fungi$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_ser_control_3_Rhizosphere_env_fungi$log2FoldChange, sigtab_ser_control_3_Rhizosphere_env_fungi$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_3_Rhizosphere_env_fungi$Genus = factor(as.character(sigtab_ser_control_3_Rhizosphere_env_fungi$Genus), levels=names(x))
ggplot(sigtab_ser_control_3_Rhizosphere_env_fungi, aes(x=Genus, y=log2FoldChange, color=Family)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Aspergillaceae,Didymellaceae,Endogonales_fam_Incertae_sedis,Nectriaceae,Pseudeurotiaceae,Unidentified_Chytridiomycota,Unidentified_Fungi,Unidentified_Sebacinales,
#Unidentified_Tremellomycetes
#####################################################################################################
Fungi_experiment_two_Soil_env_fungi= subset_samples(Fungi_experiment_two_env_fungi, Sample_type %in% c("2_Soil"));Fungi_experiment_two_Soil_env_fungi#[ 15421 taxa and 1102 samples ]
diagdds_ser_control_Soil_env_fungi = phyloseq_to_deseq2(Fungi_experiment_two_Soil_env_fungi, ~ Treatment);diagdds_ser_control_Soil_env_fungi
diagdds_ser_control_Soil_env_fungi = DESeq(diagdds_ser_control_Soil_env_fungi, test="Wald", fitType="parametric");diagdds_ser_control_Soil_env_fungi
res = results(diagdds_ser_control_Soil_env_fungi, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_ser_control_Soil_env_fungi = res[which(res$padj < alpha), ];sigtab_ser_control_Soil_env_fungi
sigtab_ser_control_Soil_env_fungi = cbind(as(sigtab_ser_control_Soil_env_fungi, "data.frame"), as(tax_table(Fungi_experiment_two_Soil_env_fungi)[rownames(sigtab_ser_control_Soil_env_fungi), ], "matrix"));sigtab_ser_control_Soil_env_fungi
head(sigtab_ser_control_Soil_env_fungi)
write.csv(sigtab_ser_control_Soil_env_fungi, "sigtab_ser_control_Soil_env_fungi.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_ser_control_Soil_env_fungi$log2FoldChange, sigtab_ser_control_Soil_env_fungi$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_Soil_env_fungi$Phylum = factor(as.character(sigtab_ser_control_Soil_env_fungi$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_ser_control_Soil_env_fungi$log2FoldChange, sigtab_ser_control_Soil_env_fungi$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_Soil_env_fungi$Genus = factor(as.character(sigtab_ser_control_Soil_env_fungi$Genus), levels=names(x))
ggplot(sigtab_ser_control_Soil_env_fungi, aes(x=Genus, y=log2FoldChange, color=Family)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Dictyosporiaceae, Endogonales_fam_Incertae_sedis,Leptosphaeriaceae,Mortierellaceae,Myxotrichaceae,Nectriaceae,Pseudeurotiaceae,Unidentified_Helotiales,Unidentified_Pleosporales
#Unidentified_Sordariales,Unidentified_Thelebolales..
#####################################################################################################
#Pottironmetal soil
Fungi_experiment_two_Pott_fungi= subset_samples(Fungi_experiment_two, Soil_type %in% c("Potting_soil"));Fungi_experiment_two_Pott_fungi#[ 15421 taxa and 1102 samples ]

#phyllosphere
Fungi_experiment_two_Phyllosphere_Pott_fungi= subset_samples(Fungi_experiment_two_Pott_fungi, Sample_type %in% c("4_Phyllosphere"));Fungi_experiment_two_Phyllosphere_Pott_fungi#[ 15421 taxa and 1102 samples ]
diagdds_ser_control_phyllosphere_Pott_fungi = phyloseq_to_deseq2(Fungi_experiment_two_Phyllosphere_Pott_fungi, ~ Treatment);diagdds_ser_control_phyllosphere_Pott_fungi
diagdds_ser_control_phyllosphere_Pott_fungi = DESeq(diagdds_ser_control_phyllosphere_Pott_fungi, test="Wald", fitType="parametric");diagdds_ser_control_phyllosphere_Pott_fungi
res = results(diagdds_ser_control_phyllosphere_Pott_fungi, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_ser_control_phyllosphere_Pott_fungi = res[which(res$padj < alpha), ];sigtab_ser_control_phyllosphere_Pott_fungi
sigtab_ser_control_phyllosphere_Pott_fungi = cbind(as(sigtab_ser_control_phyllosphere_Pott_fungi, "data.frame"), as(tax_table(Fungi_experiment_two_Phyllosphere_Pott_fungi)[rownames(sigtab_ser_control_phyllosphere_Pott_fungi), ], "matrix"));sigtab_ser_control_phyllosphere_Pott_fungi
head(sigtab_ser_control_phyllosphere_Pott_fungi)
write.csv(sigtab_ser_control_phyllosphere_Pott_fungi, "sigtab_ser_control_phyllosphere_Pott_fungi.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_ser_control_phyllosphere_Pott_fungi$log2FoldChange, sigtab_ser_control_phyllosphere_Pott_fungi$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_phyllosphere_Pott_fungi$Phylum = factor(as.character(sigtab_ser_control_phyllosphere_Pott_fungi$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_ser_control_phyllosphere_Pott_fungi$log2FoldChange, sigtab_ser_control_phyllosphere_Pott_fungi$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_phyllosphere_Pott_fungi$Genus = factor(as.character(sigtab_ser_control_phyllosphere_Pott_fungi$Genus), levels=names(x))
ggplot(sigtab_ser_control_phyllosphere_Pott_fungi, aes(x=Genus, y=log2FoldChange, color=Family)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Aspergillaceae,Aureobasidiaceae,Bulleribasidiaceae,Chaetomiaceae,Cystobasidiaceae,Didymellaceae,Endogonales_fam_Incertae_sedis,Hypocreales_fam_Incertae_sedis
#Leucosporidiaceae,Nectriaceae,Orbiliaceae,Pleosporaceae,Unidentified_Auriculariales,Unidentified_Hypocreales,Unidentified_Pleosporales,Unidentified_Helotiales,
#Unidentified_Saccharomycetales,Unidentified_Sordariales, Unidentified_Sordariomycetes
#####################################################################################################
Fungi_experiment_two_3_Rhizosphere_Pott_fungi= subset_samples(Fungi_experiment_two_Pott_fungi, Sample_type %in% c("3_Rhizosphere"));Fungi_experiment_two_3_Rhizosphere_Pott_fungi#[ 15421 taxa and 1102 samples ]
diagdds_ser_control_3_Rhizosphere_Pott_fungi = phyloseq_to_deseq2(Fungi_experiment_two_3_Rhizosphere_Pott_fungi, ~ Treatment);diagdds_ser_control_3_Rhizosphere_Pott_fungi
diagdds_ser_control_3_Rhizosphere_Pott_fungi = DESeq(diagdds_ser_control_3_Rhizosphere_Pott_fungi, test="Wald", fitType="parametric");diagdds_ser_control_3_Rhizosphere_Pott_fungi
res = results(diagdds_ser_control_3_Rhizosphere_Pott_fungi, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_ser_control_3_Rhizosphere_Pott_fungi = res[which(res$padj < alpha), ];sigtab_ser_control_3_Rhizosphere_Pott_fungi
sigtab_ser_control_3_Rhizosphere_Pott_fungi = cbind(as(sigtab_ser_control_3_Rhizosphere_Pott_fungi, "data.frame"), as(tax_table(Fungi_experiment_two_3_Rhizosphere_Pott_fungi)[rownames(sigtab_ser_control_3_Rhizosphere_Pott_fungi), ], "matrix"));sigtab_ser_control_3_Rhizosphere_Pott_fungi
head(sigtab_ser_control_3_Rhizosphere_Pott_fungi)
write.csv(sigtab_ser_control_3_Rhizosphere_Pott_fungi, "sigtab_ser_control_3_Rhizosphere_Pott_fungi.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_ser_control_3_Rhizosphere_Pott_fungi$log2FoldChange, sigtab_ser_control_3_Rhizosphere_Pott_fungi$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_3_Rhizosphere_Pott_fungi$Phylum = factor(as.character(sigtab_ser_control_3_Rhizosphere_Pott_fungi$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_ser_control_3_Rhizosphere_Pott_fungi$log2FoldChange, sigtab_ser_control_3_Rhizosphere_Pott_fungi$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_3_Rhizosphere_Pott_fungi$Genus = factor(as.character(sigtab_ser_control_3_Rhizosphere_Pott_fungi$Genus), levels=names(x))
ggplot(sigtab_ser_control_3_Rhizosphere_Pott_fungi, aes(x=Genus, y=log2FoldChange, color=Family)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Aspergillaceae,Didymellaceae,Endogonales_fam_Incertae_sedis,Leucosporidiaceae,Mortierellaceae,Pseudeurotiaceae,Unidentified_Helotiales

#####################################################################################################
Fungi_experiment_two_Soil_Pott_fungi= subset_samples(Fungi_experiment_two_Pott_fungi, Sample_type %in% c("2_Soil"));Fungi_experiment_two_Soil_Pott_fungi#[ 15421 taxa and 1102 samples ]
diagdds_ser_control_Soil_Pott_fungi = phyloseq_to_deseq2(Fungi_experiment_two_Soil_Pott_fungi, ~ Treatment);diagdds_ser_control_Soil_Pott_fungi
diagdds_ser_control_Soil_Pott_fungi = DESeq(diagdds_ser_control_Soil_Pott_fungi, test="Wald", fitType="parametric");diagdds_ser_control_Soil_Pott_fungi
res = results(diagdds_ser_control_Soil_Pott_fungi, cooksCutoff = FALSE);res
alpha = 0.05
sigtab_ser_control_Soil_Pott_fungi = res[which(res$padj < alpha), ];sigtab_ser_control_Soil_Pott_fungi
sigtab_ser_control_Soil_Pott_fungi = cbind(as(sigtab_ser_control_Soil_Pott_fungi, "data.frame"), as(tax_table(Fungi_experiment_two_Soil_Pott_fungi)[rownames(sigtab_ser_control_Soil_Pott_fungi), ], "matrix"));sigtab_ser_control_Soil_Pott_fungi
head(sigtab_ser_control_Soil_Pott_fungi)
write.csv(sigtab_ser_control_Soil_Pott_fungi, "sigtab_ser_control_Soil_Pott_fungi.csv")
library("ggplot2")
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab_ser_control_Soil_Pott_fungi$log2FoldChange, sigtab_ser_control_Soil_Pott_fungi$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_Soil_Pott_fungi$Phylum = factor(as.character(sigtab_ser_control_Soil_Pott_fungi$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab_ser_control_Soil_Pott_fungi$log2FoldChange, sigtab_ser_control_Soil_Pott_fungi$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab_ser_control_Soil_Pott_fungi$Genus = factor(as.character(sigtab_ser_control_Soil_Pott_fungi$Genus), levels=names(x))
ggplot(sigtab_ser_control_Soil_Pott_fungi, aes(x=Genus, y=log2FoldChange, color=Family)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#Aspergillaceae,Didymellaceae,Endogonales_fam_Incertae_sedis,Helotiaceae,Nectriaceae,Unidentified_Helotiales,Unidentified_Pleosporales
#####################################################################################################
library(ggplot2)
library(MicrobiotaProcess)
citation("MicrobiotaProcess")
fgentaxa_EXPT2 <- get_taxadf(obj=Fungi_experiment_two, taxlevel=5);fgentaxa_EXPT2
########################################################################################################################################
fgen_EXPT2 <- ggbartax(obj=fgentaxa_EXPT2, facetNames="Soil_type_Sample_type_Treatment", plotgroup=TRUE, topn=20) +
  xlab("") +
  ylab("relative abundance (%)") +scale_fill_manual(values=mycolors)+theme(axis.text.x = element_text(angle = 270)) +
  #scale_color_manual(values=c("chartreuse4","bisque4","deepskyblue","Sky Blue","goldenrod", "#ba0d42", "grey85","Orange","coral3","darkred","#0dba2f","#850dba","grey35","Blue","deeppink","blue","Yellow")) + 
  guides(fill= guide_legend(keywidth = 0.6, keyheight = 0.6, ncol=3))+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=0.4))+ 
  theme(axis.title.x = element_text(face="bold", colour="black", size=14),
        axis.title.y = element_text(face="bold", colour="black", size=14),
        axis.text.x = element_blank(),
        axis.text.y  = element_text(colour="black", size=11))+ labs(tag = "F")+
  theme(legend.position = "bottom")+ 
  labs(fill='Phyla')+
  theme(legend.title = element_text(color = "black", size = 12),
        legend.text = element_text(color = "black", size = 12))+
  theme(text=element_text(family="sans"));fgen_EXPT2

#Figure S4 c and f
tiff("Figure4.tiff", units="in", width=18.5, height=10, res=300)
Figure4.tiff=bfam_EXPT2+fgen_EXPT2+plot_layout(ncol = 2);Figure4.tiff
ggsave("Figure4.tiff", plot = Figure4.tiff)
dev.off()

#####################################################################################################
#PERMANOVA 
phyloseq_transform_css <- function(physeq, norm = TRUE, log = TRUE, ...){
  # require(metagenomeSeq)
  MGS <- phyloseq::phyloseq_to_metagenomeSeq(physeq)
  # get the normalized count matrix
  otu_norm <- metagenomeSeq::MRcounts(MGS, norm = norm, log = log, ...)
  # exportMat(datt.norm, file = "tmp.txt")    # norm = TRUE, log = TRUE
  ## save sample statistics (sample scaling factor, quantile value, number of identified features and library size):
  # exportStats(dattM.sf, file = "tmp_stats.txt")
  # Substitue raw abundance to the css-normalized data
  physeq.tr <- physeq
  phyloseq::otu_table(physeq.tr) <- phyloseq::otu_table(otu_norm, taxa_are_rows = T)
  return(physeq.tr)
}

##Beta diversity using CSS
Fungi_experiment_two_CSS = phyloseq_transform_css(Fungi_experiment_two);Fungi_experiment_two_CSS#########
#First get otu_table and transpose it:
dist.matrix_Fungi_experiment_two_CSS <- t(data.frame(otu_table(Fungi_experiment_two_CSS)))
sampledf_Fungi_experiment_two_CSS <- data.frame(sample_data(Fungi_experiment_two_CSS))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_experiment_two_CSS <- vegdist(dist.matrix_Fungi_experiment_two_CSS, method = "bray"); phyloseq_bray_Fungi_experiment_two_CSS
sampleda <- data.frame(sample_data(Fungi_experiment_two_CSS), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_experiment_two_CSS)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_experiment_two_CSS ~ Soil_type*Sample_type*Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#                                 Df SumOfSqs      R2       F Pr(>F)    
#Soil_type                        1   6.3113 0.37721 63.5307  0.001 ***
#Sample_type                      2   1.4153 0.08459  7.1234  0.001 ***
#Treatment                        1   0.5114 0.03057  5.1479  0.001 ***
#Soil_type:Sample_type            2   0.9984 0.05967  5.0248  0.001 ***
#Soil_type:Treatment              1   0.6351 0.03796  6.3932  0.001 ***
#Sample_type:Treatment            2   0.4413 0.02637  2.2209  0.026 *  
#Soil_type:Sample_type:Treatment  2   0.4583 0.02739  2.3068  0.018 *  
#Residual                        60   5.9606 0.35624                   
#Total                           71  16.7317 1.00000                   
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
####################################################################################################

#Phyllosphere
Fungi_experiment_two_CSS_phyllosphere= subset_samples(Fungi_experiment_two_CSS, Sample_type %in% c("4_Phyllosphere"));Fungi_experiment_two_CSS_phyllosphere#[ 15421 taxa and 1102 samples ]
#visualize
matrixdistance_Fungi_experiment_two_CSS_phyllosphere_PCoA <- ordinate(physeq = Fungi_experiment_two_CSS_phyllosphere, method = "PCoA", distance = "bray", trymax = 1000)
#Stress:     0.2069558 

Fig2_serr_cont_phy_fung=plot_ordination(physeq = Fungi_experiment_two_CSS_phyllosphere, ordination = matrixdistance_Fungi_experiment_two_CSS_phyllosphere_PCoA, color = "Treatment",shape = "Soil_type") + 
  theme(text = element_text(size = 12)) + scale_color_manual(values =myco)+theme(legend.title=element_blank())+  geom_point(aes(color =Treatment ), size = 4)+
  scale_shape_manual(values=c(19,17, 15,18))+ theme_gray(base_size = 14)+ labs(tag = "F")+
theme(legend.position="none");Fig2_serr_cont_phy_fung
tiff("Fig2_serr_cont_phy_fung.tiff", units="in", width=5, height=4, res=300)
ggsave("Fig2_serr_cont_phy_fung.tiff", plot = Fig2_serr_cont_phy_fung)
dev.off()
####################################################################################################
#3_Rhizosphere
Fungi_experiment_two_CSS_3_Rhizosphere= subset_samples(Fungi_experiment_two_CSS, Sample_type %in% c("3_Rhizosphere"));Fungi_experiment_two_CSS_3_Rhizosphere#[ 15421 taxa and 1102 samples ]
#visualize
matrixdistance_Fungi_experiment_two_CSS_3_Rhizosphere_PCoA <- ordinate(physeq = Fungi_experiment_two_CSS_3_Rhizosphere, method = "PCoA", distance = "bray", trymax = 1000)
#Stress:     0.2069558 

Fig2_serr_cont_rhi_fung=plot_ordination(physeq = Fungi_experiment_two_CSS_3_Rhizosphere, ordination = matrixdistance_Fungi_experiment_two_CSS_3_Rhizosphere_PCoA, color = "Treatment",shape = "Soil_type") + 
  theme(text = element_text(size = 12)) + scale_color_manual(values =myco)+theme(legend.title=element_blank())+  geom_point(aes(color =Treatment ), size = 4)+
  scale_shape_manual(values=c(19,17, 15,18))+ theme_gray(base_size = 14)+ labs(tag = "E")+
theme(legend.position="none");Fig2_serr_cont_rhi_fung
tiff("Fig2_serr_cont_rhi_fung.tiff", units="in", width=5, height=4, res=300)
ggsave("Fig2_serr_cont_rhi_fung.tiff", plot = Fig2_serr_cont_rhi_fung)
dev.off()
####################################################################################################
####################################################################################################
#Soil
Fungi_experiment_two_CSS_Soil= subset_samples(Fungi_experiment_two_CSS, Sample_type %in% c("2_Soil"));Fungi_experiment_two_CSS_Soil#[ 15421 taxa and 1102 samples ]
#visualize
matrixdistance_Fungi_experiment_two_CSS_Soil_PCoA <- ordinate(physeq = Fungi_experiment_two_CSS_Soil, method = "PCoA", distance = "bray", trymax = 1000)
#Stress:     0.2069558 

Fig2_serr_cont_soil_fung=plot_ordination(physeq = Fungi_experiment_two_CSS_Soil, ordination = matrixdistance_Fungi_experiment_two_CSS_Soil_PCoA, color = "Treatment",shape = "Soil_type") + 
  theme(text = element_text(size = 12)) + scale_color_manual(values =myco)+theme(legend.title=element_blank())+  geom_point(aes(color =Treatment ), size = 4)+
  scale_shape_manual(values=c(19,17, 15,18))+ theme_gray(base_size = 14)+ labs(tag = "D")+
  theme(legend.position="none");Fig2_serr_cont_soil_fung
tiff("Fig2_serr_cont_soil_fung.tiff", units="in", width=5, height=4, res=300)
ggsave("Fig2_serr_cont_soil_fung.tiff", plot = Fig2_serr_cont_soil_fung)
dev.off()
####################################################################################################
#Combine all the figures
#Figure S7 SOILS
tiff("Figure3.tiff", units="in", width=11, height=7, res=250)
Figure3.tiff=Fig2_serr_cont_soil_bac+Fig2_serr_cont_rhi_bac+Fig2_serr_cont_phy_bac+Fig2_serr_cont_soil_fung+Fig2_serr_cont_rhi_fung+Fig2_serr_cont_phy_fung+plot_layout(ncol = 3);Figure3.tiff
ggsave("Figure3.tiff", plot = Figure3.tiff)
####################################################################################################
#Environmental soil
Fungi_experiment_two_CSS_Env= subset_samples(Fungi_experiment_two_CSS, Soil_type %in% c("Environmental_soil"));Fungi_experiment_two_CSS_Env#[ 15421 taxa and 1102 samples ]

#First get otu_table and transpose it:
dist.matrix_Fungi_experiment_two_CSS_Env <- t(data.frame(otu_table(Fungi_experiment_two_CSS_Env)))
sampledf_Fungi_experiment_two_CSS_Env <- data.frame(sample_data(Fungi_experiment_two_CSS_Env))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_experiment_two_CSS_Env <- vegdist(dist.matrix_Fungi_experiment_two_CSS_Env, method = "bray"); phyloseq_bray_Fungi_experiment_two_CSS_Env
sampleda <- data.frame(sample_data(Fungi_experiment_two_CSS_Env), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_experiment_two_CSS_Env)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_experiment_two_CSS_Env ~Sample_type*Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#                       Df SumOfSqs      R2      F Pr(>F)    
#Sample_type            2   1.2112 0.20979 5.3222  0.001 ***
#Treatment              1   0.7409 0.12833 6.5113  0.001 ***
#Sample_type:Treatment  2   0.4078 0.07063 1.7920  0.006 ** 
#Residual              30   3.4136 0.59125                  
#Total                 35   5.7736 1.00000                  
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

#Subset the different compartments
#Phyllosphere
Fungi_experiment_two_CSS_Phyllosphere= subset_samples(Fungi_experiment_two_CSS_Env, Sample_type %in% c("4_Phyllosphere"));Fungi_experiment_two_CSS_Phyllosphere#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_experiment_two_CSS_Phyllosphere <- t(data.frame(otu_table(Fungi_experiment_two_CSS_Phyllosphere)))
sampledf_Fungi_experiment_two_CSS_Phyllosphere <- data.frame(sample_data(Fungi_experiment_two_CSS_Phyllosphere))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_experiment_two_CSS_Phyllosphere <- vegdist(dist.matrix_Fungi_experiment_two_CSS_Phyllosphere, method = "bray"); phyloseq_bray_Fungi_experiment_two_CSS_Phyllosphere
sampleda <- data.frame(sample_data(Fungi_experiment_two_CSS_Phyllosphere), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_experiment_two_CSS_Phyllosphere)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_experiment_two_CSS_Phyllosphere ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.57372 0.28914 4.0675  0.004 **
#Residual  10  1.41051 0.71086                 
#Total     11  1.98422 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
#####################################################################################################
#Subset the different compartments
#3_Rhizosphere
Fungi_experiment_two_CSS_3_Rhizosphere= subset_samples(Fungi_experiment_two_CSS_Env, Sample_type %in% c("3_Rhizosphere"));Fungi_experiment_two_CSS_3_Rhizosphere#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_experiment_two_CSS_3_Rhizosphere <- t(data.frame(otu_table(Fungi_experiment_two_CSS_3_Rhizosphere)))
sampledf_Fungi_experiment_two_CSS_3_Rhizosphere <- data.frame(sample_data(Fungi_experiment_two_CSS_3_Rhizosphere))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_experiment_two_CSS_3_Rhizosphere <- vegdist(dist.matrix_Fungi_experiment_two_CSS_3_Rhizosphere, method = "bray"); phyloseq_bray_Fungi_experiment_two_CSS_3_Rhizosphere
sampleda <- data.frame(sample_data(Fungi_experiment_two_CSS_3_Rhizosphere), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_experiment_two_CSS_3_Rhizosphere)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_experiment_two_CSS_3_Rhizosphere ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.36063 0.23211 3.0228  0.004 **
#Residual  10  1.19305 0.76789                 
#Total     11  1.55368 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1  
#####################################################################################################
#Subset the different compartments
#Soil
Fungi_experiment_two_CSS_Soil= subset_samples(Fungi_experiment_two_CSS_Env, Sample_type %in% c("2_Soil"));Fungi_experiment_two_CSS_Soil#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_experiment_two_CSS_Soil <- t(data.frame(otu_table(Fungi_experiment_two_CSS_Soil)))
sampledf_Fungi_experiment_two_CSS_Soil <- data.frame(sample_data(Fungi_experiment_two_CSS_Soil))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_experiment_two_CSS_Soil <- vegdist(dist.matrix_Fungi_experiment_two_CSS_Soil, method = "bray"); phyloseq_bray_Fungi_experiment_two_CSS_Soil
sampleda <- data.frame(sample_data(Fungi_experiment_two_CSS_Soil), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_experiment_two_CSS_Soil)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_experiment_two_CSS_Soil ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.21438 0.20926 2.6464  0.006 **
#Residual  10  0.81007 0.79074                 
#Total     11  1.02445 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
#####################################################################################################
#Potting soil
Fungi_experiment_two_CSS_Pott= subset_samples(Fungi_experiment_two_CSS, Soil_type %in% c("Potting_soil"));Fungi_experiment_two_CSS_Pott#[ 15421 taxa and 1102 samples ]

#First get otu_table and transpose it:
dist.matrix_Fungi_experiment_two_CSS_Pott <- t(data.frame(otu_table(Fungi_experiment_two_CSS_Pott)))
sampledf_Fungi_experiment_two_CSS_Pott <- data.frame(sample_data(Fungi_experiment_two_CSS_Pott))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_experiment_two_CSS_Pott <- vegdist(dist.matrix_Fungi_experiment_two_CSS_Pott, method = "bray"); phyloseq_bray_Fungi_experiment_two_CSS_Pott
sampleda <- data.frame(sample_data(Fungi_experiment_two_CSS_Pott), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_experiment_two_CSS_Pott)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_experiment_two_CSS_Pott ~Sample_type*Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#                       Df SumOfSqs      R2      F Pr(>F)    
#Sample_type            2   1.2025 0.25877 7.0818  0.001 ***
#Treatment              1   0.4056 0.08729 4.7778  0.001 ***
#Sample_type:Treatment  2   0.4918 0.10583 2.8963  0.001 ***
#Residual              30   2.5470 0.54811                  
#Total                 35   4.6468 1.00000                  
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

#Subset the different compartments
#Phyllosphere
Fungi_experiment_two_CSS_Phyllosphere= subset_samples(Fungi_experiment_two_CSS_Pott, Sample_type %in% c("4_Phyllosphere"));Fungi_experiment_two_CSS_Phyllosphere#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_experiment_two_CSS_Phyllosphere <- t(data.frame(otu_table(Fungi_experiment_two_CSS_Phyllosphere)))
sampledf_Fungi_experiment_two_CSS_Phyllosphere <- data.frame(sample_data(Fungi_experiment_two_CSS_Phyllosphere))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_experiment_two_CSS_Phyllosphere <- vegdist(dist.matrix_Fungi_experiment_two_CSS_Phyllosphere, method = "bray"); phyloseq_bray_Fungi_experiment_two_CSS_Phyllosphere
sampleda <- data.frame(sample_data(Fungi_experiment_two_CSS_Phyllosphere), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_experiment_two_CSS_Phyllosphere)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_experiment_two_CSS_Phyllosphere ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1   0.5409 0.27498 3.7927  0.003 **
#Residual  10   1.4261 0.72502                 
#Total     11   1.9670 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
#####################################################################################################
#Subset the different compartments
#3_Rhizosphere
Fungi_experiment_two_CSS_3_Rhizosphere= subset_samples(Fungi_experiment_two_CSS_Pott, Sample_type %in% c("3_Rhizosphere"));Fungi_experiment_two_CSS_3_Rhizosphere#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_experiment_two_CSS_3_Rhizosphere <- t(data.frame(otu_table(Fungi_experiment_two_CSS_3_Rhizosphere)))
sampledf_Fungi_experiment_two_CSS_3_Rhizosphere <- data.frame(sample_data(Fungi_experiment_two_CSS_3_Rhizosphere))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_experiment_two_CSS_3_Rhizosphere <- vegdist(dist.matrix_Fungi_experiment_two_CSS_3_Rhizosphere, method = "bray"); phyloseq_bray_Fungi_experiment_two_CSS_3_Rhizosphere
sampleda <- data.frame(sample_data(Fungi_experiment_two_CSS_3_Rhizosphere), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_experiment_two_CSS_3_Rhizosphere)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_experiment_two_CSS_3_Rhizosphere ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)   
#Treatment  1  0.19320 0.25944 3.5033  0.004 **
#Residual  10  0.55149 0.74056                 
#Total     11  0.74470 1.00000                 
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1 
#####################################################################################################
#Subset the different compartments
#Soil
Fungi_experiment_two_CSS_Soil= subset_samples(Fungi_experiment_two_CSS_Pott, Sample_type %in% c("2_Soil"));Fungi_experiment_two_CSS_Soil#[ 15421 taxa and 1102 samples ]
dist.matrix_Fungi_experiment_two_CSS_Soil <- t(data.frame(otu_table(Fungi_experiment_two_CSS_Soil)))
sampledf_Fungi_experiment_two_CSS_Soil <- data.frame(sample_data(Fungi_experiment_two_CSS_Soil))
#Then use vegdist from vegan to generate a bray distance object:
phyloseq_bray_Fungi_experiment_two_CSS_Soil <- vegdist(dist.matrix_Fungi_experiment_two_CSS_Soil, method = "bray"); phyloseq_bray_Fungi_experiment_two_CSS_Soil
sampleda <- data.frame(sample_data(Fungi_experiment_two_CSS_Soil), check.names=FALSE);sampleda
sampleda <- sampleda[match(colnames(as.matrix(phyloseq_bray_Fungi_experiment_two_CSS_Soil)),rownames(sampleda)),,drop=FALSE];sampleda
sampleda$Sample_type <- factor(sampleda$Sample_type)
adores <- adonis2(phyloseq_bray_Fungi_experiment_two_CSS_Soil ~ Treatment, data=sampleda, permutation=999);adores#change from adonis to adonis2
#           Df SumOfSqs      R2      F Pr(>F)    
#Treatment  1  0.16330 0.22289 2.8683  0.001 ***
#Residual  10  0.56933 0.77711                  
#Total     11  0.73262 1.00000                  
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
#####################################################################################################

```

